{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "163da59f-cf9c-4665-8221-8524f6a364e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "import numpy as np\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac49c984-53cf-4aca-9b10-2c20e1321d8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "idat_path = '/data/CARD/PD/GP2/raw_genotypes/GP2_idats'\n",
    "swarm_scripts_dir = f'/data/CARD/PD/GP2/swarm_scripts'\n",
    "\n",
    "ilmn_files_path = '/data/CARD/PD/GP2/ilmn_files'\n",
    "# A1 is hg19, A2 is hg38. csv needed for indel calls\n",
    "bpm_csv = f'{ilmn_files_path}/NeuroBooster_20042459_A2.csv' \n",
    "bpm = f'{ilmn_files_path}/NeuroBooster_20042459_A2.bpm'\n",
    "egt = f'{ilmn_files_path}/recluster_09272022.egt'\n",
    "iaap = f'{ilmn_files_path}/iaap-cli/iaap-cli'\n",
    "\n",
    "ref_fasta = '/data/vitaled2/ref/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna'\n",
    "\n",
    "chroms = [str(i) for i in range(1,23)] + ['X','Y','M']\n",
    "\n",
    "\n",
    "key_file = '/data/CARD/PD/GP2/clinical/key_merge/GP2_merge/GP2_master_key_full.txt'\n",
    "key = pd.read_csv(f'{key_file}', sep='\\t')\n",
    "key.loc[:,'filename'] = key.loc[:,'SentrixBarcode_A'].astype(str) + '_' + key.loc[:,'SentrixPosition_A'].astype(str)\n",
    "cohorts = [\n",
    "    'SYNAPS-KZ', \n",
    "    'CORIELL', \n",
    "    'IPDGCEA-JU', \n",
    "    'KOC', \n",
    "    'IPDGCAF-NG', \n",
    "    'IPDGCEA-MU', \n",
    "    'PAGE', \n",
    "    'NZP3', \n",
    "    'S4', \n",
    "    'BCM',\n",
    "    'BLAACPD-RUSH', \n",
    "    'BIOFIND', \n",
    "    'UMD', \n",
    "    'BLAACPD-UAB', \n",
    "    'LCC', \n",
    "    'SYNAPS-GR', \n",
    "    'MDGAP-QSBB', \n",
    "    'APGS', \n",
    "    'MDGAP-KINGS', \n",
    "    'MDGAP-IBB', \n",
    "    'BLAACPD-UC', \n",
    "    'BLAACPD-KPM', \n",
    "    'PPMI', \n",
    "    'IPDGCEA-UKM'\n",
    "]\n",
    "\n",
    "release_key = key.loc[key['study'].isin(cohorts)]\n",
    "release_key.to_csv('/data/CARD/PD/GP2/clinical/key_merge/GP2_merge/release3_key.csv', index=False, header=True)\n",
    "\n",
    "samples_list = release_key.loc[:,'filename'].unique()\n",
    "barcodes_list = list(set([x.split('_')[0] for x in samples_list]))\n",
    "\n",
    "genes = pd.read_csv('/data/CARD/PD/GP2/ref_panel/glist-hg38', sep='\\s+', header=None, names=['chr','start','end','symbol'], dtype={'chr':str,'start':int,'end':int})\n",
    "genes.columns = ['CHR','START','STOP','NAME']\n",
    "genes_out = genes.loc[genes.CHR.isin(chroms)]\n",
    "genes_out[['NAME','CHR','START','STOP']].to_csv('/data/CARD/PD/GP2/ref_panel/glist_hg38_intervals.csv', index=False)\n",
    "# gene_list originally from plink glist\n",
    "gene_list = '/data/CARD/PD/GP2/ref_panel/glist_hg38_intervals.csv'\n",
    "\n",
    "raw_geno_path = '/data/CARD/PD/GP2/raw_genotypes'\n",
    "snp_metrics_path = f'{raw_geno_path}/GP2_snp_metrics'\n",
    "idat_path = f'{raw_geno_path}/GP2_idats'\n",
    "plink_file_path = f'{raw_geno_path}/GP2_plink'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59346c9d-e1bd-47fd-b503-c9f592ff4dd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(samples_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d146821f-30ff-496e-a9ad-bd2e1b90ac08",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get snp metrics using new cluster file\n",
    "bcftools_plugins_path = '/data/vitaled2/bin'\n",
    "with open(f'{swarm_scripts_dir}/snp_metrics.swarm', 'w') as f:\n",
    "    for code in barcodes_list:\n",
    "        idat_path_ = f'{idat_path}/{code}'\n",
    "        metrics_out = f'{snp_metrics_path}'\n",
    "\n",
    "        cmd = f'\\\n",
    "python /data/vitaled2/GenoTools/run_snp_metrics_pipeline.py \\\n",
    "--idat_path {idat_path_} \\\n",
    "--bpm {bpm} \\\n",
    "--bpm_csv {bpm_csv} \\\n",
    "--egt {egt} \\\n",
    "--ref_fasta {ref_fasta} \\\n",
    "--out_path {metrics_out} \\\n",
    "--iaap {iaap} \\\n",
    "--bcftools_plugins_path {bcftools_plugins_path}'\n",
    "        f.write(f'{cmd}\\n')\n",
    "f.close()\n",
    "\n",
    "!swarm -f {swarm_scripts_dir}/snp_metrics.swarm -g 16 -t 16 --time=04:00:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cffc06e3-dfa9-4654-b599-d63c485b3b28",
   "metadata": {},
   "outputs": [],
   "source": [
    "# check if snp metrics produced across all samples\n",
    "completed_list = []\n",
    "missing_list = []\n",
    "for sample in samples_list:\n",
    "    code = sample.split('_')[0]\n",
    "    metrics_path = f'{snp_metrics_path}/{code}'\n",
    "    for chrom in chroms:\n",
    "        mfile = f'{metrics_path}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "        if os.path.isfile(mfile):\n",
    "            completed_list.append(sample)\n",
    "        else:\n",
    "            missing_list.append(sample)\n",
    "completed_final = list(set(completed_list))\n",
    "missing_final =list(set(missing_list))\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7c4dff8-569a-4d46-b4f2-186363aacb39",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(len(completed_final))\n",
    "print(len(missing_final))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcd9284a-4a05-46d9-892e-6bda06354195",
   "metadata": {},
   "outputs": [],
   "source": [
    "# now check if we have idats for missing snp metrics\n",
    "missing_idats = []\n",
    "exists_idats = []\n",
    "for sample in missing_final:\n",
    "    code = sample.split('_')[0]\n",
    "    grn = f'{idat_path}/{code}/{sample}_Grn.idat'\n",
    "    red = f'{idat_path}/{code}/{sample}_Red.idat'\n",
    "    if os.path.isfile(grn) & os.path.isfile(red):\n",
    "        exists_idats.append(sample)\n",
    "    else:\n",
    "        missing_idats.append(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc859aa9-0359-427f-ae76-0f0fb2a4596e",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(len(missing_idats))\n",
    "print(len(exists_idats))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a949c9a-3b34-422e-bc96-b224e6b86681",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_path = f'/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3'\n",
    "# using just autosomes for CNV analysis due to call quality\n",
    "chroms = [str(i) for i in range(1,23)]\n",
    "\n",
    "ancestry_labels = [x.split('/')[-1].replace('.bed','').split('_')[-1] for x in glob.glob('/data/CARD/PD/GP2/genotypes/GP2/round3/clean/GP2_round3_NEW_MERGE*.bed') if '_maf_hwe' not in x]\n",
    "\n",
    "release_key[['FID','GP2sampleID']].to_csv(f'{cnv_path}/release3.samples', sep='\\t', header=False, index=False)\n",
    "release_key[['GP2sampleID','IID']].to_csv(f'{cnv_path}/release3_sample_id_key.csv')\n",
    "release_covars = release_key.loc[:,['FID', 'GP2sampleID','sex_for_qc', 'age', 'age_of_onset']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea017911-2cff-4adf-9952-8bb3b71f5a14",
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp QC for cnvs\n",
    "for label in ancestry_labels:\n",
    "    out_name = f'GP2_round3_NEW_MERGE_{label}'\n",
    "    out_path = f'{cnv_path}/{out_name}'\n",
    "    geno_path = f'/data/CARD/PD/GP2/genotypes/GP2/round3/clean/{out_name}'\n",
    "\n",
    "    cmd1 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {geno_path} \\\n",
    "--keep {cnv_path}/release3.samples \\\n",
    "--make-bed \\\n",
    "--out {out_path}'\n",
    "\n",
    "    cmd2 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {out_path} \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.02 \\\n",
    "--hwe 5e-6 \\\n",
    "--autosome \\\n",
    "--make-bed \\\n",
    "--out {out_path}_clean'\n",
    "\n",
    "    cmd3 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {out_path}_clean \\\n",
    "--pca \\\n",
    "--out {out_path}_clean_pcs'\n",
    "\n",
    "    cmds = [cmd1, cmd2, cmd3]\n",
    "\n",
    "    for cmd in cmds:\n",
    "        !{cmd}\n",
    "\n",
    "    pcs = pd.read_csv(f'{out_path}_clean_pcs.eigenvec', sep='\\s+')\n",
    "    pc_num = pcs.iloc[:, 2:].shape[1]\n",
    "    pc_names = ['FID','GP2sampleID'] + [f'PC{i}' for i in range(1, pc_num+1)]\n",
    "    pcs.columns = pc_names\n",
    "\n",
    "    cov = pcs.merge(release_covars, on=['FID','GP2sampleID'], how='left')\n",
    "    cov.age.fillna(cov.age.mean(), inplace=True)\n",
    "    cov.age_of_onset.fillna(cov.age_of_onset.mean(), inplace=True)\n",
    "    cov.sex_for_qc.fillna(cov.sex_for_qc.median(), inplace=True)\n",
    "    cov.rename(columns={'GP2sampleID':'sampleid','sex_for_qc':'sex'}, inplace=True)\n",
    "    cov.to_csv(f'{out_path}_clean.cov', sep='\\t', header=True, index=False)\n",
    "\n",
    "    samples = cov.merge(release_key[['GP2sampleID','IID']], left_on='sampleid', right_on='GP2sampleID', how='left')\n",
    "    samples['IID'].to_csv(f'{out_path}_clean_barcode.samples', header=False, index=False)\n",
    "\n",
    "    bim = pd.read_csv(f'{out_path}_clean.bim', sep='\\s+', header=None, names=['chr','id','pos','bp','a1','a2'], usecols=['id'])\n",
    "    bim.to_csv(f'{out_path}_clean.snps', sep='\\t', header=False, index=False)\n",
    "        \n",
    "\n",
    "        \n",
    "# skip ld pruning until we have better quality rare calls\n",
    "# --exclude exclusion_regions_hg38.txt \\\n",
    "\n",
    "\n",
    "#         cmd3 = f'\\\n",
    "# plink \\\n",
    "# --bfile {geno}_release2_maf_geno_hwe \\\n",
    "# --indep-pairwise 1000 10 0.02 \\\n",
    "# --autosome \\\n",
    "# --out {geno}_release2_prune'\n",
    "    \n",
    "#         cmd4 = f'\\\n",
    "# plink \\\n",
    "# --bfile {geno}_release2_maf_geno_hwe \\\n",
    "# --extract {geno}_release2_prune.prune.in \\\n",
    "# --make-bed \\\n",
    "# --out {geno}_release2_clean'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d716faf8-48f8-4a74-9c90-8046a49bbb5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/cnvs.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        out_name = f'GP2_round3_NEW_MERGED_{label}'\n",
    "        out_path = f'{cnv_path}/{out_name}'\n",
    "#         geno_path = f'/data/CARD/PD/GP2/genotypes/GP2/round3/clean/{out_name}'\n",
    "        samples = pd.read_csv(f'{out_path}_clean_barcode.samples', header=None, names=['IID'])\n",
    "        \n",
    "        for sample in samples.IID.unique():\n",
    "            code = sample.split('_')[0]\n",
    "            for chrom in chroms:\n",
    "            \n",
    "                mfile = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "\n",
    "                out_file = f'{cnv_path}/CNV_{label}_{sample}_chr{chrom}.csv'\n",
    "                intervals = f'/data/CARD/PD/GP2/ref_panel/glist_hg38_chr{chrom}.csv'\n",
    "\n",
    "                cmd = f'\\\n",
    "python /data/vitaled2/GenoTools/run_cnv_pipeline.py \\\n",
    "--metrics {mfile} \\\n",
    "--out_path {out_file} \\\n",
    "--intervals {intervals} \\\n",
    "--min_variants 10 \\\n",
    "--kb_window 250'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa7199d9-3b87-4d27-832a-28f98b0dd800",
   "metadata": {},
   "outputs": [],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnvs.swarm -g 4 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75f31342-5241-48f9-bdf5-e73b1c46798d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# write dosages per chromosome per ancestry per cnv-type\n",
    "cnv_types = ['PERCENT_BAF_INSERTION','PERCENT_L2R_DELETION','PERCENT_L2R_DUPLICATION']\n",
    "\n",
    "with open(f'{swarm_scripts_dir}/cnv_dosages.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        for chrom in chroms:\n",
    "\n",
    "            cnv_files_list = glob.glob(f'{cnv_path}/CNV_{label}_*_chr{chrom}.csv')\n",
    "            cnv_files_df = pd.DataFrame({'filename':cnv_files_list})\n",
    "            cnv_files_df.to_csv(f'{cnv_path}/CNV_{label}_chr{chrom}_files.csv', header=False, index=False)\n",
    "\n",
    "            for cnv_type in cnv_types:\n",
    "                \n",
    "                cmd = f'\\\n",
    "python /data/vitaled2/GenoTools/run_cnv_dosage_pipeline.py \\\n",
    "--files {cnv_path}/CNV_{label}_chr{chrom}_files.csv \\\n",
    "--label {label} \\\n",
    "--chrom {chrom} \\\n",
    "--cnv_type {cnv_type} \\\n",
    "--out_path {cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n",
    "            \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5bce1bd2-a140-47d7-93cb-f6515adea344",
   "metadata": {},
   "outputs": [],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_dosages.swarm -g 32 -t 16 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8eaefe81-9ec6-4179-9f2c-776ba966b8ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/update_cnv_ids.swarm','w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        for dose in cnv_types:\n",
    "            for chrom in chroms:\n",
    "\n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                dosagefile_out = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                pheno_out = f'{cnv_path}/GP2_round2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "                cmd = f'\\\n",
    "python /data/vitaled2/GenoTools/update_cnv_ids.py \\\n",
    "--dosagefile {dosagefile} \\\n",
    "--key /data/CARD/PD/GP2/clinical/key_merge/GP2_merge/release3_key.csv \\\n",
    "--pheno_out {pheno_out} \\\n",
    "--out_path {dosagefile_out}'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b3a4dec-9b45-40ac-a1bc-0c51090c2819",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34286258-bfab-4a43-b046-eea65b7cf414",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d8fab9c-8631-496f-9d04-61735788cda5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15c41a1d-f86b-4001-968f-613ea87793e7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea2ad12d-590a-4a16-9569-49d447cf37b3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3419a76-ba11-49a0-913d-f34b85c0ba61",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ff40bf6-6316-43f4-bf4c-e8a950006191",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ecfa17e5-a635-487f-8f47-7a6527f24e94",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce5f6a9d-132b-45b4-92f2-32c2dcb73f76",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00ff6fa4-9530-4737-a080-99e5c9afcd56",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
