{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "163da59f-cf9c-4665-8221-8524f6a364e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink is found\n",
      "Plink2 is found\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "import numpy as np\n",
    "import glob\n",
    "from sklearn.preprocessing import MinMaxScaler\n",
    "import statsmodels.api as sm\n",
    "\n",
    "from QC.utils import shell_do"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ac49c984-53cf-4aca-9b10-2c20e1321d8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "############ SET THIS EVERY RELEASE ############\n",
    "release = '5'\n",
    "\n",
    "key_file = f'/data/GP2/clinical/master_key/GP2_master_key_FINAL_release5_n26728.txt'\n",
    "key = pd.read_csv(f'{key_file}')\n",
    "key.loc[:,'IID'] = key.loc[:,'SentrixBarcode_A'].astype(str) + '_' + key.loc[:,'SentrixPosition_A'].astype(str)\n",
    "key.loc[:,'FID'] = \"0\"\n",
    "\n",
    "\n",
    "idat_path = '/data/GP2/raw_genotypes/idats/'\n",
    "swarm_scripts_dir = f'/data/GP2/users/vitaled2/swarm'\n",
    "\n",
    "ilmn_files_path = '/data/GP2/utils/ilmn_files'\n",
    "# A1 is hg19, A2 is hg38. csv needed for indel calls\n",
    "bpm_csv = f'{ilmn_files_path}/NeuroBooster_20042459_A2.csv' \n",
    "bpm = f'{ilmn_files_path}/NeuroBooster_20042459_A2.bpm'\n",
    "egt = f'{ilmn_files_path}/recluster_09272022.egt'\n",
    "iaap = f'{ilmn_files_path}/iaap-cli/iaap-cli'\n",
    "\n",
    "ref_fasta = '/data/vitaled2/ref/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna'\n",
    "\n",
    "chroms = [str(i) for i in range(1,23)] + ['X','Y','M']\n",
    "\n",
    "samples_list = key.loc[:,'filename'].unique()\n",
    "barcodes_list = list(set([x.split('_')[0] for x in samples_list]))\n",
    "\n",
    "################################################################################################################\n",
    "\n",
    "# genes = pd.read_csv('/data/GP2/utils/ref_dir/glist-hg38', sep='\\s+', header=None, names=['chr','start','end','symbol'], dtype={'chr':str,'start':int,'end':int})\n",
    "# genes.columns = ['CHR','START','STOP','NAME']\n",
    "# genes_out = genes.loc[genes.CHR.isin(chroms)]\n",
    "# genes_out[['NAME','CHR','START','STOP']].to_csv('/data/GP2/utils/ref_dir/glist_hg38_intervals.csv', index=False)\n",
    "# # gene_list originally from plink glist\n",
    "# gene_list = '/data/GP2/utils/ref_dir/glist_hg38_intervals.csv'\n",
    "# !cp /data/CARD/PD/GP2/ref_panel/glist_hg38_chr*.csv /data/GP2/utils/ref_dir/\n",
    "\n",
    "raw_geno_path = '/data/GP2/raw_genotypes'\n",
    "snp_metrics_path = f'{raw_geno_path}/snp_metrics'\n",
    "idat_path = f'{raw_geno_path}/idats'\n",
    "plink_file_path = f'{raw_geno_path}/ped_bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "59346c9d-e1bd-47fd-b503-c9f592ff4dd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GP2ID</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>manifest_id</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>pheno_for_qc</th>\n",
       "      <th>other_pheno</th>\n",
       "      <th>sex_for_qc</th>\n",
       "      <th>age</th>\n",
       "      <th>age_of_onset</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>study</th>\n",
       "      <th>pruned</th>\n",
       "      <th>pruned_reason</th>\n",
       "      <th>label</th>\n",
       "      <th>related</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>filename</th>\n",
       "      <th>IID</th>\n",
       "      <th>FID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IMMUNEPD_000001</td>\n",
       "      <td>IMMUNEPD_000001_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>2</td>\n",
       "      <td>PD</td>\n",
       "      <td>1</td>\n",
       "      <td>61.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>61.0</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R01C01</td>\n",
       "      <td>206966380021_R01C01</td>\n",
       "      <td>206966380021_R01C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IMMUNEPD_000002</td>\n",
       "      <td>IMMUNEPD_000002_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>2</td>\n",
       "      <td>PD</td>\n",
       "      <td>1</td>\n",
       "      <td>66.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>66.0</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>206966380021_R02C01</td>\n",
       "      <td>206966380021_R02C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IMMUNEPD_000003</td>\n",
       "      <td>IMMUNEPD_000003_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>55.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R03C01</td>\n",
       "      <td>206966380021_R03C01</td>\n",
       "      <td>206966380021_R03C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IMMUNEPD_000004</td>\n",
       "      <td>IMMUNEPD_000004_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>50.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>206966380021_R04C01</td>\n",
       "      <td>206966380021_R04C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IMMUNEPD_000005</td>\n",
       "      <td>IMMUNEPD_000005_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>74.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>206966380021_R05C01</td>\n",
       "      <td>206966380021_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26723</th>\n",
       "      <td>MDGAP-EBB_000149</td>\n",
       "      <td>MDGAP-EBB_000149_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>206412050061_R04C01</td>\n",
       "      <td>206412050061_R04C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26724</th>\n",
       "      <td>MDGAP-EBB_000150</td>\n",
       "      <td>MDGAP-EBB_000150_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>206412050061_R05C01</td>\n",
       "      <td>206412050061_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26725</th>\n",
       "      <td>MDGAP-EBB_000151</td>\n",
       "      <td>MDGAP-EBB_000151_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>206412050061_R06C01</td>\n",
       "      <td>206412050061_R06C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26726</th>\n",
       "      <td>MDGAP-EBB_000152</td>\n",
       "      <td>MDGAP-EBB_000152_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R07C01</td>\n",
       "      <td>206412050061_R07C01</td>\n",
       "      <td>206412050061_R07C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26727</th>\n",
       "      <td>MDGAP-EBB_000153</td>\n",
       "      <td>MDGAP-EBB_000153_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R08C01</td>\n",
       "      <td>206412050061_R08C01</td>\n",
       "      <td>206412050061_R08C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26728 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  GP2ID          GP2sampleID manifest_id phenotype  \\\n",
       "0       IMMUNEPD_000001   IMMUNEPD_000001_s1          m1        PD   \n",
       "1       IMMUNEPD_000002   IMMUNEPD_000002_s1          m1        PD   \n",
       "2       IMMUNEPD_000003   IMMUNEPD_000003_s1          m1   Control   \n",
       "3       IMMUNEPD_000004   IMMUNEPD_000004_s1          m1   Control   \n",
       "4       IMMUNEPD_000005   IMMUNEPD_000005_s1          m1   Control   \n",
       "...                 ...                  ...         ...       ...   \n",
       "26723  MDGAP-EBB_000149  MDGAP-EBB_000149_s1          m4   Control   \n",
       "26724  MDGAP-EBB_000150  MDGAP-EBB_000150_s1          m4   Control   \n",
       "26725  MDGAP-EBB_000151  MDGAP-EBB_000151_s1          m4   Control   \n",
       "26726  MDGAP-EBB_000152  MDGAP-EBB_000152_s1          m4   Control   \n",
       "26727  MDGAP-EBB_000153  MDGAP-EBB_000153_s1          m4   Control   \n",
       "\n",
       "       pheno_for_qc other_pheno  sex_for_qc   age  age_of_onset  \\\n",
       "0                 2          PD           1  61.0           NaN   \n",
       "1                 2          PD           1  66.0           NaN   \n",
       "2                 1     Control           2  55.0           NaN   \n",
       "3                 1     Control           1  50.0           NaN   \n",
       "4                 1     Control           2  74.0           NaN   \n",
       "...             ...         ...         ...   ...           ...   \n",
       "26723             1     Control           1   NaN           NaN   \n",
       "26724             1     Control           2   NaN           NaN   \n",
       "26725             1     Control           1   NaN           NaN   \n",
       "26726             1     Control           2   NaN           NaN   \n",
       "26727             1     Control           1   NaN           NaN   \n",
       "\n",
       "       age_at_diagnosis  ...      study pruned   pruned_reason label related  \\\n",
       "0                  61.0  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "1                  66.0  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "2                   NaN  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "3                   NaN  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "4                   NaN  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "...                 ...  ...        ...    ...             ...   ...     ...   \n",
       "26723               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26724               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26725               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26726               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26727               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "\n",
       "       SentrixBarcode_A SentrixPosition_A             filename  \\\n",
       "0          206966380021            R01C01  206966380021_R01C01   \n",
       "1          206966380021            R02C01  206966380021_R02C01   \n",
       "2          206966380021            R03C01  206966380021_R03C01   \n",
       "3          206966380021            R04C01  206966380021_R04C01   \n",
       "4          206966380021            R05C01  206966380021_R05C01   \n",
       "...                 ...               ...                  ...   \n",
       "26723      206412050061            R04C01  206412050061_R04C01   \n",
       "26724      206412050061            R05C01  206412050061_R05C01   \n",
       "26725      206412050061            R06C01  206412050061_R06C01   \n",
       "26726      206412050061            R07C01  206412050061_R07C01   \n",
       "26727      206412050061            R08C01  206412050061_R08C01   \n",
       "\n",
       "                       IID  FID  \n",
       "0      206966380021_R01C01    0  \n",
       "1      206966380021_R02C01    0  \n",
       "2      206966380021_R03C01    0  \n",
       "3      206966380021_R04C01    0  \n",
       "4      206966380021_R05C01    0  \n",
       "...                    ...  ...  \n",
       "26723  206412050061_R04C01    0  \n",
       "26724  206412050061_R05C01    0  \n",
       "26725  206412050061_R06C01    0  \n",
       "26726  206412050061_R07C01    0  \n",
       "26727  206412050061_R08C01    0  \n",
       "\n",
       "[26728 rows x 24 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "d146821f-30ff-496e-a9ad-bd2e1b90ac08",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get snp metrics\n",
    "metrics_exist_list = []\n",
    "bcftools_plugins_path = 'bin'\n",
    "with open(f'{swarm_scripts_dir}/snp_metrics.swarm', 'w') as f:\n",
    "    for code in barcodes_list:\n",
    "        idat_path_ = f'{idat_path}/{code}'\n",
    "        metrics_out = f'{snp_metrics_path}'\n",
    "        # metrics_code_path = f'{snp_metrics_path}/{code}'\n",
    "        cmd = f'\\\n",
    "python run_snp_metrics_pipeline.py \\\n",
    "--idat_path {idat_path_} \\\n",
    "--bpm {bpm} \\\n",
    "--bpm_csv {bpm_csv} \\\n",
    "--egt {egt} \\\n",
    "--ref_fasta {ref_fasta} \\\n",
    "--out_path {metrics_out} \\\n",
    "--iaap {iaap} \\\n",
    "--bcftools_plugins_path {bcftools_plugins_path}'\n",
    "        f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "117e910d-509a-4ff5-af24-e5f3d07da1fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "63708079\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/snp_metrics.swarm -g 16 -t 16 --time=04:00:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a63c4a2e-b1c6-4cd0-a72d-cd99f60f1106",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr2 metrics exist: 26578\n",
      "chr2 missing metrics: 150\n",
      "samples with all chroms: 26578.0\n",
      "chr files missing from samples with chr2 0\n"
     ]
    }
   ],
   "source": [
    "#check if output for chr2 exists for all samples \n",
    "exist_metrics = []\n",
    "missing_metrics = []\n",
    "\n",
    "for sample in samples_list:\n",
    "    code = sample.split('_')[0]\n",
    "    metrics_file = f'/data/GP2/raw_genotypes/snp_metrics/{code}/snp_metrics_{sample}_chr2.parquet'\n",
    "    \n",
    "    if os.path.isfile(metrics_file):\n",
    "        exist_metrics.append(sample)\n",
    "    else:\n",
    "        missing_metrics.append(sample)\n",
    "\n",
    "chroms = [x for x in range(1,23)]\n",
    "exist_chrom = []\n",
    "missing_chrom = []\n",
    "# check if all chromosomes exist for samples for which chr2 exists\n",
    "for sample in exist_metrics:\n",
    "    code = sample.split('_')[0]\n",
    "    for chrom in chroms:\n",
    "        \n",
    "        metrics_path = f'/data/GP2/raw_genotypes/snp_metrics/{code}/snp_metrics_{sample}_chr{chrom}.parquet'\n",
    "        \n",
    "        if os.path.isfile(metrics_path):\n",
    "            exist_chrom.append(metrics_path)\n",
    "        else:\n",
    "            missing_chrom.append(metrics_path)\n",
    "    \n",
    "    \n",
    "\n",
    "        \n",
    "print(f'chr2 metrics exist: {len(exist_metrics)}')\n",
    "print(f'chr2 missing metrics: {len(missing_metrics)}')\n",
    "print(f'samples with all chroms: {len(exist_chrom)/22}')\n",
    "print(f'chr files missing from samples with chr2 {len(missing_chrom)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bcd9284a-4a05-46d9-892e-6bda06354195",
   "metadata": {},
   "outputs": [],
   "source": [
    "# now check if we have idats for missing snp metrics\n",
    "missing_idats = []\n",
    "exists_idats = []\n",
    "for sample in missing_metrics:\n",
    "    code = sample.split('_')[0]\n",
    "    grn = f'{idat_path}/{code}/{sample}_Grn.idat'\n",
    "    red = f'{idat_path}/{code}/{sample}_Red.idat'\n",
    "    if os.path.isfile(grn) & os.path.isfile(red):\n",
    "        exists_idats.append(sample)\n",
    "    else:\n",
    "        missing_idats.append(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cc859aa9-0359-427f-ae76-0f0fb2a4596e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "93\n",
      "57\n"
     ]
    }
   ],
   "source": [
    "print(len(missing_idats))\n",
    "print(len(exists_idats))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8a949c9a-3b34-422e-bc96-b224e6b86681",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_path = f'/data/GP2/raw_genotypes/cnvs/release{release}'\n",
    "covar_path = f'{cnv_path}/key/release{release}_covar.csv'\n",
    "# using just autosomes for CNV analysis due to call quality\n",
    "chroms = [str(i) for i in range(1,23)]\n",
    "\n",
    "ancestry_labels = [x.split('/')[-1].replace('.bed','').split('_')[-1] for x in glob.glob(f'/data/GP2/quality_control/release{release}/genotype_qc/GP2_round5_APRIL_2023_*.bed') if '_maf_hwe' not in x]\n",
    "\n",
    "key[['FID','GP2sampleID']].to_csv(f'{cnv_path}/key/release{release}.samples', sep='\\t', header=False, index=False)\n",
    "key[['GP2sampleID','IID']].to_csv(f'{cnv_path}/key/release{release}_sample_id_key.csv')\n",
    "release_covars = key.loc[:,['FID', 'GP2sampleID','sex_for_qc', 'age', 'age_of_onset']]\n",
    "release_covars.to_csv(covar_path)\n",
    "\n",
    "cnv_types = ['PERCENT_BAF_INSERTION','PERCENT_L2R_DELETION','PERCENT_L2R_DUPLICATION']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "eb84cf01-270c-429f-8758-d700fe19740a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_FIN --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_FIN_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_EUR --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_EUR_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_EAS --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_EAS_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_AAC --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AAC_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_AJ --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AJ_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_MDE --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_MDE_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_SAS --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_SAS_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_CAS --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_CAS_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_AMR --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AMR_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n",
      "python run_cnv_qc_pipeline.py --geno_path /data/GP2/quality_control/release5/genotype_qc/GP2_round5_APRIL_2023_AFR --out_path /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AFR_clean --covar_path /data/GP2/raw_genotypes/cnvs/release5/key/release5_covar.csv --maf 0.01 --geno 0.2 --hwe 5e-6 --indep_pairwise 1000 10 0.01 --samples_path /data/GP2/raw_genotypes/cnvs/release5/key/release5.samples\n"
     ]
    }
   ],
   "source": [
    "with open(f'{swarm_scripts_dir}/cnv_geno_qc.swarm', 'w') as f:\n",
    "\n",
    "    for label in ancestry_labels:\n",
    "        geno_name = f'GP2_round5_APRIL_2023_{label}'\n",
    "        out_path = f'{cnv_path}/genotypes/{geno_name}_clean'\n",
    "        geno_path = f'/data/GP2/quality_control/release{release}/genotype_qc/{geno_name}'\n",
    "    \n",
    "        cmd = f'\\\n",
    "python run_cnv_qc_pipeline.py \\\n",
    "--geno_path {geno_path} \\\n",
    "--out_path {out_path} \\\n",
    "--covar_path {covar_path} \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.2 \\\n",
    "--hwe 5e-6 \\\n",
    "--indep_pairwise 1000 10 0.01 \\\n",
    "--samples_path {cnv_path}/key/release{release}.samples'\n",
    "        \n",
    "        f.write(f'{cmd}\\n')\n",
    "f.close()\n",
    "!cat {swarm_scripts_dir}/cnv_geno_qc.swarm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b3362f7d-e67e-4b3d-b692-06f7d5a4c421",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "63958236\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_geno_qc.swarm -g 16 -t 16 --time=02:00:00 --logdir {swarm_scripts_dir}/logs --module python/3.8,plink/1.9 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "aeb8e8f0-3185-4f74-83a1-c2d0f153d205",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_FIN_clean.bed\n",
      "fam: ['16']\n",
      "cov: ['16']\n",
      "eigenvec: ['16']\n",
      "bim: ['1152']\n",
      "snps: ['1152']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_EUR_clean.bed\n",
      "fam: ['15356']\n",
      "cov: ['15356']\n",
      "eigenvec: ['15356']\n",
      "bim: ['14573']\n",
      "snps: ['14573']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_EAS_clean.bed\n",
      "fam: ['2913']\n",
      "cov: ['2913']\n",
      "eigenvec: ['2913']\n",
      "bim: ['17255']\n",
      "snps: ['17255']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AAC_clean.bed\n",
      "fam: ['1331']\n",
      "cov: ['1331']\n",
      "eigenvec: ['1331']\n",
      "bim: ['13110']\n",
      "snps: ['13110']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AJ_clean.bed\n",
      "fam: ['1546']\n",
      "cov: ['1546']\n",
      "eigenvec: ['1546']\n",
      "bim: ['4747']\n",
      "snps: ['4747']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_MDE_clean.bed\n",
      "fam: ['149']\n",
      "cov: ['149']\n",
      "eigenvec: ['149']\n",
      "bim: ['2672']\n",
      "snps: ['2672']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_SAS_clean.bed\n",
      "fam: ['220']\n",
      "cov: ['220']\n",
      "eigenvec: ['220']\n",
      "bim: ['3273']\n",
      "snps: ['3273']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_CAS_clean.bed\n",
      "fam: ['478']\n",
      "cov: ['478']\n",
      "eigenvec: ['478']\n",
      "bim: ['5367']\n",
      "snps: ['5367']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AMR_clean.bed\n",
      "fam: ['573']\n",
      "cov: ['573']\n",
      "eigenvec: ['573']\n",
      "bim: ['5020']\n",
      "snps: ['5020']\n",
      "\n",
      "\n",
      "/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_AFR_clean.bed\n",
      "fam: ['2353']\n",
      "cov: ['2353']\n",
      "eigenvec: ['2353']\n",
      "bim: ['18843']\n",
      "snps: ['18843']\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# check outputs from qc pipeline\n",
    "\n",
    "for label in ancestry_labels:\n",
    "    extensions = ['fam','cov','eigenvec','bim', 'snps']\n",
    "    out_geno = f'/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_{label}_clean.bed'\n",
    "    print(f'{out_geno}')\n",
    "    for ext in extensions:\n",
    "        out_geno_path = f'/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_{label}_clean.{ext}'\n",
    "        if os.path.isfile(out_geno):\n",
    "            count = !cat {out_geno_path} | wc -l\n",
    "            print(f'{ext}: {count}') \n",
    "            \n",
    "        else:\n",
    "            print(f'{out_geno} does not exist!!!')\n",
    "    print()\n",
    "    print()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c0a75eb9-9d5f-4cfd-b81a-f97ac19841f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FIN | total_n: 13 | n_cases: 10 | n_controls: 3 | n_snps: 1152\n",
      "\n",
      "EUR | total_n: 14196 | n_cases: 9230 | n_controls: 4966 | n_snps: 14573\n",
      "\n",
      "EAS | total_n: 2900 | n_cases: 878 | n_controls: 2022 | n_snps: 17255\n",
      "\n",
      "AAC | total_n: 1320 | n_cases: 289 | n_controls: 1031 | n_snps: 13110\n",
      "\n",
      "AJ | total_n: 1126 | n_cases: 740 | n_controls: 386 | n_snps: 4747\n",
      "\n",
      "MDE | total_n: 124 | n_cases: 106 | n_controls: 18 | n_snps: 2672\n",
      "\n",
      "SAS | total_n: 215 | n_cases: 80 | n_controls: 135 | n_snps: 3273\n",
      "\n",
      "CAS | total_n: 474 | n_cases: 195 | n_controls: 279 | n_snps: 5367\n",
      "\n",
      "AMR | total_n: 550 | n_cases: 345 | n_controls: 205 | n_snps: 5020\n",
      "\n",
      "AFR | total_n: 2343 | n_cases: 855 | n_controls: 1488 | n_snps: 18843\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "total_sample_n = 0\n",
    "for label in ancestry_labels:\n",
    "    out_name = f'GP2_round5_APRIL_2023_{label}_clean'\n",
    "    out_path = f'{cnv_path}/genotypes/{out_name}'\n",
    "    fam = pd.read_csv(f'{out_path}.fam', sep='\\s+', header=None, names=['fid','iid','pat','mat','sex','pheno'])\n",
    "    pheno_counts = fam.pheno.value_counts()\n",
    "    ncases = pheno_counts[2]\n",
    "    ncontrols = pheno_counts[1]\n",
    "    ntotal = ncases + ncontrols\n",
    "    total_sample_n += ntotal\n",
    "    nsnps = !cat {out_path}.bim | wc -l\n",
    "    print(f'{label} | total_n: {ntotal} | n_cases: {ncases} | n_controls: {ncontrols} | n_snps: {int(nsnps[0])}')\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "f3c351ba-eae3-4d15-8afd-7e409e967bab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GP2ID</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>manifest_id</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>pheno_for_qc</th>\n",
       "      <th>other_pheno</th>\n",
       "      <th>sex_for_qc</th>\n",
       "      <th>age</th>\n",
       "      <th>age_of_onset</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>study</th>\n",
       "      <th>pruned</th>\n",
       "      <th>pruned_reason</th>\n",
       "      <th>label</th>\n",
       "      <th>related</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>filename</th>\n",
       "      <th>IID</th>\n",
       "      <th>FID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IMMUNEPD_000001</td>\n",
       "      <td>IMMUNEPD_000001_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>2</td>\n",
       "      <td>PD</td>\n",
       "      <td>1</td>\n",
       "      <td>61.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>61.0</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R01C01</td>\n",
       "      <td>206966380021_R01C01</td>\n",
       "      <td>206966380021_R01C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IMMUNEPD_000002</td>\n",
       "      <td>IMMUNEPD_000002_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>2</td>\n",
       "      <td>PD</td>\n",
       "      <td>1</td>\n",
       "      <td>66.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>66.0</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>206966380021_R02C01</td>\n",
       "      <td>206966380021_R02C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IMMUNEPD_000003</td>\n",
       "      <td>IMMUNEPD_000003_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>55.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R03C01</td>\n",
       "      <td>206966380021_R03C01</td>\n",
       "      <td>206966380021_R03C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IMMUNEPD_000004</td>\n",
       "      <td>IMMUNEPD_000004_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>50.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>206966380021_R04C01</td>\n",
       "      <td>206966380021_R04C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IMMUNEPD_000005</td>\n",
       "      <td>IMMUNEPD_000005_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>74.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>IMMUNEPD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206966380021</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>206966380021_R05C01</td>\n",
       "      <td>206966380021_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26723</th>\n",
       "      <td>MDGAP-EBB_000149</td>\n",
       "      <td>MDGAP-EBB_000149_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>206412050061_R04C01</td>\n",
       "      <td>206412050061_R04C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26724</th>\n",
       "      <td>MDGAP-EBB_000150</td>\n",
       "      <td>MDGAP-EBB_000150_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>206412050061_R05C01</td>\n",
       "      <td>206412050061_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26725</th>\n",
       "      <td>MDGAP-EBB_000151</td>\n",
       "      <td>MDGAP-EBB_000151_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>206412050061_R06C01</td>\n",
       "      <td>206412050061_R06C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26726</th>\n",
       "      <td>MDGAP-EBB_000152</td>\n",
       "      <td>MDGAP-EBB_000152_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R07C01</td>\n",
       "      <td>206412050061_R07C01</td>\n",
       "      <td>206412050061_R07C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26727</th>\n",
       "      <td>MDGAP-EBB_000153</td>\n",
       "      <td>MDGAP-EBB_000153_s1</td>\n",
       "      <td>m4</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>Control</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>MDGAP-EBB</td>\n",
       "      <td>1</td>\n",
       "      <td>callrate_prune</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206412050061</td>\n",
       "      <td>R08C01</td>\n",
       "      <td>206412050061_R08C01</td>\n",
       "      <td>206412050061_R08C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26728 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  GP2ID          GP2sampleID manifest_id phenotype  \\\n",
       "0       IMMUNEPD_000001   IMMUNEPD_000001_s1          m1        PD   \n",
       "1       IMMUNEPD_000002   IMMUNEPD_000002_s1          m1        PD   \n",
       "2       IMMUNEPD_000003   IMMUNEPD_000003_s1          m1   Control   \n",
       "3       IMMUNEPD_000004   IMMUNEPD_000004_s1          m1   Control   \n",
       "4       IMMUNEPD_000005   IMMUNEPD_000005_s1          m1   Control   \n",
       "...                 ...                  ...         ...       ...   \n",
       "26723  MDGAP-EBB_000149  MDGAP-EBB_000149_s1          m4   Control   \n",
       "26724  MDGAP-EBB_000150  MDGAP-EBB_000150_s1          m4   Control   \n",
       "26725  MDGAP-EBB_000151  MDGAP-EBB_000151_s1          m4   Control   \n",
       "26726  MDGAP-EBB_000152  MDGAP-EBB_000152_s1          m4   Control   \n",
       "26727  MDGAP-EBB_000153  MDGAP-EBB_000153_s1          m4   Control   \n",
       "\n",
       "       pheno_for_qc other_pheno  sex_for_qc   age  age_of_onset  \\\n",
       "0                 2          PD           1  61.0           NaN   \n",
       "1                 2          PD           1  66.0           NaN   \n",
       "2                 1     Control           2  55.0           NaN   \n",
       "3                 1     Control           1  50.0           NaN   \n",
       "4                 1     Control           2  74.0           NaN   \n",
       "...             ...         ...         ...   ...           ...   \n",
       "26723             1     Control           1   NaN           NaN   \n",
       "26724             1     Control           2   NaN           NaN   \n",
       "26725             1     Control           1   NaN           NaN   \n",
       "26726             1     Control           2   NaN           NaN   \n",
       "26727             1     Control           1   NaN           NaN   \n",
       "\n",
       "       age_at_diagnosis  ...      study pruned   pruned_reason label related  \\\n",
       "0                  61.0  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "1                  66.0  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "2                   NaN  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "3                   NaN  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "4                   NaN  ...   IMMUNEPD      0             NaN   EUR     NaN   \n",
       "...                 ...  ...        ...    ...             ...   ...     ...   \n",
       "26723               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26724               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26725               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26726               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "26727               NaN  ...  MDGAP-EBB      1  callrate_prune   NaN     NaN   \n",
       "\n",
       "       SentrixBarcode_A SentrixPosition_A             filename  \\\n",
       "0          206966380021            R01C01  206966380021_R01C01   \n",
       "1          206966380021            R02C01  206966380021_R02C01   \n",
       "2          206966380021            R03C01  206966380021_R03C01   \n",
       "3          206966380021            R04C01  206966380021_R04C01   \n",
       "4          206966380021            R05C01  206966380021_R05C01   \n",
       "...                 ...               ...                  ...   \n",
       "26723      206412050061            R04C01  206412050061_R04C01   \n",
       "26724      206412050061            R05C01  206412050061_R05C01   \n",
       "26725      206412050061            R06C01  206412050061_R06C01   \n",
       "26726      206412050061            R07C01  206412050061_R07C01   \n",
       "26727      206412050061            R08C01  206412050061_R08C01   \n",
       "\n",
       "                       IID  FID  \n",
       "0      206966380021_R01C01    0  \n",
       "1      206966380021_R02C01    0  \n",
       "2      206966380021_R03C01    0  \n",
       "3      206966380021_R04C01    0  \n",
       "4      206966380021_R05C01    0  \n",
       "...                    ...  ...  \n",
       "26723  206412050061_R04C01    0  \n",
       "26724  206412050061_R05C01    0  \n",
       "26725  206412050061_R06C01    0  \n",
       "26726  206412050061_R07C01    0  \n",
       "26727  206412050061_R08C01    0  \n",
       "\n",
       "[26728 rows x 24 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d56bffd1-35e2-4cfa-ba0f-b9c4c816ec69",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d88db2ff-516c-4e0e-9e50-2ae5763cd9e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for label in ['EUR']:\n",
    "    out_name = f'GP2_round5_APRIL_2023_{label}_clean'\n",
    "    geno_path = f'{cnv_path}/genotypes/{out_name}'\n",
    "    bim_path = f'{geno_path}.bim'\n",
    "    fam = pd.read_csv(f'{geno_path}.fam', sep='\\s+', header=None, names=['FID','IID','pat','mat','sex','pheno'])\n",
    "    fam_key = fam.merge(key, left_on='IID', right_on='GP2sampleID', how='left')\n",
    "    \n",
    "    # samples = pd.read_csv(f'{geno_path}_barcode.samples', header=None, names=['IID'])\n",
    "    label_dir = f'{cnv_path}/cnv_calls/{label}'\n",
    "    os.makedirs(label_dir, exist_ok=True)\n",
    "    for sample in fam_key.filename.unique():\n",
    "        code = sample.split('_')[0]\n",
    "        for chrom in chroms:\n",
    "\n",
    "            mfile = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "\n",
    "            cnv_out = f'{label_dir}/CNV_{label}_{sample}_chr{chrom}.csv'\n",
    "            intervals = f'/data/GP2/utils/ref_dir/glist_hg38_chr{chrom}.csv'\n",
    "\n",
    "            cmd = f'\\\n",
    "python run_cnv_pipeline.py \\\n",
    "--metrics {mfile} \\\n",
    "--bim {bim_path} \\\n",
    "--out_path {cnv_out} \\\n",
    "--intervals {intervals} \\\n",
    "--min_variants 10 \\\n",
    "--kb_window 250 \\\n",
    "--min_gentrain 0.2'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e5943a0e-afc6-4605-93e8-5e308777bd45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID_x</th>\n",
       "      <th>IID_x</th>\n",
       "      <th>pat</th>\n",
       "      <th>mat</th>\n",
       "      <th>sex</th>\n",
       "      <th>pheno</th>\n",
       "      <th>GP2ID</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>manifest_id</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>...</th>\n",
       "      <th>study</th>\n",
       "      <th>pruned</th>\n",
       "      <th>pruned_reason</th>\n",
       "      <th>label</th>\n",
       "      <th>related</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>filename</th>\n",
       "      <th>IID_y</th>\n",
       "      <th>FID_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000001_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>APGS_000001</td>\n",
       "      <td>APGS_000001_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>APGS</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206677820008</td>\n",
       "      <td>R01C01</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000002_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>APGS_000002</td>\n",
       "      <td>APGS_000002_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>APGS</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206677820008</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>206677820008_R02C01</td>\n",
       "      <td>206677820008_R02C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000003_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>APGS_000003</td>\n",
       "      <td>APGS_000003_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>APGS</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206677820008</td>\n",
       "      <td>R03C01</td>\n",
       "      <td>206677820008_R03C01</td>\n",
       "      <td>206677820008_R03C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000004_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>APGS_000004</td>\n",
       "      <td>APGS_000004_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>APGS</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>1.0</td>\n",
       "      <td>206677820008</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>206677820008_R04C01</td>\n",
       "      <td>206677820008_R04C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000005_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>APGS_000005</td>\n",
       "      <td>APGS_000005_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>APGS</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>206677820008</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>206677820008_R05C01</td>\n",
       "      <td>206677820008_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15351</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000475_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>UMD_000475</td>\n",
       "      <td>UMD_000475_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>UMD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204701870092</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>204701870092_R04C01</td>\n",
       "      <td>204701870092_R04C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15352</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000476_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>UMD_000476</td>\n",
       "      <td>UMD_000476_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>...</td>\n",
       "      <td>UMD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204701870051</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>204701870051_R05C01</td>\n",
       "      <td>204701870051_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15353</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000477_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>UMD_000477</td>\n",
       "      <td>UMD_000477_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>UMD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204701870092</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>204701870092_R02C01</td>\n",
       "      <td>204701870092_R02C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15354</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000478_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>UMD_000478</td>\n",
       "      <td>UMD_000478_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>Control</td>\n",
       "      <td>...</td>\n",
       "      <td>UMD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204701860092</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>204701860092_R05C01</td>\n",
       "      <td>204701860092_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15355</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000479_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>UMD_000479</td>\n",
       "      <td>UMD_000479_s1</td>\n",
       "      <td>m1</td>\n",
       "      <td>PD</td>\n",
       "      <td>...</td>\n",
       "      <td>UMD</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EUR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204701870092</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>204701870092_R05C01</td>\n",
       "      <td>204701870092_R05C01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15356 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID_x           IID_x  pat  mat  sex  pheno        GP2ID  \\\n",
       "0          0  APGS_000001_s1    0    0    1      2  APGS_000001   \n",
       "1          0  APGS_000002_s1    0    0    1      2  APGS_000002   \n",
       "2          0  APGS_000003_s1    0    0    1      2  APGS_000003   \n",
       "3          0  APGS_000004_s1    0    0    2      2  APGS_000004   \n",
       "4          0  APGS_000005_s1    0    0    1      2  APGS_000005   \n",
       "...      ...             ...  ...  ...  ...    ...          ...   \n",
       "15351      0   UMD_000475_s1    0    0    1      2   UMD_000475   \n",
       "15352      0   UMD_000476_s1    0    0    1      1   UMD_000476   \n",
       "15353      0   UMD_000477_s1    0    0    2      2   UMD_000477   \n",
       "15354      0   UMD_000478_s1    0    0    1      1   UMD_000478   \n",
       "15355      0   UMD_000479_s1    0    0    2      2   UMD_000479   \n",
       "\n",
       "          GP2sampleID manifest_id phenotype  ...  study pruned  pruned_reason  \\\n",
       "0      APGS_000001_s1          m1        PD  ...   APGS      0            NaN   \n",
       "1      APGS_000002_s1          m1        PD  ...   APGS      0            NaN   \n",
       "2      APGS_000003_s1          m1        PD  ...   APGS      0            NaN   \n",
       "3      APGS_000004_s1          m1        PD  ...   APGS      0            NaN   \n",
       "4      APGS_000005_s1          m1        PD  ...   APGS      0            NaN   \n",
       "...               ...         ...       ...  ...    ...    ...            ...   \n",
       "15351   UMD_000475_s1          m1        PD  ...    UMD      0            NaN   \n",
       "15352   UMD_000476_s1          m1   Control  ...    UMD      0            NaN   \n",
       "15353   UMD_000477_s1          m1        PD  ...    UMD      0            NaN   \n",
       "15354   UMD_000478_s1          m1   Control  ...    UMD      0            NaN   \n",
       "15355   UMD_000479_s1          m1        PD  ...    UMD      0            NaN   \n",
       "\n",
       "       label  related  SentrixBarcode_A  SentrixPosition_A  \\\n",
       "0        EUR      NaN      206677820008             R01C01   \n",
       "1        EUR      NaN      206677820008             R02C01   \n",
       "2        EUR      NaN      206677820008             R03C01   \n",
       "3        EUR      1.0      206677820008             R04C01   \n",
       "4        EUR      NaN      206677820008             R05C01   \n",
       "...      ...      ...               ...                ...   \n",
       "15351    EUR      NaN      204701870092             R04C01   \n",
       "15352    EUR      NaN      204701870051             R05C01   \n",
       "15353    EUR      NaN      204701870092             R02C01   \n",
       "15354    EUR      NaN      204701860092             R05C01   \n",
       "15355    EUR      NaN      204701870092             R05C01   \n",
       "\n",
       "                  filename                IID_y FID_y  \n",
       "0      206677820008_R01C01  206677820008_R01C01     0  \n",
       "1      206677820008_R02C01  206677820008_R02C01     0  \n",
       "2      206677820008_R03C01  206677820008_R03C01     0  \n",
       "3      206677820008_R04C01  206677820008_R04C01     0  \n",
       "4      206677820008_R05C01  206677820008_R05C01     0  \n",
       "...                    ...                  ...   ...  \n",
       "15351  204701870092_R04C01  204701870092_R04C01     0  \n",
       "15352  204701870051_R05C01  204701870051_R05C01     0  \n",
       "15353  204701870092_R02C01  204701870092_R02C01     0  \n",
       "15354  204701860092_R05C01  204701860092_R05C01     0  \n",
       "15355  204701870092_R05C01  204701870092_R05C01     0  \n",
       "\n",
       "[15356 rows x 30 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "d716faf8-48f8-4a74-9c90-8046a49bbb5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "with open(f'{swarm_scripts_dir}/cnvs.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "    # for label in ['EUR']:\n",
    "        out_name = f'GP2_round5_APRIL_2023_{label}_clean'\n",
    "        geno_path = f'{cnv_path}/genotypes/{out_name}'\n",
    "        bim_path = f'{geno_path}.bim'\n",
    "        fam = pd.read_csv(f'{geno_path}.fam', sep='\\s+', header=None, names=['FID','IID','pat','mat','sex','pheno'])\n",
    "        fam_key = fam.merge(key, left_on='IID', right_on='GP2sampleID', how='left')\n",
    "\n",
    "        # samples = pd.read_csv(f'{geno_path}_barcode.samples', header=None, names=['IID'])\n",
    "        label_dir = f'{cnv_path}/cnv_calls/{label}'\n",
    "        os.makedirs(label_dir, exist_ok=True)\n",
    "        for sample in fam_key.filename.unique():\n",
    "            code = sample.split('_')[0]\n",
    "            for chrom in chroms:\n",
    "\n",
    "                mfile = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.parquet'\n",
    "\n",
    "                cnv_out = f'{label_dir}/CNV_{label}_{sample}_chr{chrom}.parquet'\n",
    "                intervals = f'/data/GP2/utils/ref_dir/glist_hg38_chr{chrom}.csv'\n",
    "\n",
    "                cmd = f'\\\n",
    "python run_cnv_pipeline.py \\\n",
    "--metrics {mfile} \\\n",
    "--bim {bim_path} \\\n",
    "--out_path {cnv_out} \\\n",
    "--intervals {intervals} \\\n",
    "--min_variants 10 \\\n",
    "--kb_window 250 \\\n",
    "--min_gentrain 0.2'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "fa7199d9-3b87-4d27-832a-28f98b0dd800",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "64115371\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnvs.swarm -g 4 -t 8 --time=00:15:00 --logdir {swarm_scripts_dir}/logs --module python/3.8 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "88b934f7-2ff6-422d-83a7-486020aa2ff1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink is found\n",
      "Plink2 is found\n"
     ]
    }
   ],
   "source": [
    "# !cat {swarm_scripts_dir}/logs/swarm_64098176_989.e\n",
    "# !cat {swarm_scripts_dir}/cnvs.swarm | grep EUR\n",
    "!python run_cnv_pipeline.py --metrics /data/GP2/raw_genotypes/snp_metrics/206677820008/snp_metrics_206677820008_R01C01_chr1.parquet --bim /data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_EUR_clean.bim --out_path /data/GP2/raw_genotypes/cnvs/release5/cnv_calls/EUR/CNV_EUR_206677820008_R01C01_chr1.parquet --intervals /data/GP2/utils/ref_dir/glist_hg38_chr1.csv --min_variants 10 --kb_window 250 --min_gentrain 0.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "b17553a5-bc57-440a-93e5-3b81d771c7c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>INTERVAL</th>\n",
       "      <th>NUM_VARIANTS</th>\n",
       "      <th>PERCENT_BAF_INSERTION</th>\n",
       "      <th>PERCENT_L2R_DELETION</th>\n",
       "      <th>PERCENT_L2R_DUPLICATION</th>\n",
       "      <th>START_PLUS_WINDOW</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>STOP_PLUS_WINDOW</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33056765</td>\n",
       "      <td>33306765</td>\n",
       "      <td>33321098</td>\n",
       "      <td>33571098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AADACL3</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12466114</td>\n",
       "      <td>12716114</td>\n",
       "      <td>12728759</td>\n",
       "      <td>12978759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AADACL4</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12394546</td>\n",
       "      <td>12644546</td>\n",
       "      <td>12667086</td>\n",
       "      <td>12917086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA4</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>93742837</td>\n",
       "      <td>93992837</td>\n",
       "      <td>94121149</td>\n",
       "      <td>94371149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCB10</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>229266581</td>\n",
       "      <td>229516581</td>\n",
       "      <td>229558695</td>\n",
       "      <td>229808695</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  INTERVAL  NUM_VARIANTS  PERCENT_BAF_INSERTION  PERCENT_L2R_DELETION  \\\n",
       "0  A3GALT2             0                    NaN                   NaN   \n",
       "1  AADACL3             0                    NaN                   NaN   \n",
       "2  AADACL4             0                    NaN                   NaN   \n",
       "3    ABCA4             0                    NaN                   NaN   \n",
       "4   ABCB10             0                    NaN                   NaN   \n",
       "\n",
       "   PERCENT_L2R_DUPLICATION  START_PLUS_WINDOW      START       STOP  \\\n",
       "0                      NaN           33056765   33306765   33321098   \n",
       "1                      NaN           12466114   12716114   12728759   \n",
       "2                      NaN           12394546   12644546   12667086   \n",
       "3                      NaN           93742837   93992837   94121149   \n",
       "4                      NaN          229266581  229516581  229558695   \n",
       "\n",
       "   STOP_PLUS_WINDOW  \n",
       "0          33571098  \n",
       "1          12978759  \n",
       "2          12917086  \n",
       "3          94371149  \n",
       "4         229808695  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = pd.read_parquet('/data/GP2/raw_genotypes/cnvs/release5/cnv_calls/EUR/CNV_EUR_206677820008_R01C01_chr1.parquet')\n",
    "test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "7abdc92d-b571-4864-bfa3-b71262ef332c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>NAME</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>33306765</td>\n",
       "      <td>33321098</td>\n",
       "      <td>A3GALT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>12716114</td>\n",
       "      <td>12728759</td>\n",
       "      <td>AADACL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>12644546</td>\n",
       "      <td>12667086</td>\n",
       "      <td>AADACL4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>93992837</td>\n",
       "      <td>94121149</td>\n",
       "      <td>ABCA4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>229516581</td>\n",
       "      <td>229558695</td>\n",
       "      <td>ABCB10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2807</th>\n",
       "      <td>1</td>\n",
       "      <td>33472630</td>\n",
       "      <td>33496395</td>\n",
       "      <td>ZSCAN20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2808</th>\n",
       "      <td>1</td>\n",
       "      <td>45016403</td>\n",
       "      <td>45206578</td>\n",
       "      <td>ZSWIM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2809</th>\n",
       "      <td>1</td>\n",
       "      <td>52842510</td>\n",
       "      <td>52894575</td>\n",
       "      <td>ZYG11A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2810</th>\n",
       "      <td>1</td>\n",
       "      <td>52726458</td>\n",
       "      <td>52827341</td>\n",
       "      <td>ZYG11B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2811</th>\n",
       "      <td>1</td>\n",
       "      <td>77564504</td>\n",
       "      <td>77682658</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2812 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHR      START       STOP     NAME\n",
       "0       1   33306765   33321098  A3GALT2\n",
       "1       1   12716114   12728759  AADACL3\n",
       "2       1   12644546   12667086  AADACL4\n",
       "3       1   93992837   94121149    ABCA4\n",
       "4       1  229516581  229558695   ABCB10\n",
       "...   ...        ...        ...      ...\n",
       "2807    1   33472630   33496395  ZSCAN20\n",
       "2808    1   45016403   45206578   ZSWIM5\n",
       "2809    1   52842510   52894575   ZYG11A\n",
       "2810    1   52726458   52827341   ZYG11B\n",
       "2811    1   77564504   77682658     ZZZ3\n",
       "\n",
       "[2812 rows x 4 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intervals_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "ce294b79-1b6b-4b44-8637-391b34724ef2",
   "metadata": {},
   "outputs": [],
   "source": [
    "kb_window=250\n",
    "min_variants = 10\n",
    "min_gentrain = 0.2\n",
    "\n",
    "metrics_df = pd.read_parquet('/data/GP2/raw_genotypes/snp_metrics/206677820008/snp_metrics_206677820008_R01C01_chr1.parquet')\n",
    "bim = pd.read_csv('/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_EUR_clean.bim', sep='\\s+', header=None, names=['chr','id','cm','bp','a1','a2'], usecols=['id'])\n",
    "intervals_df = pd.read_csv('/data/GP2/utils/ref_dir/glist_hg38_chr1.csv', dtype={'CHR':str})\n",
    "sample_df = metrics_df.loc[(metrics_df.snpID.isin(bim.id)) & (metrics_df.GenTrain_Score>=min_gentrain)]\n",
    "\n",
    "# Merge the dataframes\n",
    "sample_df = sample_df.merge(intervals_df, left_on='chromosome', right_on='CHR', suffixes=('', '_interval'))\n",
    "\n",
    "# Calculate start and stop plus window\n",
    "sample_df['START_PLUS_WINDOW'] = sample_df['START'] - kb_window * 1000\n",
    "sample_df['STOP_PLUS_WINDOW'] = sample_df['STOP'] + kb_window * 1000\n",
    "\n",
    "# Filter the data\n",
    "sample_df = sample_df[(sample_df['position'] >= sample_df['START_PLUS_WINDOW']) & (sample_df['position'] <= sample_df['STOP_PLUS_WINDOW'])]\n",
    "\n",
    "\n",
    "\n",
    "# Calculate new columns\n",
    "sample_df['BAF_insertion'] = ((sample_df['BAlleleFreq'].between(0.65, 0.85, inclusive='neither')) | (sample_df['BAlleleFreq'].between(0.15, 0.35, inclusive='neither'))).astype(int)\n",
    "sample_df['L2R_deletion'] = (sample_df['LogRRatio'] < -0.2).astype(int)\n",
    "sample_df['L2R_insertion'] = (sample_df['LogRRatio'] > 0.2).astype(int)\n",
    "\n",
    "# Group by interval and calculate the mean\n",
    "results = sample_df.groupby(['NAME', 'CHR', 'START', 'STOP']).agg(\n",
    "    NUM_VARIANTS=('position', 'count'),\n",
    "    PERCENT_BAF_INSERTION=('BAF_insertion', 'mean'),\n",
    "    PERCENT_L2R_DELETION=('L2R_deletion', 'mean'),\n",
    "    PERCENT_L2R_INSERTION=('L2R_insertion', 'mean')\n",
    ").reset_index()\n",
    "\n",
    "\n",
    "\n",
    "# Filter out intervals with less than min_variants\n",
    "results = results[results['NUM_VARIANTS'] >= min_variants]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "e22b1c69-02c8-4d04-8cc6-c12cf5d86047",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/Anaconda/envs/py3.8/lib/python3.8/site-packages/dask/dataframe/multi.py:1254: UserWarning: Concatenating dataframes with unknown divisions.\n",
      "We're assuming that the indices of each dataframes are \n",
      " aligned. This assumption is not generally safe.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "import dask.dataframe as dd\n",
    "from dask import delayed\n",
    "\n",
    "# Custom function to filter the sample_df based on intervals_df\n",
    "def filter_by_intervals(partition, intervals_df, kb_window):\n",
    "    filtered = []\n",
    "    for _, row in intervals_df.iterrows():\n",
    "        temp = partition.loc[(partition['chromosome'] == row['CHR']) &\n",
    "                             (partition['position'] >= row['START'] - kb_window * 1000) &\n",
    "                             (partition['position'] <= row['STOP'] + kb_window * 1000)]\n",
    "        temp = temp.assign(NAME=row['NAME'], CHR=row['CHR'], START=row['START'], STOP=row['STOP'])\n",
    "        filtered.append(temp)\n",
    "    return pd.concat(filtered)\n",
    "\n",
    "\n",
    "kb_window = 250\n",
    "min_variants = 10\n",
    "min_gentrain = 0.2\n",
    "\n",
    "metrics_df = dd.read_parquet('/data/GP2/raw_genotypes/snp_metrics/206677820008/snp_metrics_206677820008_R01C01_chr1.parquet')\n",
    "bim = dd.read_csv('/data/GP2/raw_genotypes/cnvs/release5/genotypes/GP2_round5_APRIL_2023_EUR_clean.bim', sep='\\s+', header=None, names=['chr','id','cm','bp','a1','a2'], usecols=['id'])\n",
    "intervals_df = dd.read_csv('/data/GP2/utils/ref_dir/glist_hg38_chr1.csv', dtype={'CHR':str})\n",
    "\n",
    "sample_df = metrics_df.loc[(metrics_df.snpID.isin(bim.id.compute())) & (metrics_df.GenTrain_Score >= min_gentrain)]\n",
    "\n",
    "# Apply the custom filter function to each partition of the sample_df\n",
    "filtered_sample_df = sample_df.map_partitions(filter_by_intervals, intervals_df=intervals_df, kb_window=kb_window, meta=sample_df.assign(NAME=pd.Series(dtype=object), CHR=pd.Series(dtype=object), START=pd.Series(dtype=int), STOP=pd.Series(dtype=int)))\n",
    "\n",
    "# Use filtered_sample_df instead of sample_df in the following lines\n",
    "filtered_sample_df['BAF_insertion'] = ((filtered_sample_df['BAlleleFreq'].between(0.65, 0.85, inclusive='neither')) | (filtered_sample_df['BAlleleFreq'].between(0.15, 0.35, inclusive='neither'))).astype(int)\n",
    "filtered_sample_df['L2R_deletion'] = (filtered_sample_df['LogRRatio'] < -0.2).astype(int)\n",
    "filtered_sample_df['L2R_insertion'] = (filtered_sample_df['LogRRatio'] > 0.2).astype(int)\n",
    "\n",
    "# Group by interval\n",
    "grouped = filtered_sample_df.groupby(['NAME', 'CHR', 'START', 'STOP'])\n",
    "\n",
    "# Calculate the aggregations separately\n",
    "NUM_VARIANTS = grouped['position'].count().rename('NUM_VARIANTS')\n",
    "PERCENT_BAF_INSERTION = grouped['BAF_insertion'].mean().rename('PERCENT_BAF_INSERTION')\n",
    "PERCENT_L2R_DELETION = grouped['L2R_deletion'].mean().rename('PERCENT_L2R_DELETION')\n",
    "PERCENT_L2R_INSERTION = grouped['L2R_insertion'].mean().rename('PERCENT_L2R_INSERTION')\n",
    "\n",
    "# Concatenate the aggregation results\n",
    "results = dd.concat([NUM_VARIANTS, PERCENT_BAF_INSERTION, PERCENT_L2R_DELETION, PERCENT_L2R_INSERTION], axis=1).reset_index()\n",
    "\n",
    "# Filter out intervals with less than min_variants\n",
    "results = results[results['NUM_VARIANTS'] >= min_variants]\n",
    "# results.to_parquet(out_path, engine='auto', compression='brotli')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "30f6c24a-0c99-4dff-ac15-ef3b764acf48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><strong>Dask DataFrame Structure:</strong></div>\n",
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>snpID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>ALLELE_A</th>\n",
       "      <th>ALLELE_B</th>\n",
       "      <th>BAlleleFreq</th>\n",
       "      <th>LogRRatio</th>\n",
       "      <th>R</th>\n",
       "      <th>Theta</th>\n",
       "      <th>GenTrain_Score</th>\n",
       "      <th>GType</th>\n",
       "      <th>Alt1</th>\n",
       "      <th>Alt2</th>\n",
       "      <th>GT</th>\n",
       "      <th>a1</th>\n",
       "      <th>a2</th>\n",
       "      <th>maf</th>\n",
       "      <th>BAF_insertion</th>\n",
       "      <th>L2R_deletion</th>\n",
       "      <th>L2R_insertion</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>npartitions=1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <td>object</td>\n",
       "      <td>int64</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>int64</td>\n",
       "      <td>int64</td>\n",
       "      <td>float64</td>\n",
       "      <td>float64</td>\n",
       "      <td>float64</td>\n",
       "      <td>float64</td>\n",
       "      <td>float64</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>float64</td>\n",
       "      <td>int64</td>\n",
       "      <td>int64</td>\n",
       "      <td>int64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "<div>Dask Name: assign, 25 tasks</div>"
      ],
      "text/plain": [
       "Dask DataFrame Structure:\n",
       "              chromosome position   snpID Sample_ID     Ref     Alt ALLELE_A ALLELE_B BAlleleFreq LogRRatio        R    Theta GenTrain_Score   GType    Alt1    Alt2      GT      a1      a2      maf BAF_insertion L2R_deletion L2R_insertion\n",
       "npartitions=1                                                                                                                                                                                                                                 \n",
       "                  object    int64  object    object  object  object    int64    int64     float64   float64  float64  float64        float64  object  object  object  object  object  object  float64         int64        int64         int64\n",
       "                     ...      ...     ...       ...     ...     ...      ...      ...         ...       ...      ...      ...            ...     ...     ...     ...     ...     ...     ...      ...           ...          ...           ...\n",
       "Dask Name: assign, 25 tasks"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_sample_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "6d06700f-a898-4adf-a75b-783a55435783",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAME</th>\n",
       "      <th>CHR</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>NUM_VARIANTS</th>\n",
       "      <th>PERCENT_BAF_INSERTION</th>\n",
       "      <th>PERCENT_L2R_DELETION</th>\n",
       "      <th>PERCENT_L2R_INSERTION</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>ARHGEF16</td>\n",
       "      <td>1</td>\n",
       "      <td>3454582</td>\n",
       "      <td>3481113</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.363636</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>C1orf101</td>\n",
       "      <td>1</td>\n",
       "      <td>244461370</td>\n",
       "      <td>244640360</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>1</td>\n",
       "      <td>6785323</td>\n",
       "      <td>7769706</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>CSMD2</td>\n",
       "      <td>1</td>\n",
       "      <td>33513998</td>\n",
       "      <td>34165842</td>\n",
       "      <td>13</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.307692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>DAB1</td>\n",
       "      <td>1</td>\n",
       "      <td>56997905</td>\n",
       "      <td>58250539</td>\n",
       "      <td>12</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.083333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>EDARADD</td>\n",
       "      <td>1</td>\n",
       "      <td>236394379</td>\n",
       "      <td>236484708</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>ERO1LB</td>\n",
       "      <td>1</td>\n",
       "      <td>236215121</td>\n",
       "      <td>236282039</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>ESRRG</td>\n",
       "      <td>1</td>\n",
       "      <td>216503245</td>\n",
       "      <td>217137755</td>\n",
       "      <td>12</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.083333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>FHAD1</td>\n",
       "      <td>1</td>\n",
       "      <td>15247271</td>\n",
       "      <td>15398126</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>FMN2</td>\n",
       "      <td>1</td>\n",
       "      <td>240091884</td>\n",
       "      <td>240475189</td>\n",
       "      <td>13</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.076923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>783</th>\n",
       "      <td>IGSF21</td>\n",
       "      <td>1</td>\n",
       "      <td>18107745</td>\n",
       "      <td>18378483</td>\n",
       "      <td>12</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>KAZN</td>\n",
       "      <td>1</td>\n",
       "      <td>14598716</td>\n",
       "      <td>15118048</td>\n",
       "      <td>16</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.187500</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>LOC100506795</td>\n",
       "      <td>1</td>\n",
       "      <td>234527890</td>\n",
       "      <td>234531779</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1047</th>\n",
       "      <td>LOC339529</td>\n",
       "      <td>1</td>\n",
       "      <td>243917401</td>\n",
       "      <td>244047318</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1139</th>\n",
       "      <td>MEGF6</td>\n",
       "      <td>1</td>\n",
       "      <td>3487941</td>\n",
       "      <td>3611495</td>\n",
       "      <td>12</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1360</th>\n",
       "      <td>NAV1</td>\n",
       "      <td>1</td>\n",
       "      <td>201648321</td>\n",
       "      <td>201826974</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1385</th>\n",
       "      <td>NFIA</td>\n",
       "      <td>1</td>\n",
       "      <td>61077273</td>\n",
       "      <td>61462788</td>\n",
       "      <td>14</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.071429</td>\n",
       "      <td>0.071429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1518</th>\n",
       "      <td>PAX7</td>\n",
       "      <td>1</td>\n",
       "      <td>18631005</td>\n",
       "      <td>18748866</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519</th>\n",
       "      <td>PBX1</td>\n",
       "      <td>1</td>\n",
       "      <td>164559359</td>\n",
       "      <td>164851823</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1571</th>\n",
       "      <td>PKP1</td>\n",
       "      <td>1</td>\n",
       "      <td>201283451</td>\n",
       "      <td>201332993</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1634</th>\n",
       "      <td>PRDM16</td>\n",
       "      <td>1</td>\n",
       "      <td>3069177</td>\n",
       "      <td>3438621</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.266667</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1718</th>\n",
       "      <td>RGS7</td>\n",
       "      <td>1</td>\n",
       "      <td>240775513</td>\n",
       "      <td>241357230</td>\n",
       "      <td>12</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1964</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>1</td>\n",
       "      <td>237042401</td>\n",
       "      <td>237833988</td>\n",
       "      <td>17</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.058824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2063</th>\n",
       "      <td>SLC35F3</td>\n",
       "      <td>1</td>\n",
       "      <td>233904932</td>\n",
       "      <td>234324516</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2179</th>\n",
       "      <td>TARBP1</td>\n",
       "      <td>1</td>\n",
       "      <td>234391312</td>\n",
       "      <td>234479103</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2224</th>\n",
       "      <td>TMEM51</td>\n",
       "      <td>1</td>\n",
       "      <td>15152531</td>\n",
       "      <td>15220478</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2251</th>\n",
       "      <td>TNNI1</td>\n",
       "      <td>1</td>\n",
       "      <td>201403766</td>\n",
       "      <td>201421746</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              NAME CHR      START       STOP  NUM_VARIANTS  \\\n",
       "85        ARHGEF16   1    3454582    3481113            11   \n",
       "160       C1orf101   1  244461370  244640360            10   \n",
       "234         CAMTA1   1    6785323    7769706            10   \n",
       "370          CSMD2   1   33513998   34165842            13   \n",
       "394           DAB1   1   56997905   58250539            12   \n",
       "463        EDARADD   1  236394379  236484708            10   \n",
       "499         ERO1LB   1  236215121  236282039            10   \n",
       "503          ESRRG   1  216503245  217137755            12   \n",
       "582          FHAD1   1   15247271   15398126            10   \n",
       "593           FMN2   1  240091884  240475189            13   \n",
       "783         IGSF21   1   18107745   18378483            12   \n",
       "826           KAZN   1   14598716   15118048            16   \n",
       "985   LOC100506795   1  234527890  234531779            10   \n",
       "1047     LOC339529   1  243917401  244047318            10   \n",
       "1139         MEGF6   1    3487941    3611495            12   \n",
       "1360          NAV1   1  201648321  201826974            10   \n",
       "1385          NFIA   1   61077273   61462788            14   \n",
       "1518          PAX7   1   18631005   18748866            10   \n",
       "1519          PBX1   1  164559359  164851823            11   \n",
       "1571          PKP1   1  201283451  201332993            10   \n",
       "1634        PRDM16   1    3069177    3438621            15   \n",
       "1718          RGS7   1  240775513  241357230            12   \n",
       "1964          RYR2   1  237042401  237833988            17   \n",
       "2063       SLC35F3   1  233904932  234324516            11   \n",
       "2179        TARBP1   1  234391312  234479103            10   \n",
       "2224        TMEM51   1   15152531   15220478            10   \n",
       "2251         TNNI1   1  201403766  201421746            10   \n",
       "\n",
       "      PERCENT_BAF_INSERTION  PERCENT_L2R_DELETION  PERCENT_L2R_INSERTION  \n",
       "85                   0.0000              0.363636               0.000000  \n",
       "160                  0.0000              0.000000               0.000000  \n",
       "234                  0.0000              0.000000               0.100000  \n",
       "370                  0.0000              0.000000               0.307692  \n",
       "394                  0.0000              0.083333               0.083333  \n",
       "463                  0.0000              0.000000               0.100000  \n",
       "499                  0.0000              0.000000               0.100000  \n",
       "503                  0.0000              0.083333               0.083333  \n",
       "582                  0.0000              0.200000               0.000000  \n",
       "593                  0.0000              0.000000               0.076923  \n",
       "783                  0.0000              0.083333               0.000000  \n",
       "826                  0.0625              0.187500               0.000000  \n",
       "985                  0.0000              0.100000               0.000000  \n",
       "1047                 0.0000              0.000000               0.000000  \n",
       "1139                 0.0000              0.333333               0.000000  \n",
       "1360                 0.0000              0.000000               0.000000  \n",
       "1385                 0.0000              0.071429               0.071429  \n",
       "1518                 0.0000              0.100000               0.000000  \n",
       "1519                 0.0000              0.000000               0.000000  \n",
       "1571                 0.0000              0.100000               0.100000  \n",
       "1634                 0.0000              0.266667               0.000000  \n",
       "1718                 0.0000              0.000000               0.000000  \n",
       "1964                 0.0000              0.000000               0.058824  \n",
       "2063                 0.0000              0.090909               0.000000  \n",
       "2179                 0.0000              0.100000               0.000000  \n",
       "2224                 0.0000              0.200000               0.000000  \n",
       "2251                 0.0000              0.100000               0.100000  "
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Compute the final result and convert it to a Pandas DataFrame\n",
    "test_results = results.compute()\n",
    "test_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "15407d9e-b374-47db-9501-552a4628bff7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>snpID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>ALLELE_A</th>\n",
       "      <th>ALLELE_B</th>\n",
       "      <th>BAlleleFreq</th>\n",
       "      <th>LogRRatio</th>\n",
       "      <th>R</th>\n",
       "      <th>Theta</th>\n",
       "      <th>GenTrain_Score</th>\n",
       "      <th>GType</th>\n",
       "      <th>Alt1</th>\n",
       "      <th>Alt2</th>\n",
       "      <th>GT</th>\n",
       "      <th>a1</th>\n",
       "      <th>a2</th>\n",
       "      <th>maf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52888</th>\n",
       "      <td>1</td>\n",
       "      <td>3245479</td>\n",
       "      <td>rs74048402</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.021550</td>\n",
       "      <td>-0.216438</td>\n",
       "      <td>1.007590</td>\n",
       "      <td>0.974296</td>\n",
       "      <td>0.825628</td>\n",
       "      <td>AA</td>\n",
       "      <td>A</td>\n",
       "      <td>None</td>\n",
       "      <td>BB</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52931</th>\n",
       "      <td>1</td>\n",
       "      <td>3300587</td>\n",
       "      <td>rs2455137</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.494529</td>\n",
       "      <td>0.002118</td>\n",
       "      <td>0.540867</td>\n",
       "      <td>0.513139</td>\n",
       "      <td>0.771683</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52979</th>\n",
       "      <td>1</td>\n",
       "      <td>3345118</td>\n",
       "      <td>rs4415513</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.494152</td>\n",
       "      <td>-0.165281</td>\n",
       "      <td>0.759863</td>\n",
       "      <td>0.582220</td>\n",
       "      <td>0.838088</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52987</th>\n",
       "      <td>1</td>\n",
       "      <td>3353146</td>\n",
       "      <td>rs12076855</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.011877</td>\n",
       "      <td>-0.287709</td>\n",
       "      <td>0.650681</td>\n",
       "      <td>0.024299</td>\n",
       "      <td>0.852240</td>\n",
       "      <td>AA</td>\n",
       "      <td>C</td>\n",
       "      <td>None</td>\n",
       "      <td>AA</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52991</th>\n",
       "      <td>1</td>\n",
       "      <td>3355977</td>\n",
       "      <td>JHU_1.3272540</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.002809</td>\n",
       "      <td>-0.077267</td>\n",
       "      <td>1.285940</td>\n",
       "      <td>0.018699</td>\n",
       "      <td>0.826347</td>\n",
       "      <td>BB</td>\n",
       "      <td>A</td>\n",
       "      <td>None</td>\n",
       "      <td>AA</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.437500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53013</th>\n",
       "      <td>1</td>\n",
       "      <td>3384243</td>\n",
       "      <td>rs868688</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.475585</td>\n",
       "      <td>-0.189619</td>\n",
       "      <td>0.881796</td>\n",
       "      <td>0.453435</td>\n",
       "      <td>0.854678</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.285714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53017</th>\n",
       "      <td>1</td>\n",
       "      <td>3386882</td>\n",
       "      <td>rs2245703</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.002190</td>\n",
       "      <td>-0.012652</td>\n",
       "      <td>1.065500</td>\n",
       "      <td>0.982750</td>\n",
       "      <td>0.931019</td>\n",
       "      <td>BB</td>\n",
       "      <td>C</td>\n",
       "      <td>None</td>\n",
       "      <td>BB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.437500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53300</th>\n",
       "      <td>1</td>\n",
       "      <td>3534441</td>\n",
       "      <td>rs12119711</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.011486</td>\n",
       "      <td>-0.123429</td>\n",
       "      <td>0.772890</td>\n",
       "      <td>0.042202</td>\n",
       "      <td>0.868157</td>\n",
       "      <td>BB</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>AA</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.375000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53350</th>\n",
       "      <td>1</td>\n",
       "      <td>3595370</td>\n",
       "      <td>exm7149</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.049961</td>\n",
       "      <td>-0.330753</td>\n",
       "      <td>0.415949</td>\n",
       "      <td>0.107751</td>\n",
       "      <td>0.791691</td>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>None</td>\n",
       "      <td>AA</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.142857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53383</th>\n",
       "      <td>1</td>\n",
       "      <td>3621937</td>\n",
       "      <td>rs4364818</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.440968</td>\n",
       "      <td>0.072807</td>\n",
       "      <td>1.406750</td>\n",
       "      <td>0.447423</td>\n",
       "      <td>0.861270</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.357143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53508</th>\n",
       "      <td>1</td>\n",
       "      <td>3718901</td>\n",
       "      <td>rs3765767</td>\n",
       "      <td>206677820008_R01C01</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.017782</td>\n",
       "      <td>-0.402822</td>\n",
       "      <td>0.499591</td>\n",
       "      <td>0.039802</td>\n",
       "      <td>0.854595</td>\n",
       "      <td>AA</td>\n",
       "      <td>C</td>\n",
       "      <td>None</td>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  position          snpID            Sample_ID Ref Alt  \\\n",
       "52888          1   3245479     rs74048402  206677820008_R01C01   G   A   \n",
       "52931          1   3300587      rs2455137  206677820008_R01C01   C   T   \n",
       "52979          1   3345118      rs4415513  206677820008_R01C01   C   T   \n",
       "52987          1   3353146     rs12076855  206677820008_R01C01   A   C   \n",
       "52991          1   3355977  JHU_1.3272540  206677820008_R01C01   G   A   \n",
       "53013          1   3384243       rs868688  206677820008_R01C01   C   T   \n",
       "53017          1   3386882      rs2245703  206677820008_R01C01   T   C   \n",
       "53300          1   3534441     rs12119711  206677820008_R01C01   C   T   \n",
       "53350          1   3595370        exm7149  206677820008_R01C01   A   G   \n",
       "53383          1   3621937      rs4364818  206677820008_R01C01   C   T   \n",
       "53508          1   3718901      rs3765767  206677820008_R01C01   G   C   \n",
       "\n",
       "       ALLELE_A  ALLELE_B  BAlleleFreq  LogRRatio         R     Theta  \\\n",
       "52888         1         0     1.021550  -0.216438  1.007590  0.974296   \n",
       "52931         1         0     0.494529   0.002118  0.540867  0.513139   \n",
       "52979         1         0     0.494152  -0.165281  0.759863  0.582220   \n",
       "52987         0         1    -0.011877  -0.287709  0.650681  0.024299   \n",
       "52991         1         0    -0.002809  -0.077267  1.285940  0.018699   \n",
       "53013         1         0     0.475585  -0.189619  0.881796  0.453435   \n",
       "53017         0         1     1.002190  -0.012652  1.065500  0.982750   \n",
       "53300         1         0     0.011486  -0.123429  0.772890  0.042202   \n",
       "53350         0         1     0.049961  -0.330753  0.415949  0.107751   \n",
       "53383         1         0     0.440968   0.072807  1.406750  0.447423   \n",
       "53508         0         1     0.017782  -0.402822  0.499591  0.039802   \n",
       "\n",
       "       GenTrain_Score GType Alt1  Alt2  GT a1 a2       maf  \n",
       "52888        0.825628    AA    A  None  BB  A  G  0.000000  \n",
       "52931        0.771683    AB    T  None  AB  T  C  0.500000  \n",
       "52979        0.838088    AB    T  None  AB  T  C  0.500000  \n",
       "52987        0.852240    AA    C  None  AA  A  C  0.000000  \n",
       "52991        0.826347    BB    A  None  AA  A  G  0.437500  \n",
       "53013        0.854678    AB    T  None  AB  T  C  0.285714  \n",
       "53017        0.931019    BB    C  None  BB  T  C  0.437500  \n",
       "53300        0.868157    BB    T  None  AA  T  C  0.375000  \n",
       "53350        0.791691    AA    G  None  AA  A  G  0.142857  \n",
       "53383        0.861270    AB    T  None  AB  T  C  0.357143  \n",
       "53508        0.854595    AA    C  None  AA  G  C  0.500000  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_test = sample_df.compute()\n",
    "sample_test.loc[(sample_test['chromosome']=='1') & (sample_test['position'] >= 3204582) &\t(sample_test['position']<= 3731113)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "f8f016e0-1f30-4c28-8188-d7f7278dc723",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chromosome', 'position', 'snpID', 'Sample_ID', 'Ref', 'Alt',\n",
       "       'ALLELE_A', 'ALLELE_B', 'BAlleleFreq', 'LogRRatio', 'R', 'Theta',\n",
       "       'GenTrain_Score', 'GType', 'Alt1', 'Alt2', 'GT', 'a1', 'a2', 'maf',\n",
       "       'CHR', 'START', 'STOP', 'NAME', 'START_PLUS_WINDOW', 'STOP_PLUS_WINDOW',\n",
       "       'BAF_insertion', 'L2R_deletion', 'L2R_insertion'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_test.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f3184f1-1047-4cf2-9931-673a2ab57e68",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2e9e6292-58c4-4565-8074-cfa5a39f883a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MDE\n",
      "120\n",
      "\n",
      "\n",
      "AMR\n",
      "489\n",
      "\n",
      "\n",
      "AJ\n",
      "1434\n",
      "\n",
      "\n",
      "AAC\n",
      "1304\n",
      "\n",
      "\n",
      "EUR\n",
      "13611\n",
      "\n",
      "\n",
      "EAS\n",
      "154\n",
      "\n",
      "\n",
      "CAS\n",
      "313\n",
      "\n",
      "\n",
      "SAS\n",
      "79\n",
      "\n",
      "\n",
      "FIN\n",
      "11\n",
      "\n",
      "\n",
      "AFR\n",
      "404\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "missing_cnvs = list()\n",
    "missing_metrics = list()\n",
    "for label in ancestry_labels:\n",
    "    # for label in ['EUR']:\n",
    "    out_name = f'GP2_round4_JAN_24_2023_{label}'\n",
    "    label_dir = f'{cnv_path}/cnv_calls/{label}'\n",
    "    geno_path = f'{cnv_path}/genotypes/{out_name}_clean'\n",
    "    samples = pd.read_csv(f'{geno_path}_barcode.samples', header=None, names=['IID'])\n",
    "    \n",
    "    print(label)\n",
    "    print(samples.shape[0])\n",
    "    print()\n",
    "    print()\n",
    "    \n",
    "    \n",
    "    for sample in samples.IID.unique():\n",
    "        code = sample.split('_')[0]\n",
    "        for chrom in chroms:\n",
    "            \n",
    "            mfile = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "            cnv_out = f'{label_dir}/CNV_{label}_{sample}_chr{chrom}.csv'\n",
    "            \n",
    "            if os.path.isfile(mfile):\n",
    "                pass\n",
    "            else:\n",
    "                missing_metrics.append(mfile)\n",
    "                \n",
    "            if os.path.isfile(cnv_out):\n",
    "                pass\n",
    "            else:\n",
    "                missing_cnvs.append(cnv_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "75f31342-5241-48f9-bdf5-e73b1c46798d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# write dosages per chromosome per ancestry per cnv-type\n",
    "\n",
    "with open(f'{swarm_scripts_dir}/cnv_dosages.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        label_dir = f'{cnv_path}/dosages/{label}'\n",
    "        os.makedirs(label_dir, exist_ok=True)\n",
    "        for chrom in chroms:\n",
    "\n",
    "            cnv_files_list = glob.glob(f'{cnv_path}/cnv_calls/{label}/CNV_{label}_*_chr{chrom}.csv')\n",
    "            cnv_files_df = pd.DataFrame({'filename': cnv_files_list})\n",
    "            cnv_files_df.to_csv(f'{label_dir}/CNV_{label}_chr{chrom}_files.csv', header=False, index=False)\n",
    "\n",
    "            for cnv_type in cnv_types:\n",
    "                \n",
    "                cmd = f'\\\n",
    "python /data/vitaled2/CNVs/run_cnv_dosage_pipeline.py \\\n",
    "--files {label_dir}/CNV_{label}_chr{chrom}_files.csv \\\n",
    "--label {label} \\\n",
    "--chrom {chrom} \\\n",
    "--cnv_type {cnv_type} \\\n",
    "--out_path {label_dir}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "5bce1bd2-a140-47d7-93cb-f6515adea344",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58322813\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_dosages.swarm -g 32 -t 16 --logdir {swarm_scripts_dir}/logs --module python/3.8 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "ab5d95cc-a362-4abe-8e1f-3ea30fd5041d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "label = 'EUR'\n",
    "label_dir = f'/data/GP2/raw_genotypes/cnvs/release4/cnv_calls/{label}'\n",
    "chrom = '1'\n",
    "cnv_files = f'{cnv_path}/dosages/{label}/CNV_{label}_chr{chrom}_files.csv' \n",
    "cnv_type = 'PERCENT_BAF_INSERTION'\n",
    "\n",
    "cnv_files_df = pd.read_csv(cnv_files, header=None)\n",
    "files_list = cnv_files_df.loc[:,0]\n",
    "\n",
    "dosage_ = pd.DataFrame()\n",
    "\n",
    "for sample_chrom_file in files_list:\n",
    "\n",
    "    sample = sample_chrom_file.split('/')[-1].replace(f'CNV_{label}_','').replace(f'_chr{chrom}.csv','')\n",
    "    cnvs = pd.read_csv(sample_chrom_file, usecols=['INTERVAL', cnv_type, 'NUM_VARIANTS'])\n",
    "    cnvs.loc[:,'sampleid'] = sample\n",
    "    cnvs_final = cnvs.loc[~cnvs[cnv_type].isna()]\n",
    "    dosage = cnvs_final.loc[:,[cnv_type,'INTERVAL','sampleid']]\n",
    "    dosage_pivot = dosage.pivot(index='sampleid', columns='INTERVAL', values=cnv_type)\n",
    "\n",
    "    dosage_ = pd.concat([dosage_, dosage_pivot])\n",
    "# dosage_.to_csv(out_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5493f8c-6cdc-4cdc-9157-680924abd629",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d8eecf1-4cbf-4aa6-a81c-e21239f97a19",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e67bb1ce-7d63-47aa-a0ab-0fb64fdf1a47",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c273d2fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cnv_dosage = pd.read_csv('/data/GP2/raw_genotypes/cnvs/release4/dosages/EUR/CNV_EUR_chr1_PERCENT_BAF_INSERTION.csv')\n",
    "dose_vc = cnv_dosage.apply(lambda x: x.isnull().value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "3b0970fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleid</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>AADACL3</th>\n",
       "      <th>AADACL4</th>\n",
       "      <th>ABCA4</th>\n",
       "      <th>ABCB10</th>\n",
       "      <th>ABCD3</th>\n",
       "      <th>ABL2</th>\n",
       "      <th>ACADM</th>\n",
       "      <th>ACAP3</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNHIT6</th>\n",
       "      <th>ZP4</th>\n",
       "      <th>ZRANB2</th>\n",
       "      <th>ZRANB2-AS1</th>\n",
       "      <th>ZRANB2-AS2</th>\n",
       "      <th>ZSCAN20</th>\n",
       "      <th>ZSWIM5</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>13611.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>True</th>\n",
       "      <td>NaN</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>...</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 2581 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       sampleid  A3GALT2  AADACL3  AADACL4    ABCA4   ABCB10    ABCD3  \\\n",
       "False   13611.0      NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "True        NaN  13611.0  13611.0  13611.0  13611.0  13611.0  13611.0   \n",
       "\n",
       "          ABL2    ACADM    ACAP3  ...   ZNHIT6      ZP4   ZRANB2  ZRANB2-AS1  \\\n",
       "False      NaN      NaN      NaN  ...      NaN      NaN      NaN         NaN   \n",
       "True   13611.0  13611.0  13611.0  ...  13611.0  13611.0  13611.0     13611.0   \n",
       "\n",
       "       ZRANB2-AS2  ZSCAN20   ZSWIM5   ZYG11A   ZYG11B     ZZZ3  \n",
       "False         NaN      NaN      NaN      NaN      NaN      NaN  \n",
       "True      13611.0  13611.0  13611.0  13611.0  13611.0  13611.0  \n",
       "\n",
       "[2 rows x 2581 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dose_vc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "8eaefe81-9ec6-4179-9f2c-776ba966b8ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/update_cnv_ids.swarm','w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        label_dir = f'{cnv_path}/dosages/{label}'\n",
    "        out_dir = f'/data/GP2/releases/gp2tier2/release4_14022023/cnvs/{label}'\n",
    "        os.makedirs(out_dir, exist_ok=True)\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "\n",
    "                dosagefile = f'{label_dir}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                dosagefile_out = f'{out_dir}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                pheno_out = f'{label_dir}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "                cmd = f'\\\n",
    "python update_cnv_ids.py \\\n",
    "--dosagefile {dosagefile} \\\n",
    "--key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv \\\n",
    "--pheno_out {pheno_out} \\\n",
    "--out_path {dosagefile_out}'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "051dc35b-f2f2-4eb8-877a-a720e822fb04",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58356484\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/update_cnv_ids.swarm -g 8 -t 8 --logdir {swarm_scripts_dir}/logs --module python/3.8 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "7936e3e6-fb58-4957-bf60-17ae52a0a4a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr1_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr1_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr1_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr2_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr2_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr2_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr3_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr3_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr3_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr4_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr4_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr4_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr5_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr5_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr5_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr6_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr6_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr6_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr7_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr7_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr7_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr8_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr8_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr8_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr9_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr9_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr9_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr10_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr10_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr10_PERCENT_BAF_INSERTION_gp2ids.csv\n"
     ]
    }
   ],
   "source": [
    "!head {swarm_scripts_dir}/update_cnv_ids.swarm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "3b3a4dec-9b45-40ac-a1bc-0c51090c2819",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_out_path = '/data/GP2/raw_genotypes/cnvwas'\n",
    "with open(f'{swarm_scripts_dir}/cnvwas.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "\n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                pheno = f'{cnv_path}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "                covar = f'{cnv_path}/GP2_round4_JAN_24_2023_{label}_clean.cov'\n",
    "                out_path = f'{cnv_out_path}/GP2_{label}_chr{chrom}_{cnv_type}_cnvwas.tab'\n",
    "\n",
    "                cmd = f'\\\n",
    "python run_cnvwas_pipeline.py \\\n",
    "--cnv_dosage_file {dosagefile} \\\n",
    "--out_path {out_path} \\\n",
    "--pheno {pheno} \\\n",
    "--covar {covar}'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "9098b284-30f8-4c79-9944-9dc8da375090",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58077962\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnvwas.swarm -g 32 -t 16 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "931f3e00-37b1-4a89-b7b3-529d0ca4c7ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n",
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n",
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from GWAS.gwas import calculate_inflation\n",
    "label = 'EUR'\n",
    "# cnv_type = 'PERCENT_BAF_INSERTION'\n",
    "# cnv_type = 'PERCENT_L2R_DUPLICATION'\n",
    "# chrom = '1'\n",
    "pheno_df = pd.read_csv(f'{cnv_path}/GP2_EUR_chr1_{cnv_type}.pheno', sep='\\t', engine='python')\n",
    "ncontrols = pheno_df.pheno.value_counts()[0]\n",
    "ncases = pheno_df.pheno.value_counts()[1]\n",
    "cnv_lambda_dict = dict()\n",
    "for cnv_type in cnv_types:\n",
    "    total_cnvtype_df = pd.DataFrame()\n",
    "    for chrom in chroms:\n",
    "        tab = f'{cnv_out_path}/GP2_{label}_chr{chrom}_{cnv_type}_cnvwas.tab'\n",
    "\n",
    "        if os.path.isfile(tab):\n",
    "\n",
    "            sumstats = pd.read_csv(tab, sep='\\s+', engine='python')\n",
    "            # print(tab)\n",
    "            # infl = calculate_inflation(sumstats.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "            # print(infl)\n",
    "            # print()\n",
    "            # print()\n",
    "            total_cnvtype_df = pd.concat([total_cnvtype_df, sumstats], ignore_index=True)\n",
    "        else:\n",
    "            print(f'{tab} does not exist!')\n",
    "\n",
    "    infl = calculate_inflation(total_cnvtype_df.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "    cnv_lambda_dict[cnv_type] = infl\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "69500012-0c99-45cf-8c91-0a8afa7e3ff4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'PERCENT_BAF_INSERTION': {'pass': True,\n",
       "  'step': 'lambda_calculation',\n",
       "  'metrics': {'inflation': 2.101750589371727}},\n",
       " 'PERCENT_L2R_DELETION': {'pass': True,\n",
       "  'step': 'lambda_calculation',\n",
       "  'metrics': {'inflation': 14.555569305672298}},\n",
       " 'PERCENT_L2R_DUPLICATION': {'pass': True,\n",
       "  'step': 'lambda_calculation',\n",
       "  'metrics': {'inflation': 1.3423052908502526}}}"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_lambda_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "20feb548-fbfc-49d5-8d92-98ee2afc218a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'lambda_calculation',\n",
       " 'metrics': {'inflation': 1.3423052908502526}}"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "pheno = pd.read_csv(f'{cnv_path}/GP2_EUR_chr1_{cnv_type}.pheno', sep='\\t', engine='python')\n",
    "ncontrols = pheno.pheno.value_counts()[0]\n",
    "ncases = pheno.pheno.value_counts()[1]\n",
    "\n",
    "infl = calculate_inflation(total_cnvtype_df.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "infl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "0e4d119f-b74a-4af8-8548-dd4c2b844198",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8066"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "02570930-f095-4a36-a28d-d6d0b93c723c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13911"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno.pheno.value_counts()[1] + pheno.pheno.value_counts()[0] + 1475"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d68612b4-bb57-481d-8506-b1cd1ab71891",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    8066\n",
       "0    4370\n",
       "Name: pheno, dtype: int64"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ce8cb4aa-172a-4258-911f-2e38dd93c0a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       " 2    8711\n",
       " 1    4876\n",
       "-9    1475\n",
       "Name: pheno, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam = pd.read_csv(f'/data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_{label}.fam', sep='\\t', header=None, names=['fid','iid','pat','mat','sex','pheno'], engine='python')\n",
    "fam.pheno.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06cd40f1-e86e-4a8e-9f2a-4f2c36cd8eba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "3087cdf2-024f-4d96-bdb2-5cfb319558f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/GP2/raw_genotypes/cnvs/release4/CNV_EUR_chr1_PERCENT_L2R_DUPLICATION_gp2ids.csv'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "cnv_dosage_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "f68a016a-a0e2-4fd9-a513-49c333668b0f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638843\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639250\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639028\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639249\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639250\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639149\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.636706\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639198\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639250\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639085\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638965\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639251\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639251\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639252\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639221\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639227\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639244\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638848\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639203\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639254\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639143\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639249\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638589\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639251\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639117\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639252\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638968\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639045\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639113\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639254\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638554\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639185\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639046\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638780\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639253\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639052\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639190\n",
      "         Iterations 6\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import numpy as np\n",
    "from numpy.core.numeric import NaN\n",
    "import pandas as pd\n",
    "import glob\n",
    "import shutil\n",
    "from sklearn.preprocessing import MinMaxScaler\n",
    "import statsmodels.api as sm\n",
    "\n",
    "# Supress copy warning.\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "from QC.utils import shell_do\n",
    "\n",
    "label = 'EUR'\n",
    "chrom = '1'\n",
    "cnv_type = 'PERCENT_L2R_DUPLICATION'\n",
    "dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "pheno = f'{cnv_path}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "covar = f'{cnv_path}/GP2_round4_JAN_24_2023_{label}_clean.cov'\n",
    "out_path = f'{cnv_out_path}/GP2_{label}_chr{chrom}_{cnv_type}_cnvwas.tab'\n",
    "cnv_dosage_file = dosagefile\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "scaler = MinMaxScaler()\n",
    "dosage_df = pd.read_csv(cnv_dosage_file)\n",
    "#fix column names\n",
    "dosage_df.columns = [x.replace('-','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace('.','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace(' ','_') for x in dosage_df.columns]\n",
    "\n",
    "pheno_df = pd.read_csv(pheno, sep='\\t')\n",
    "covar_df = pd.read_csv(covar, sep='\\t')\n",
    "\n",
    "\n",
    "if covar_df.age_of_onset.isna().all():\n",
    "    covar_df.drop(columns=['age_of_onset'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age_of_onset'] = scaler.fit_transform(covar_df[['age_of_onset']])\n",
    "\n",
    "if covar_df.age.isna().all():\n",
    "    covar_df.drop(columns=['age'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age'] = scaler.fit_transform(covar_df[['age']])\n",
    "\n",
    "if covar_df.sex.isna().all():\n",
    "    covar_df.drop(columns=['sex'], inplace=True)\n",
    "\n",
    "covar_df.drop(columns=['FID'], inplace=True)\n",
    "covar_df.rename(columns={'GP2sampleID':'sampleid'}, inplace=True)\n",
    "\n",
    "data_df = dosage_df.merge(covar_df, on='sampleid', how='left').merge(pheno_df, on='sampleid', how='left').set_index('sampleid')\n",
    "\n",
    "rm_pred = [f'PC{i}' for i in range(1,21)] + ['sex','age_of_onset','age','pheno']\n",
    "\n",
    "pred_list = [x for x in data_df.columns if x not in rm_pred]\n",
    "covars_list = [x for x in data_df.columns if x not in pred_list + [f'PC{i}' for i in range(11,21)] + ['pheno']]\n",
    "\n",
    "\n",
    "genes_list = dosage_df.columns.drop('sampleid')\n",
    "data_df_final = data_df.copy()\n",
    "ctrl_stds = data_df.loc[data_df['pheno']==0, genes_list].std()\n",
    "ctrl_means = data_df.loc[data_df['pheno']==0, genes_list].mean()\n",
    "\n",
    "for gene in genes_list:\n",
    "    lower = (ctrl_means[gene]-(ctrl_stds[gene]*2))\n",
    "    upper = (ctrl_means[gene]+(ctrl_stds[gene]*2))         \n",
    "    data_df_final.loc[:, gene] = data_df_final.loc[:, gene].between(lower, upper, inclusive='both')\n",
    "    \n",
    "data_df_final.loc[:, genes_list] = data_df_final.loc[:, genes_list].astype(int)\n",
    "\n",
    "results = []\n",
    "fails = []\n",
    "\n",
    "for pred in range(len(pred_list)):\n",
    "    pred_name = pred_list[pred]\n",
    "    formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "\n",
    "    fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df_final).fit()\n",
    "\n",
    "    beta_coef  = fitted.params.loc[pred_name]\n",
    "    beta_se  = fitted.bse.loc[pred_name]\n",
    "    p_val = fitted.pvalues.loc[pred_name]\n",
    "\n",
    "    results.append((pred_name, beta_coef, beta_se, p_val))\n",
    "\n",
    "\n",
    "output = pd.DataFrame(results, columns=('PREDICTOR', 'BETA_COEF', 'BETA_SE','P_VAL'))\n",
    "# output.to_csv(out_path, sep='\\t', header=True, index=False)\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# def CNV_WAS(cnv_dosage_file, pheno, covar, out_path):\n",
    "#     scaler = MinMaxScaler()\n",
    "#     dosage_df = pd.read_csv(cnv_dosage_file)\n",
    "#     #fix column names\n",
    "#     dosage_df.columns = [x.replace('-','_') for x in dosage_df.columns]\n",
    "#     dosage_df.columns = [x.replace('.','_') for x in dosage_df.columns]\n",
    "#     dosage_df.columns = [x.replace(' ','_') for x in dosage_df.columns]\n",
    "\n",
    "#     pheno_df = pd.read_csv(pheno, sep='\\t')\n",
    "#     covar_df = pd.read_csv(covar, sep='\\t')\n",
    "\n",
    "\n",
    "#     if covar_df.age_of_onset.isna().all():\n",
    "#         covar_df.drop(columns=['age_of_onset'], inplace=True)\n",
    "#     else:\n",
    "#         covar_df.loc[:,'age_of_onset'] = scaler.fit_transform(covar_df[['age_of_onset']])\n",
    "\n",
    "#     if covar_df.age.isna().all():\n",
    "#         covar_df.drop(columns=['age'], inplace=True)\n",
    "#     else:\n",
    "#         covar_df.loc[:,'age'] = scaler.fit_transform(covar_df[['age']])\n",
    "\n",
    "#     if covar_df.sex.isna().all():\n",
    "#         covar_df.drop(columns=['sex'], inplace=True)\n",
    "\n",
    "#     covar_df.drop(columns=['FID'], inplace=True)\n",
    "#     covar_df.rename(columns={'GP2sampleID':'sampleid'}, inplace=True)\n",
    "\n",
    "#     data_df = dosage_df.merge(covar_df, on='sampleid', how='left').merge(pheno_df, on='sampleid', how='left').set_index('sampleid')\n",
    "\n",
    "#     rm_pred = [f'PC{i}' for i in range(1,21)] + ['sex','age_of_onset','age','pheno']\n",
    "\n",
    "#     pred_list = [x for x in data_df.columns if x not in rm_pred]\n",
    "#     covars_list = [x for x in data_df.columns if x not in pred_list + [f'PC{i}' for i in range(11,21)] + ['pheno']]\n",
    "\n",
    "\n",
    "#     genes_list = dosage_df.columns.drop('sampleid')\n",
    "#     data_df_final = data_df.copy()\n",
    "#     ctrl_stds = data_df.loc[data_df['pheno']==0, genes_list].std()\n",
    "#     ctrl_means = data_df.loc[data_df['pheno']==0, genes_list].mean()\n",
    "\n",
    "#     for gene in genes_list:\n",
    "#         lower = (ctrl_means[gene]-(ctrl_stds[gene]*2))\n",
    "#         upper = (ctrl_means[gene]+(ctrl_stds[gene]*2))         \n",
    "#         data_df_final.loc[:, gene] = data_df_final.loc[:, gene].between(lower, upper, inclusive='both')\n",
    "\n",
    "#     data_df_final.loc[:, genes_list] = data_df_final.loc[:, genes_list].astype(int)\n",
    "\n",
    "#     results = []\n",
    "#     fails = []\n",
    "\n",
    "#     for pred in range(len(pred_list)):\n",
    "#         pred_name = pred_list[pred]\n",
    "#         formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "\n",
    "#         fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df_final).fit()\n",
    "#         beta_coef  = fitted.params.loc[pred_name]\n",
    "#         beta_se  = fitted.bse.loc[pred_name]\n",
    "#         p_val = fitted.pvalues.loc[pred_name]\n",
    "\n",
    "#         results.append((pred_name, beta_coef, beta_se, p_val))\n",
    "\n",
    "\n",
    "#     output = pd.DataFrame(results, columns=('PREDICTOR', 'BETA_COEF', 'BETA_SE','P_VAL'))\n",
    "#     output.to_csv(out_path, sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "9b025936-4e9c-4281-8f63-f3ee1a68ac49",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'lambda_calculation',\n",
       " 'metrics': {'inflation': 1.4366622566944616}}"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from GWAS.gwas import calculate_inflation\n",
    "# pheno_df = pd.read_csv(f'{cnv_path}/GP2_EUR_chr1_{cnv_type}.pheno', sep='\\t', engine='python')\n",
    "ncontrols = pheno_df.pheno.value_counts()[0]\n",
    "ncases = pheno_df.pheno.value_counts()[1]\n",
    "\n",
    "infl = calculate_inflation(output.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "infl\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "20016864-9700-45f7-9cc5-c9c07d995d62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'pheno ~ WDR64 + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + sex + age + age_of_onset'"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "formula"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "2b45610b-ad6e-4de8-a407-512f1e04503c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACTN2</th>\n",
       "      <th>AGTRAP</th>\n",
       "      <th>ARHGEF16</th>\n",
       "      <th>C1orf101</th>\n",
       "      <th>C1orf167</th>\n",
       "      <th>CAMTA1</th>\n",
       "      <th>CSMD2</th>\n",
       "      <th>DAB1</th>\n",
       "      <th>DRAXIN</th>\n",
       "      <th>EDARADD</th>\n",
       "      <th>...</th>\n",
       "      <th>PC15</th>\n",
       "      <th>PC16</th>\n",
       "      <th>PC17</th>\n",
       "      <th>PC18</th>\n",
       "      <th>PC19</th>\n",
       "      <th>PC20</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>age_of_onset</th>\n",
       "      <th>pheno</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CORIELL_000178_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.006651</td>\n",
       "      <td>0.009075</td>\n",
       "      <td>-0.007526</td>\n",
       "      <td>0.000607</td>\n",
       "      <td>-0.005277</td>\n",
       "      <td>0.016480</td>\n",
       "      <td>1</td>\n",
       "      <td>0.646341</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAGE_000432_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.009945</td>\n",
       "      <td>0.006582</td>\n",
       "      <td>0.005732</td>\n",
       "      <td>-0.005267</td>\n",
       "      <td>-0.017189</td>\n",
       "      <td>-0.012584</td>\n",
       "      <td>1</td>\n",
       "      <td>0.767308</td>\n",
       "      <td>0.741228</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_000783_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001914</td>\n",
       "      <td>-0.002026</td>\n",
       "      <td>-0.007720</td>\n",
       "      <td>-0.002451</td>\n",
       "      <td>-0.003864</td>\n",
       "      <td>-0.006665</td>\n",
       "      <td>2</td>\n",
       "      <td>0.524390</td>\n",
       "      <td>0.578947</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_004856_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006815</td>\n",
       "      <td>-0.005230</td>\n",
       "      <td>-0.006799</td>\n",
       "      <td>0.006651</td>\n",
       "      <td>0.005691</td>\n",
       "      <td>-0.003396</td>\n",
       "      <td>1</td>\n",
       "      <td>0.621951</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COPN_000306_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004314</td>\n",
       "      <td>0.003814</td>\n",
       "      <td>0.000499</td>\n",
       "      <td>0.010819</td>\n",
       "      <td>0.001542</td>\n",
       "      <td>0.018862</td>\n",
       "      <td>1</td>\n",
       "      <td>0.548780</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAGE_001054_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.008115</td>\n",
       "      <td>-0.003489</td>\n",
       "      <td>0.001157</td>\n",
       "      <td>-0.002412</td>\n",
       "      <td>-0.000864</td>\n",
       "      <td>-0.004991</td>\n",
       "      <td>1</td>\n",
       "      <td>0.722300</td>\n",
       "      <td>0.653509</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LCC_000146_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008954</td>\n",
       "      <td>-0.009757</td>\n",
       "      <td>0.003442</td>\n",
       "      <td>-0.000514</td>\n",
       "      <td>-0.001066</td>\n",
       "      <td>-0.003567</td>\n",
       "      <td>1</td>\n",
       "      <td>0.548780</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_002459_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001957</td>\n",
       "      <td>0.003340</td>\n",
       "      <td>0.000891</td>\n",
       "      <td>-0.007043</td>\n",
       "      <td>-0.000739</td>\n",
       "      <td>0.000489</td>\n",
       "      <td>1</td>\n",
       "      <td>0.536585</td>\n",
       "      <td>0.484211</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCM_000036_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.006514</td>\n",
       "      <td>-0.008966</td>\n",
       "      <td>-0.010393</td>\n",
       "      <td>-0.007601</td>\n",
       "      <td>0.000784</td>\n",
       "      <td>-0.014942</td>\n",
       "      <td>2</td>\n",
       "      <td>0.571582</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_001134_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006826</td>\n",
       "      <td>0.008794</td>\n",
       "      <td>-0.007286</td>\n",
       "      <td>0.005986</td>\n",
       "      <td>-0.005052</td>\n",
       "      <td>-0.010628</td>\n",
       "      <td>1</td>\n",
       "      <td>0.304878</td>\n",
       "      <td>0.378947</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12436 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   ACTN2  AGTRAP  ARHGEF16  C1orf101  C1orf167  CAMTA1  CSMD2  \\\n",
       "sampleid                                                                        \n",
       "CORIELL_000178_s1      1       1         1         1         1       1      1   \n",
       "PAGE_000432_s1         1       1         1         1         1       1      0   \n",
       "CORIELL_000783_s1      1       1         1         1         1       1      1   \n",
       "CORIELL_004856_s1      1       1         1         1         1       1      1   \n",
       "COPN_000306_s1         1       1         1         1         1       1      1   \n",
       "...                  ...     ...       ...       ...       ...     ...    ...   \n",
       "PAGE_001054_s1         1       1         1         1         1       1      1   \n",
       "LCC_000146_s1          1       1         1         1         1       1      1   \n",
       "CORIELL_002459_s1      1       1         1         1         1       1      1   \n",
       "BCM_000036_s1          1       1         1         1         1       1      1   \n",
       "CORIELL_001134_s1      1       1         0         1         1       1      0   \n",
       "\n",
       "                   DAB1  DRAXIN  EDARADD  ...      PC15      PC16      PC17  \\\n",
       "sampleid                                  ...                                 \n",
       "CORIELL_000178_s1     1       1        1  ... -0.006651  0.009075 -0.007526   \n",
       "PAGE_000432_s1        0       1        1  ... -0.009945  0.006582  0.005732   \n",
       "CORIELL_000783_s1     1       1        1  ...  0.001914 -0.002026 -0.007720   \n",
       "CORIELL_004856_s1     1       1        1  ...  0.006815 -0.005230 -0.006799   \n",
       "COPN_000306_s1        1       1        1  ...  0.004314  0.003814  0.000499   \n",
       "...                 ...     ...      ...  ...       ...       ...       ...   \n",
       "PAGE_001054_s1        1       1        1  ... -0.008115 -0.003489  0.001157   \n",
       "LCC_000146_s1         1       1        1  ...  0.008954 -0.009757  0.003442   \n",
       "CORIELL_002459_s1     1       1        1  ...  0.001957  0.003340  0.000891   \n",
       "BCM_000036_s1         1       1        1  ... -0.006514 -0.008966 -0.010393   \n",
       "CORIELL_001134_s1     1       1        1  ...  0.006826  0.008794 -0.007286   \n",
       "\n",
       "                       PC18      PC19      PC20  sex       age  age_of_onset  \\\n",
       "sampleid                                                                       \n",
       "CORIELL_000178_s1  0.000607 -0.005277  0.016480    1  0.646341      0.600000   \n",
       "PAGE_000432_s1    -0.005267 -0.017189 -0.012584    1  0.767308      0.741228   \n",
       "CORIELL_000783_s1 -0.002451 -0.003864 -0.006665    2  0.524390      0.578947   \n",
       "CORIELL_004856_s1  0.006651  0.005691 -0.003396    1  0.621951      0.577225   \n",
       "COPN_000306_s1     0.010819  0.001542  0.018862    1  0.548780      0.577225   \n",
       "...                     ...       ...       ...  ...       ...           ...   \n",
       "PAGE_001054_s1    -0.002412 -0.000864 -0.004991    1  0.722300      0.653509   \n",
       "LCC_000146_s1     -0.000514 -0.001066 -0.003567    1  0.548780      0.577225   \n",
       "CORIELL_002459_s1 -0.007043 -0.000739  0.000489    1  0.536585      0.484211   \n",
       "BCM_000036_s1     -0.007601  0.000784 -0.014942    2  0.571582      0.577225   \n",
       "CORIELL_001134_s1  0.005986 -0.005052 -0.010628    1  0.304878      0.378947   \n",
       "\n",
       "                   pheno  \n",
       "sampleid                  \n",
       "CORIELL_000178_s1      1  \n",
       "PAGE_000432_s1         1  \n",
       "CORIELL_000783_s1      1  \n",
       "CORIELL_004856_s1      0  \n",
       "COPN_000306_s1         1  \n",
       "...                  ...  \n",
       "PAGE_001054_s1         1  \n",
       "LCC_000146_s1          0  \n",
       "CORIELL_002459_s1      1  \n",
       "BCM_000036_s1          1  \n",
       "CORIELL_001134_s1      1  \n",
       "\n",
       "[12436 rows x 61 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "data_df_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d5d2df9-3901-4321-8672-2d427082e1ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41dda188-496d-4149-887d-990acff9b742",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f4b0d49-c3a2-42b8-9156-9172bc4aadf6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41cb69b5-c464-4cc4-be12-d8e1704e84d4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "6081e4d8-a887-412f-98b2-77240260d7b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# output final release samples\n",
    "cnv_out_path = '/data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs'\n",
    "release3_samples = '/data/CARD/PD/GP2/genotypes/GP2/round3/clean/release3.samples'\n",
    "release_key_final = pd.read_csv('/data/CARD/PD/GP2/genotypes/GP2/round3/clean/release3_master_key.csv')\n",
    "release_key_out = release_key_final.loc[release_key_final.pheno!=-9]\n",
    "release_key_out['GP2sampleID'].to_csv(release3_samples, index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "82dc85bb-6d29-405e-bed6-efab5c1cc434",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/cnv_dosage_release_split.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        sample_file = f'/data/CARD/PD/GP2/genotypes/GP2/round3/clean/{label}_release3.samples'\n",
    "        label_samples = release_key_final.loc[release_key_final['label']==label]\n",
    "        label_samples['GP2sampleID'].to_csv(sample_file, header=False, index=False)\n",
    "\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "                \n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                dosage_out = f'{cnv_out_path}/CNV_{label}_chr{chrom}_{cnv_type}_release3.csv'\n",
    "                \n",
    "                cmd = f'\\\n",
    "python split_release_samples.py \\\n",
    "--cnv_dosage_file {dosagefile} \\\n",
    "--samples {sample_file} \\\n",
    "--out_path {dosage_out}'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "56353619-d819-4347-ab81-aec8d05de647",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "53715605\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_dosage_release_split.swarm -g 8 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "64f9b5b4-92c2-430c-a9ec-b09aae6aab6b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AMR PERCENT_BAF_INSERTION\n",
      "412\n",
      "\n",
      "AMR PERCENT_L2R_DELETION\n",
      "412\n",
      "\n",
      "AMR PERCENT_L2R_DUPLICATION\n",
      "412\n",
      "\n",
      "EUR PERCENT_BAF_INSERTION\n",
      "10727\n",
      "\n",
      "EUR PERCENT_L2R_DELETION\n",
      "10727\n",
      "\n",
      "EUR PERCENT_L2R_DUPLICATION\n",
      "10727\n",
      "\n",
      "SAS PERCENT_BAF_INSERTION\n",
      "59\n",
      "\n",
      "SAS PERCENT_L2R_DELETION\n",
      "59\n",
      "\n",
      "SAS PERCENT_L2R_DUPLICATION\n",
      "59\n",
      "\n",
      "MDE PERCENT_BAF_INSERTION\n",
      "62\n",
      "\n",
      "MDE PERCENT_L2R_DELETION\n",
      "62\n",
      "\n",
      "MDE PERCENT_L2R_DUPLICATION\n",
      "62\n",
      "\n",
      "EAS PERCENT_BAF_INSERTION\n",
      "137\n",
      "\n",
      "EAS PERCENT_L2R_DELETION\n",
      "137\n",
      "\n",
      "EAS PERCENT_L2R_DUPLICATION\n",
      "137\n",
      "\n",
      "AAC PERCENT_BAF_INSERTION\n",
      "1215\n",
      "\n",
      "AAC PERCENT_L2R_DELETION\n",
      "1215\n",
      "\n",
      "AAC PERCENT_L2R_DUPLICATION\n",
      "1215\n",
      "\n",
      "AFR PERCENT_BAF_INSERTION\n",
      "404\n",
      "\n",
      "AFR PERCENT_L2R_DELETION\n",
      "404\n",
      "\n",
      "AFR PERCENT_L2R_DUPLICATION\n",
      "404\n",
      "\n",
      "CAS PERCENT_BAF_INSERTION\n",
      "303\n",
      "\n",
      "CAS PERCENT_L2R_DELETION\n",
      "303\n",
      "\n",
      "CAS PERCENT_L2R_DUPLICATION\n",
      "303\n",
      "\n",
      "AJ PERCENT_BAF_INSERTION\n",
      "923\n",
      "\n",
      "AJ PERCENT_L2R_DELETION\n",
      "923\n",
      "\n",
      "AJ PERCENT_L2R_DUPLICATION\n",
      "923\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# CHECK COUNTS\n",
    "total_dosages = 0\n",
    "for label in ancestry_labels:\n",
    "    for cnv_type in cnv_types:\n",
    "        dosage_out = f'{cnv_out_path}/CNV_{label}_chr1_{cnv_type}_release3.csv'\n",
    "        print(label, cnv_type)\n",
    "        label_count = !cat {dosage_out} | wc -l\n",
    "        label_count = int(label_count[0])\n",
    "        total_dosages += label_count\n",
    "        print(label_count)\n",
    "        print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "3df93e96-b169-4e45-ad32-76904be6cf22",
   "metadata": {},
   "outputs": [],
   "source": [
    "# push to release bucket\n",
    "with open(f'{swarm_scripts_dir}/gcp_push_cnvs.swarm','w') as f:\n",
    "    \n",
    "    for label in ancestry_labels:\n",
    "        for chrom in chroms:\n",
    "    \n",
    "#         !module load google-cloud-sdk/397.0.0; gsutil cp {cnv_out_path}/CNV_{label}_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/\n",
    "            cp1 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "            cp2 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_L2R_DELETION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "            cp3 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_L2R_DUPLICATION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "\n",
    "            cps = [cp1, cp2, cp3]\n",
    "\n",
    "            for cp in cps:\n",
    "                f.write(f'{cp}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ddb097af-919f-41f3-a434-2b624c4dbde4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "53719849\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/gcp_push_cnvs.swarm -g 8 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --module google-cloud-sdk/397.0.0 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "70c7c14a-cb24-4e02-961f-f4fb69834166",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "594\n"
     ]
    }
   ],
   "source": [
    "!module load google-cloud-sdk/397.0.0; gsutil cp /data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs/CNV_AMR_chr1_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/AMR/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "2a133f0b-dc2b-4bce-91d5-5040ca024786",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-rw----+ 1 vitaled2 CARD 4.2K Dec  5 19:27 /data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs/CNV_AJ_chr22_PERCENT_BAF_INSERTION_release3.csv\n"
     ]
    }
   ],
   "source": [
    "!ls -lh {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_BAF_INSERTION_release3.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e4aa47ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11696\n"
     ]
    }
   ],
   "source": [
    "!cat /data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_chr1_PERCENT_BAF_INSERTION_gp2ids.csv | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6951e7ee-bbc6-412f-8e18-921bd5c37b9a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93cea6d2-bf8f-4ad4-841a-7689d75fc5c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16b30b49-d588-442c-bec0-b8613fbb2089",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae39c1ad-f5e6-4c4b-85da-ea425681c9f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "{cnv_path}/genotypes/"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.8",
   "language": "python",
   "name": "py3.8"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
