{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "163da59f-cf9c-4665-8221-8524f6a364e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "import numpy as np\n",
    "import glob\n",
    "from sklearn.preprocessing import MinMaxScaler\n",
    "import statsmodels.api as sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ac49c984-53cf-4aca-9b10-2c20e1321d8d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_11989/1889251462.py:17: DtypeWarning: Columns (7,10,27,36,37,39) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  key = pd.read_csv(f'{key_file}', sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "idat_path = '/data/GP2/raw_genotypes/idats/'\n",
    "swarm_scripts_dir = f'/data/GP2/users/vitaled2/swarm'\n",
    "\n",
    "ilmn_files_path = '/data/CARD/PD/GP2/ilmn_files'\n",
    "# A1 is hg19, A2 is hg38. csv needed for indel calls\n",
    "bpm_csv = f'{ilmn_files_path}/NeuroBooster_20042459_A2.csv' \n",
    "bpm = f'{ilmn_files_path}/NeuroBooster_20042459_A2.bpm'\n",
    "egt = f'{ilmn_files_path}/recluster_09272022.egt'\n",
    "iaap = f'{ilmn_files_path}/iaap-cli/iaap-cli'\n",
    "\n",
    "ref_fasta = '/data/vitaled2/ref/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna'\n",
    "\n",
    "chroms = [str(i) for i in range(1,23)] + ['X','Y','M']\n",
    "\n",
    "\n",
    "key_file = '/data/GP2/clinical/master_key/GP2_master_key_APRIL2023_ALL_cohorts_n35802.txt'\n",
    "key = pd.read_csv(f'{key_file}', sep='\\t')\n",
    "key.loc[:,'filename'] = key.loc[:,'SentrixBarcode_A'].astype(str) + '_' + key.loc[:,'SentrixPosition_A'].astype(str)\n",
    "\n",
    "cohorts = ['APGS',\n",
    " 'BCM',\n",
    " 'BIOFIND',\n",
    " 'BLAACPD-KPM',\n",
    " 'BLAACPD-RUSH',\n",
    " 'BLAACPD-UAB',\n",
    " 'BLAACPD-UC',\n",
    " 'COPN',\n",
    " 'CORIELL',\n",
    " 'KOC',\n",
    " 'LCC',\n",
    " 'MDGAP-EBB',\n",
    " 'MDGAP-KINGS',\n",
    " 'MDGAP-QSBB',\n",
    " 'NZP3',\n",
    " 'PAGE',\n",
    " 'PPMI',\n",
    " 'S4',\n",
    " 'SYDBB',\n",
    " 'SYNAPS-KZ',\n",
    " 'UMD']\n",
    "\n",
    "release_key = key.loc[key['study'].isin(cohorts)]\n",
    "release_key.to_csv('/data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv', index=False, header=True)\n",
    "\n",
    "###################### EDIT THIS BACK IN ONCE WE HAVE FULL LIST OF RELEASE SAMPLES!!!! #########################\n",
    "# samples_list = release_key.loc[:,'filename'].unique()\n",
    "# barcodes_list = list(set([x.split('_')[0] for x in samples_list]))\n",
    "\n",
    "samples_list = key.loc[:,'filename'].unique()\n",
    "barcodes_list = list(set([x.split('_')[0] for x in samples_list]))\n",
    "\n",
    "################################################################################################################\n",
    "\n",
    "# genes = pd.read_csv('/data/GP2/utils/ref_dir/glist-hg38', sep='\\s+', header=None, names=['chr','start','end','symbol'], dtype={'chr':str,'start':int,'end':int})\n",
    "# genes.columns = ['CHR','START','STOP','NAME']\n",
    "# genes_out = genes.loc[genes.CHR.isin(chroms)]\n",
    "# genes_out[['NAME','CHR','START','STOP']].to_csv('/data/GP2/utils/ref_dir/glist_hg38_intervals.csv', index=False)\n",
    "# # gene_list originally from plink glist\n",
    "# gene_list = '/data/GP2/utils/ref_dir/glist_hg38_intervals.csv'\n",
    "# !cp /data/CARD/PD/GP2/ref_panel/glist_hg38_chr*.csv /data/GP2/utils/ref_dir/\n",
    "\n",
    "raw_geno_path = '/data/GP2/raw_genotypes'\n",
    "snp_metrics_path = f'{raw_geno_path}/snp_metrics'\n",
    "idat_path = f'{raw_geno_path}/idats'\n",
    "plink_file_path = f'{raw_geno_path}/ped_bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "59346c9d-e1bd-47fd-b503-c9f592ff4dd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "35802"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# len(samples_list)\n",
    "# sorted(cohorts)\n",
    "# len(release_key)\n",
    "len(key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d146821f-30ff-496e-a9ad-bd2e1b90ac08",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get snp metrics\n",
    "metrics_exist_list = []\n",
    "bcftools_plugins_path = 'bin'\n",
    "with open(f'{swarm_scripts_dir}/snp_metrics.swarm', 'w') as f:\n",
    "    for code in barcodes_list:\n",
    "        idat_path_ = f'{idat_path}/{code}'\n",
    "        metrics_out = f'{snp_metrics_path}'\n",
    "        metrics_code_path = f'{snp_metrics_path}/{code}'\n",
    "        cmd = f'\\\n",
    "python run_snp_metrics_pipeline.py \\\n",
    "--idat_path {idat_path_} \\\n",
    "--bpm {bpm} \\\n",
    "--bpm_csv {bpm_csv} \\\n",
    "--egt {egt} \\\n",
    "--ref_fasta {ref_fasta} \\\n",
    "--out_path {metrics_out} \\\n",
    "--iaap {iaap} \\\n",
    "--bcftools_plugins_path {bcftools_plugins_path}'\n",
    "        f.write(f'{cmd}\\n')\n",
    "    f.close()\n",
    "\n",
    "#         if not os.path.exists(metrics_code_path):\n",
    "#             cmd = f'\\\n",
    "# python run_snp_metrics_pipeline.py \\\n",
    "# --idat_path {idat_path_} \\\n",
    "# --bpm {bpm} \\\n",
    "# --bpm_csv {bpm_csv} \\\n",
    "# --egt {egt} \\\n",
    "# --ref_fasta {ref_fasta} \\\n",
    "# --out_path {metrics_out} \\\n",
    "# --iaap {iaap} \\\n",
    "# --bcftools_plugins_path {bcftools_plugins_path}'\n",
    "#             f.write(f'{cmd}\\n')\n",
    "        \n",
    "#         else:\n",
    "#             metrics_exist_list.append(code)\n",
    "        \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "117e910d-509a-4ff5-af24-e5f3d07da1fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "63604578\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/snp_metrics.swarm -g 16 -t 16 --time=04:00:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "933d2fbb-7730-4392-bf53-c543edac17f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4523\n"
     ]
    }
   ],
   "source": [
    "!cat {swarm_scripts_dir}/snp_metrics.swarm | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2e36d7b-97b5-442e-9448-a12a29a6b1fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206455490026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990154/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300080/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260069/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260017/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260071/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000136/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990119/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260139/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260027/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990124/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610073/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260044/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610077/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610076/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290023/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610095/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590038/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380170/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700077/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610030/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380089/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070180/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990114/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300091/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990161/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590096/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206455490034/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260062/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610071/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070119/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990065/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610046/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590079/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240027/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990078/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260039/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380114/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000095/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260097/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680046/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260054/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610052/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206455490036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990171/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990120/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240051/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000137/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206917560031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990017/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060050/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990104/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260053/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070169/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000109/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680088/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990087/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070150/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260056/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700074/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380002/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750047/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990101/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070069/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750017/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610056/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260041/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680069/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000175/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990116/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290004/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000111/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610108/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205209570035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205047670018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260147/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380001/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380067/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990047/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190089/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207086640005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590074/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610115/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990084/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260067/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590080/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990054/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190106/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580070/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990041/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000163/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680047/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180039/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680054/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890350105/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990040/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206917560011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990133/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000153/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205243960008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083620031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700030/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590047/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300089/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590105/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990091/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206714030032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590100/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680097/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380133/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180097/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610057/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205951460068/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810023/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000079/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750001/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610050/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260113/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060194/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700004/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000094/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070139/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610119/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260068/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206887890067/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260076/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990162/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590062/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590061/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610060/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590103/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680023/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590077/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190126/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990128/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590095/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990115/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260073/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700047/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260092/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680041/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380169/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990058/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380086/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700038/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590075/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990121/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810004/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300100/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990147/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000142/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260081/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240058/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000165/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260101/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000073/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206391680085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990125/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680068/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610101/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990044/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260111/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260040/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070170/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260112/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990181/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240030/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590046/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206714030033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070192/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990183/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610078/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260060/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610057/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000144/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240044/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260119/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580131/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610087/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990155/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680076/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070173/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990062/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680057/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260074/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207086640085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204620380005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260058/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053530090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890350100/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206063060016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990034/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260168/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750055/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360075/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260030/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280060/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207086640029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380143/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205951460040/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810017/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380113/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070185/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590081/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206917560022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610027/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990112/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750023/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700045/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380088/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260075/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053510141/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680095/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206544470035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580075/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190168/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610065/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180127/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204701870011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300107/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190045/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380061/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590053/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000092/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260106/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990126/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190063/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206714030007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380062/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070138/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053520032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260118/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380060/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083620065/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204835570015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610038/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810034/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060034/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000050/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207086640035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060185/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610055/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610078/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680056/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260118/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260023/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380096/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680040/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990152/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380044/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000096/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590097/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990164/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000134/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990050/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290017/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990055/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990052/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680070/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610047/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990086/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000127/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990068/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000116/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990045/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260045/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990030/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260084/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610106/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990063/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000070/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260004/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680094/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180107/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680071/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053520105/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053510083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990066/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070167/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990061/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053530071/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260107/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260098/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750045/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060164/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990023/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206964480012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810030/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190189/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260001/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990049/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053510142/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240050/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204835570009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890350108/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380091/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204701870012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590101/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380077/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000135/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990156/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990139/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206714030008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204842400179/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680092/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150017/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190107/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380171/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205209570010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260115/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060098/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610070/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000164/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380087/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206714030031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205047670011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204701870110/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240041/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260104/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000097/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053530120/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610094/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380065/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610077/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610099/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060183/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280052/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990109/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990186/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260110/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610092/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990098/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260117/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300084/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206714030009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205951460038/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000168/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990004/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990089/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190133/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580132/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000081/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180038/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990129/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380041/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190023/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380046/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590107/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190051/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300097/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206455490013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083620058/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060106/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750036/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083620113/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580162/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610098/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810001/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070104/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053510164/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700041/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205209570083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150069/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590106/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580144/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990051/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260146/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000164/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190179/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000091/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990180/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700040/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070142/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990131/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700006/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260055/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680087/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680091/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000052/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060124/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590045/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260087/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280066/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205951460032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053530148/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206904780010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680052/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610131/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590078/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990076/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260063/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260108/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990122/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680034/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300095/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190148/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610103/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060049/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190178/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990142/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810002/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206455490029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680031/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260072/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590099/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053520018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580137/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990099/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180165/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280058/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205275450005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700001/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180071/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070175/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610060/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180112/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053530100/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204820990007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205178650019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000141/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590056/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050073/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360091/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590055/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700002/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206038620003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990060/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290001/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590052/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380058/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990110/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204842400062/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260002/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680089/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206917560010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206544470037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260109/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580115/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280034/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610061/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810027/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590073/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380063/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260050/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610089/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083620026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580126/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260114/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070149/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180100/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207086640126/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610084/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990053/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260077/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260038/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204701860020/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204701870042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260088/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070156/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000098/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053510069/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610041/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260100/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180050/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204835450143/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360057/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590086/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260109/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610027/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190150/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206887870058/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206455490042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000013/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300088/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360109/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000123/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053530007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206917560032/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000080/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680067/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070105/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060016/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060129/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260087/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680096/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590051/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260148/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204701860097/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890350082/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300106/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206112300079/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260120/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204701860064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205243960102/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000154/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206917560024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990081/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590084/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590054/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070168/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380065/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380026/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380094/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610076/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083610043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380125/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207078060048/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050008/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380085/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000059/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260116/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205271260070/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205209570007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380145/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060137/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260078/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000138/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990080/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050075/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070155/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580064/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053520053/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260080/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205951460065/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204842400154/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070172/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990135/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590057/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260061/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260057/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070068/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700075/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260115/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990132/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260070/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000007/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380112/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260009/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070191/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990039/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610105/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205053530025/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260005/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083620019/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260091/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260028/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990117/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207087150018/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260089/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990190/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580150/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206154290022/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260117/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000138/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380095/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990106/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610104/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206299380043/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680033/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206890360076/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990037/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070154/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180084/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046190029/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700076/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206952000113/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000099/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207083620069/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206904780011/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700073/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207086640021/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060113/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070120/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205951460051/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990141/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206046180035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990093/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206455490040/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700046/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680044/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590094/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/205746280035/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610067/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206451070179/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412060024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990151/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990144/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680090/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000105/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990067/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680084/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260024/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206412050003/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206978580152/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680012/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990177/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206315610072/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204835450010/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/204835450115/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000072/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680045/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260034/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260065/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700078/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260105/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700017/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206949990100/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680074/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206862810015/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207121700044/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260042/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/207138680027/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206966380044/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206838590083/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206530750014/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206191240049/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206200000063/*.csv’: No such file or directory\n",
      "rm: cannot remove ‘/data/GP2/raw_genotypes/snp_metrics/206117260026/*.csv’: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "# !cat {swarm_scripts_dir}/logs/swarm_63587794_948.e\n",
    "# m_exists = []\n",
    "# m_nonexist = []\n",
    "for code in barcodes_list:\n",
    "    metrics = f'{snp_metrics_path}/{code}/*.csv'\n",
    "    !rm {metrics}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c829e9f8-ce80-4c2a-a5c7-b73f1ca12767",
   "metadata": {},
   "outputs": [],
   "source": [
    "metrics = pd.read_csv('/data/GP2/raw_genotypes/snp_metrics/206412060175/snp_metrics_206412060175_R02C01_chr2.csv')\n",
    "metrics.to_parquet('/data/GP2/raw_genotypes/snp_metrics/206412060175/snp_metrics_206412060175_R02C01_chr2.parquet', compression='brotli')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "fb0c0853-783b-4123-b576-998a1881e30b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-r----- 1 vitaled2 GP2  18M Feb  3 19:07 /data/GP2/raw_genotypes/snp_metrics/206412060175/snp_metrics_206412060175_R02C01_chr2.csv\n",
      "-rw-r----- 1 vitaled2 GP2 6.2M Apr 24 22:46 /data/GP2/raw_genotypes/snp_metrics/206412060175/snp_metrics_206412060175_R02C01_chr2.parquet\n"
     ]
    }
   ],
   "source": [
    "!ls -lh {snp_metrics_path}/{code}/snp_metrics_206412060175_R02C01_chr2.*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3dedbf84-fce1-4533-b1e3-86b74cf57bac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c865eab1-13b6-4457-a5b4-81e4b10b1c94",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "cffc06e3-dfa9-4654-b599-d63c485b3b28",
   "metadata": {},
   "outputs": [],
   "source": [
    "# check if snp metrics produced across all samples\n",
    "completed_list = []\n",
    "missing_list = []\n",
    "for sample in samples_list:\n",
    "    code = sample.split('_')[0]\n",
    "    metrics_path = f'{snp_metrics_path}/{code}'\n",
    "    for chrom in chroms:\n",
    "        mfile = f'{metrics_path}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "        if os.path.isfile(mfile):\n",
    "            completed_list.append(sample)\n",
    "        else:\n",
    "            missing_list.append(sample)\n",
    "completed_final = list(set(completed_list))\n",
    "missing_final =list(set(missing_list))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "b7c4dff8-569a-4d46-b4f2-186363aacb39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18974\n",
      "83\n"
     ]
    }
   ],
   "source": [
    "print(len(completed_final))\n",
    "print(len(missing_final))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "bcd9284a-4a05-46d9-892e-6bda06354195",
   "metadata": {},
   "outputs": [],
   "source": [
    "# now check if we have idats for missing snp metrics\n",
    "missing_idats = []\n",
    "exists_idats = []\n",
    "for sample in missing_final:\n",
    "    code = sample.split('_')[0]\n",
    "    grn = f'{idat_path}/{code}/{sample}_Grn.idat'\n",
    "    red = f'{idat_path}/{code}/{sample}_Red.idat'\n",
    "    if os.path.isfile(grn) & os.path.isfile(red):\n",
    "        exists_idats.append(sample)\n",
    "    else:\n",
    "        missing_idats.append(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "cc859aa9-0359-427f-ae76-0f0fb2a4596e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "49\n",
      "34\n"
     ]
    }
   ],
   "source": [
    "print(len(missing_idats))\n",
    "print(len(exists_idats))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8a949c9a-3b34-422e-bc96-b224e6b86681",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_path = f'/data/GP2/raw_genotypes/cnvs/release4'\n",
    "# using just autosomes for CNV analysis due to call quality\n",
    "chroms = [str(i) for i in range(1,23)]\n",
    "\n",
    "ancestry_labels = [x.split('/')[-1].replace('.bed','').split('_')[-1] for x in glob.glob('/data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023*.bed') if '_maf_hwe' not in x]\n",
    "\n",
    "release_key[['FID','GP2sampleID']].to_csv(f'{cnv_path}/key/release4.samples', sep='\\t', header=False, index=False)\n",
    "release_key[['GP2sampleID','IID']].to_csv(f'{cnv_path}/key/release4_sample_id_key.csv')\n",
    "release_covars = release_key.loc[:,['FID', 'GP2sampleID','sex_for_qc', 'age', 'age_of_onset']]\n",
    "\n",
    "cnv_types = ['PERCENT_BAF_INSERTION','PERCENT_L2R_DELETION','PERCENT_L2R_DUPLICATION']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb45e769-cf67-4ecf-ba94-56cfdb6b4ec4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "for label in ancestry_labels:\n",
    "    geno_name = f'GP2_round4_JAN_24_2023_{label}'\n",
    "    out_path = f'{cnv_path}/genotypes/{geno_name}'\n",
    "    geno_path = f'/data/GP2/quality_control/release4/genotype_qc/{geno_name}'\n",
    "    \n",
    "def cnv_qc(cnv_path, release_covars, release_key):\n",
    "\n",
    "        cmd1 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {geno_path} \\\n",
    "--keep {cnv_path}/genotypes/release4.samples \\\n",
    "--make-bed \\\n",
    "--out {out_path}'\n",
    "\n",
    "        cmd2 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {out_path} \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.02 \\\n",
    "--hwe 5e-6 \\\n",
    "--autosome \\\n",
    "--make-bed \\\n",
    "--out {out_path}_maf_geno_hwe'\n",
    "\n",
    "        cmd3 = f'\\\n",
    "module load plink/1.9; \\\n",
    "plink \\\n",
    "--bfile {out_path}_maf_geno_hwe \\\n",
    "--indep-pairwise 1000 10 0.01 \\\n",
    "--autosome \\\n",
    "--out {out_path}_maf_geno_hwe_prune'\n",
    "        \n",
    "        cmd4 = f'\\\n",
    "module load plink/1.9; \\\n",
    "plink \\\n",
    "--bfile {out_path}_maf_geno_hwe \\\n",
    "--extract {out_path}_maf_geno_hwe_prune.prune.in \\\n",
    "--make-bed \\\n",
    "--out {out_path}_clean'\n",
    "\n",
    "        cmd5 = f'\\\n",
    "module load plink/1.9; \\\n",
    "plink \\\n",
    "--bfile {out_path}_clean \\\n",
    "--pca \\\n",
    "--out {out_path}_clean_pcs'\n",
    "\n",
    "        cmds = [cmd1, cmd2, cmd3, cmd4, cmd5]\n",
    "\n",
    "        for cmd in cmds:\n",
    "            !{cmd}\n",
    "\n",
    "        pcs = pd.read_csv(f'{out_path}_clean_pcs.eigenvec', sep='\\s+')\n",
    "        pc_num = pcs.iloc[:, 2:].shape[1]\n",
    "        pc_names = ['FID','GP2sampleID'] + [f'PC{i}' for i in range(1, pc_num+1)]\n",
    "        pcs.columns = pc_names\n",
    "\n",
    "        cov = pcs.merge(release_covars, on=['FID','GP2sampleID'], how='left')\n",
    "        cov.age.fillna(cov.age.mean(), inplace=True)\n",
    "        cov.age_of_onset.fillna(cov.age_of_onset.mean(), inplace=True)\n",
    "        cov.sex_for_qc.fillna(cov.sex_for_qc.median(), inplace=True)\n",
    "        cov.rename(columns={'GP2sampleID':'sampleid','sex_for_qc':'sex'}, inplace=True)\n",
    "        cov.to_csv(f'{out_path}_clean.cov', sep='\\t', header=True, index=False)\n",
    "\n",
    "        samples = cov.merge(release_key[['GP2sampleID','IID']], left_on='sampleid', right_on='GP2sampleID', how='left')\n",
    "        samples['IID'].to_csv(f'{out_path}_clean_barcode.samples', header=False, index=False)\n",
    "\n",
    "        bim = pd.read_csv(f'{out_path}_clean.bim', sep='\\s+', header=None, names=['chr','id','pos','bp','a1','a2'], usecols=['id'])\n",
    "        bim.to_csv(f'{out_path}_clean.snps', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb84cf01-270c-429f-8758-d700fe19740a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48e85f69-2cdc-47a8-aa07-495dca8c0437",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07a6f708-21b3-429a-a7d9-ed3c4aaeeb82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90185b68-e80d-4c03-95c2-dfc06437c9de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ea017911-2cff-4adf-9952-8bb3b71f5a14",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_MDE\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1945776 variants loaded from .bim file.\n",
      "128 people (91 males, 37 females) loaded from .fam.\n",
      "97 phenotype values loaded from .fam.\n",
      "--keep: 121 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 121 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 982 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.997968.\n",
      "1945776 variants and 121 people pass filters and QC.\n",
      "Among remaining phenotypes, 77 are cases and 13 are controls.  (31 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1864661 variants loaded from .bim file.\n",
      "121 people (89 males, 32 females) loaded from .fam.\n",
      "90 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 121 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998046.\n",
      "43282 variants removed due to missing genotype data (--geno).\n",
      "Warning: --hwe observation counts vary by more than 10%.  Consider using\n",
      "--geno, and/or applying different p-value thresholds to distinct subsets of\n",
      "your data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "976342 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "845037 variants and 121 people pass filters and QC.\n",
      "Among remaining phenotypes, 77 are cases and 13 are controls.  (31 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "845037 variants loaded from .bim file.\n",
      "121 people (89 males, 32 females) loaded from .fam.\n",
      "90 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 121 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998798.\n",
      "845037 variants and 121 people pass filters and QC.\n",
      "Among remaining phenotypes, 77 are cases and 13 are controls.  (31 phenotypes\n",
      "are missing.)\n",
      "Pruned 64791 variants from chromosome 1, leaving 169.\n",
      "Pruned 71518 variants from chromosome 2, leaving 204.\n",
      "Pruned 61443 variants from chromosome 3, leaving 164.\n",
      "Pruned 56143 variants from chromosome 4, leaving 155.\n",
      "Pruned 51016 variants from chromosome 5, leaving 133.\n",
      "Pruned 65810 variants from chromosome 6, leaving 140.\n",
      "Pruned 47053 variants from chromosome 7, leaving 133.\n",
      "Pruned 44199 variants from chromosome 8, leaving 114.\n",
      "Pruned 36584 variants from chromosome 9, leaving 104.\n",
      "Pruned 41501 variants from chromosome 10, leaving 118.\n",
      "Pruned 40971 variants from chromosome 11, leaving 115.\n",
      "Pruned 39576 variants from chromosome 12, leaving 98.\n",
      "Pruned 29883 variants from chromosome 13, leaving 84.\n",
      "Pruned 26715 variants from chromosome 14, leaving 74.\n",
      "Pruned 25915 variants from chromosome 15, leaving 80.\n",
      "Pruned 27688 variants from chromosome 16, leaving 81.\n",
      "Pruned 25119 variants from chromosome 17, leaving 66.\n",
      "Pruned 25142 variants from chromosome 18, leaving 72.\n",
      "Pruned 17413 variants from chromosome 19, leaving 46.\n",
      "Pruned 20537 variants from chromosome 20, leaving 57.\n",
      "Pruned 11932 variants from chromosome 21, leaving 35.\n",
      "Pruned 11812 variants from chromosome 22, leaving 34.\n",
      "Pruning complete.  842761 of 845037 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "845037 variants loaded from .bim file.\n",
      "121 people (89 males, 32 females) loaded from .fam.\n",
      "90 phenotype values loaded from .fam.\n",
      "--extract: 2276 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 121 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999183.\n",
      "2276 variants and 121 people pass filters and QC.\n",
      "Among remaining phenotypes, 77 are cases and 13 are controls.  (31 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "2276 variants loaded from .bim file.\n",
      "121 people (89 males, 32 females) loaded from .fam.\n",
      "90 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 121 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999183.\n",
      "2276 variants and 121 people pass filters and QC.\n",
      "Among remaining phenotypes, 77 are cases and 13 are controls.  (31 phenotypes\n",
      "are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_MDE_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_AMR\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1942516 variants loaded from .bim file.\n",
      "490 people (268 males, 222 females) loaded from .fam.\n",
      "467 phenotype values loaded from .fam.\n",
      "--keep: 490 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 490 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 2752 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.997954.\n",
      "1942516 variants and 490 people pass filters and QC.\n",
      "Among remaining phenotypes, 259 are cases and 208 are controls.  (23 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1860884 variants loaded from .bim file.\n",
      "490 people (268 males, 222 females) loaded from .fam.\n",
      "467 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 490 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998013.\n",
      "32341 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "895018 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "933525 variants and 490 people pass filters and QC.\n",
      "Among remaining phenotypes, 259 are cases and 208 are controls.  (23 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "933525 variants loaded from .bim file.\n",
      "490 people (268 males, 222 females) loaded from .fam.\n",
      "467 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 490 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998312.\n",
      "933525 variants and 490 people pass filters and QC.\n",
      "Among remaining phenotypes, 259 are cases and 208 are controls.  (23 phenotypes\n",
      "are missing.)\n",
      "Pruned 71853 variants from chromosome 1, leaving 328.\n",
      "Pruned 79565 variants from chromosome 2, leaving 352.\n",
      "Pruned 68404 variants from chromosome 3, leaving 300.\n",
      "Pruned 61757 variants from chromosome 4, leaving 281.\n",
      "Pruned 56035 variants from chromosome 5, leaving 248.\n",
      "Pruned 70878 variants from chromosome 6, leaving 262.\n",
      "Pruned 51712 variants from chromosome 7, leaving 258.\n",
      "Pruned 48865 variants from chromosome 8, leaving 219.\n",
      "Pruned 40393 variants from chromosome 9, leaving 206.\n",
      "Pruned 45568 variants from chromosome 10, leaving 223.\n",
      "Pruned 45674 variants from chromosome 11, leaving 198.\n",
      "Pruned 44018 variants from chromosome 12, leaving 194.\n",
      "Pruned 33031 variants from chromosome 13, leaving 174.\n",
      "Pruned 29212 variants from chromosome 14, leaving 125.\n",
      "Pruned 28564 variants from chromosome 15, leaving 140.\n",
      "Pruned 30684 variants from chromosome 16, leaving 154.\n",
      "Pruned 27478 variants from chromosome 17, leaving 151.\n",
      "Pruned 27418 variants from chromosome 18, leaving 163.\n",
      "Pruned 19635 variants from chromosome 19, leaving 125.\n",
      "Pruned 22449 variants from chromosome 20, leaving 131.\n",
      "Pruned 12906 variants from chromosome 21, leaving 71.\n",
      "Pruned 13045 variants from chromosome 22, leaving 78.\n",
      "Pruning complete.  929144 of 933525 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "933525 variants loaded from .bim file.\n",
      "490 people (268 males, 222 females) loaded from .fam.\n",
      "467 phenotype values loaded from .fam.\n",
      "--extract: 4381 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 490 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998303.\n",
      "4381 variants and 490 people pass filters and QC.\n",
      "Among remaining phenotypes, 259 are cases and 208 are controls.  (23 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "4381 variants loaded from .bim file.\n",
      "490 people (268 males, 222 females) loaded from .fam.\n",
      "467 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 490 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998303.\n",
      "4381 variants and 490 people pass filters and QC.\n",
      "Among remaining phenotypes, 259 are cases and 208 are controls.  (23 phenotypes\n",
      "are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AMR_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_AJ\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1897776 variants loaded from .bim file.\n",
      "1444 people (919 males, 525 females) loaded from .fam.\n",
      "1016 phenotype values loaded from .fam.\n",
      "--keep: 1435 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1435 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 10877 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.997697.\n",
      "1897776 variants and 1435 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 370 are controls.  (421\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1817320 variants loaded from .bim file.\n",
      "1435 people (914 males, 521 females) loaded from .fam.\n",
      "1014 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1435 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997785.\n",
      "40937 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "977524 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "798859 variants and 1435 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 370 are controls.  (421\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "798859 variants loaded from .bim file.\n",
      "1435 people (914 males, 521 females) loaded from .fam.\n",
      "1014 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1435 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998305.\n",
      "798859 variants and 1435 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 370 are controls.  (421\n",
      "phenotypes are missing.)\n",
      "Pruned 61093 variants from chromosome 1, leaving 383.\n",
      "Pruned 67709 variants from chromosome 2, leaving 362.\n",
      "Pruned 57836 variants from chromosome 3, leaving 313.\n",
      "Pruned 53214 variants from chromosome 4, leaving 255.\n",
      "Pruned 48135 variants from chromosome 5, leaving 273.\n",
      "Pruned 61999 variants from chromosome 6, leaving 281.\n",
      "Pruned 44114 variants from chromosome 7, leaving 245.\n",
      "Pruned 41852 variants from chromosome 8, leaving 213.\n",
      "Pruned 34601 variants from chromosome 9, leaving 204.\n",
      "Pruned 39036 variants from chromosome 10, leaving 222.\n",
      "Pruned 38936 variants from chromosome 11, leaving 202.\n",
      "Pruned 37668 variants from chromosome 12, leaving 240.\n",
      "Pruned 28313 variants from chromosome 13, leaving 173.\n",
      "Pruned 25403 variants from chromosome 14, leaving 172.\n",
      "Pruned 24251 variants from chromosome 15, leaving 169.\n",
      "Pruned 25905 variants from chromosome 16, leaving 161.\n",
      "Pruned 23123 variants from chromosome 17, leaving 178.\n",
      "Pruned 23535 variants from chromosome 18, leaving 161.\n",
      "Pruned 16175 variants from chromosome 19, leaving 146.\n",
      "Pruned 19138 variants from chromosome 20, leaving 159.\n",
      "Pruned 11295 variants from chromosome 21, leaving 103.\n",
      "Pruned 10808 variants from chromosome 22, leaving 105.\n",
      "Pruning complete.  794139 of 798859 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "798859 variants loaded from .bim file.\n",
      "1435 people (914 males, 521 females) loaded from .fam.\n",
      "1014 phenotype values loaded from .fam.\n",
      "--extract: 4720 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1435 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998053.\n",
      "4720 variants and 1435 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 370 are controls.  (421\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "4720 variants loaded from .bim file.\n",
      "1435 people (914 males, 521 females) loaded from .fam.\n",
      "1014 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 1435 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998053.\n",
      "4720 variants and 1435 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 370 are controls.  (421\n",
      "phenotypes are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AJ_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_AAC\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1902738 variants loaded from .bim file.\n",
      "1360 people (551 males, 809 females) loaded from .fam.\n",
      "1347 phenotype values loaded from .fam.\n",
      "--keep: 1305 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1305 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 7863 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.997688.\n",
      "1902738 variants and 1305 people pass filters and QC.\n",
      "Among remaining phenotypes, 170 are cases and 1122 are controls.  (13\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1822250 variants loaded from .bim file.\n",
      "1305 people (514 males, 791 females) loaded from .fam.\n",
      "1292 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1305 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997758.\n",
      "37157 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "741881 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1043212 variants and 1305 people pass filters and QC.\n",
      "Among remaining phenotypes, 170 are cases and 1122 are controls.  (13\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1043212 variants loaded from .bim file.\n",
      "1305 people (514 males, 791 females) loaded from .fam.\n",
      "1292 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1305 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998117.\n",
      "1043212 variants and 1305 people pass filters and QC.\n",
      "Among remaining phenotypes, 170 are cases and 1122 are controls.  (13\n",
      "phenotypes are missing.)\n",
      "Pruned 79567 variants from chromosome 1, leaving 1061.\n",
      "Pruned 90353 variants from chromosome 2, leaving 1012.\n",
      "Pruned 76913 variants from chromosome 3, leaving 901.\n",
      "Pruned 69222 variants from chromosome 4, leaving 849.\n",
      "Pruned 62502 variants from chromosome 5, leaving 818.\n",
      "Pruned 75648 variants from chromosome 6, leaving 807.\n",
      "Pruned 57476 variants from chromosome 7, leaving 721.\n",
      "Pruned 53951 variants from chromosome 8, leaving 627.\n",
      "Pruned 44212 variants from chromosome 9, leaving 653.\n",
      "Pruned 50003 variants from chromosome 10, leaving 677.\n",
      "Pruned 51163 variants from chromosome 11, leaving 622.\n",
      "Pruned 48395 variants from chromosome 12, leaving 646.\n",
      "Pruned 36482 variants from chromosome 13, leaving 494.\n",
      "Pruned 32488 variants from chromosome 14, leaving 467.\n",
      "Pruned 31370 variants from chromosome 15, leaving 504.\n",
      "Pruned 33944 variants from chromosome 16, leaving 550.\n",
      "Pruned 30558 variants from chromosome 17, leaving 509.\n",
      "Pruned 30598 variants from chromosome 18, leaving 508.\n",
      "Pruned 21486 variants from chromosome 19, leaving 420.\n",
      "Pruned 24738 variants from chromosome 20, leaving 428.\n",
      "Pruned 13852 variants from chromosome 21, leaving 276.\n",
      "Pruned 14451 variants from chromosome 22, leaving 290.\n",
      "Pruning complete.  1029372 of 1043212 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1043212 variants loaded from .bim file.\n",
      "1305 people (514 males, 791 females) loaded from .fam.\n",
      "1292 phenotype values loaded from .fam.\n",
      "--extract: 13840 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1305 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997987.\n",
      "13840 variants and 1305 people pass filters and QC.\n",
      "Among remaining phenotypes, 170 are cases and 1122 are controls.  (13\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "13840 variants loaded from .bim file.\n",
      "1305 people (514 males, 791 females) loaded from .fam.\n",
      "1292 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 1305 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997987.\n",
      "13840 variants and 1305 people pass filters and QC.\n",
      "Among remaining phenotypes, 170 are cases and 1122 are controls.  (13\n",
      "phenotypes are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AAC_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_EUR\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1707223 variants loaded from .bim file.\n",
      "15062 people (9303 males, 5759 females) loaded from .fam.\n",
      "13587 phenotype values loaded from .fam.\n",
      "--keep: 13612 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 13612 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 82029 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.998189.\n",
      "1707223 variants and 13612 people pass filters and QC.\n",
      "Among remaining phenotypes, 8066 are cases and 4371 are controls.  (1175\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1632537 variants loaded from .bim file.\n",
      "13612 people (8412 males, 5200 females) loaded from .fam.\n",
      "12437 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 13612 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99827.\n",
      "21434 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 1 variant removed due to Hardy-Weinberg exact test.\n",
      "884230 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "726872 variants and 13612 people pass filters and QC.\n",
      "Among remaining phenotypes, 8066 are cases and 4371 are controls.  (1175\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "726872 variants loaded from .bim file.\n",
      "13612 people (8412 males, 5200 females) loaded from .fam.\n",
      "12437 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 13612 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99849.\n",
      "726872 variants and 13612 people pass filters and QC.\n",
      "Among remaining phenotypes, 8066 are cases and 4371 are controls.  (1175\n",
      "phenotypes are missing.)\n",
      "Pruned 55053 variants from chromosome 1, leaving 1151.\n",
      "Pruned 61465 variants from chromosome 2, leaving 1048.\n",
      "Pruned 52769 variants from chromosome 3, leaving 920.\n",
      "Pruned 48449 variants from chromosome 4, leaving 883.\n",
      "Pruned 44315 variants from chromosome 5, leaving 834.\n",
      "Pruned 53816 variants from chromosome 6, leaving 825.\n",
      "Pruned 39621 variants from chromosome 7, leaving 785.\n",
      "Pruned 38209 variants from chromosome 8, leaving 646.\n",
      "Pruned 30982 variants from chromosome 9, leaving 670.\n",
      "Pruned 34873 variants from chromosome 10, leaving 725.\n",
      "Pruned 34988 variants from chromosome 11, leaving 691.\n",
      "Pruned 33961 variants from chromosome 12, leaving 695.\n",
      "Pruned 25740 variants from chromosome 13, leaving 539.\n",
      "Pruned 23028 variants from chromosome 14, leaving 490.\n",
      "Pruned 21606 variants from chromosome 15, leaving 534.\n",
      "Pruned 22726 variants from chromosome 16, leaving 550.\n",
      "Pruned 19880 variants from chromosome 17, leaving 553.\n",
      "Pruned 21446 variants from chromosome 18, leaving 510.\n",
      "Pruned 13167 variants from chromosome 19, leaving 492.\n",
      "Pruned 16982 variants from chromosome 20, leaving 478.\n",
      "Pruned 10078 variants from chromosome 21, leaving 311.\n",
      "Pruned 9089 variants from chromosome 22, leaving 299.\n",
      "Pruning complete.  712243 of 726872 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "726872 variants loaded from .bim file.\n",
      "13612 people (8412 males, 5200 females) loaded from .fam.\n",
      "12437 phenotype values loaded from .fam.\n",
      "--extract: 14629 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 13612 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998345.\n",
      "14629 variants and 13612 people pass filters and QC.\n",
      "Among remaining phenotypes, 8066 are cases and 4371 are controls.  (1175\n",
      "phenotypes are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "14629 variants loaded from .bim file.\n",
      "13612 people (8412 males, 5200 females) loaded from .fam.\n",
      "12437 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 13612 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998345.\n",
      "14629 variants and 13612 people pass filters and QC.\n",
      "Among remaining phenotypes, 8066 are cases and 4371 are controls.  (1175\n",
      "phenotypes are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EUR_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_EAS\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1789739 variants loaded from .bim file.\n",
      "1901 people (1564 males, 337 females) loaded from .fam.\n",
      "1890 phenotype values loaded from .fam.\n",
      "--keep: 155 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 155 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 335 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.998559.\n",
      "1789739 variants and 155 people pass filters and QC.\n",
      "Among remaining phenotypes, 111 are cases and 33 are controls.  (11 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1724110 variants loaded from .bim file.\n",
      "155 people (103 males, 52 females) loaded from .fam.\n",
      "144 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 155 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998569.\n",
      "19007 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 1 variant removed due to Hardy-Weinberg exact test.\n",
      "1031368 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "673734 variants and 155 people pass filters and QC.\n",
      "Among remaining phenotypes, 111 are cases and 33 are controls.  (11 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "673734 variants loaded from .bim file.\n",
      "155 people (103 males, 52 females) loaded from .fam.\n",
      "144 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 155 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999019.\n",
      "673734 variants and 155 people pass filters and QC.\n",
      "Among remaining phenotypes, 111 are cases and 33 are controls.  (11 phenotypes\n",
      "are missing.)\n",
      "Pruned 51979 variants from chromosome 1, leaving 175.\n",
      "Pruned 57199 variants from chromosome 2, leaving 176.\n",
      "Pruned 49917 variants from chromosome 3, leaving 168.\n",
      "Pruned 44958 variants from chromosome 4, leaving 143.\n",
      "Pruned 40875 variants from chromosome 5, leaving 132.\n",
      "Pruned 52421 variants from chromosome 6, leaving 148.\n",
      "Pruned 37323 variants from chromosome 7, leaving 125.\n",
      "Pruned 34889 variants from chromosome 8, leaving 116.\n",
      "Pruned 29246 variants from chromosome 9, leaving 104.\n",
      "Pruned 33064 variants from chromosome 10, leaving 120.\n",
      "Pruned 32614 variants from chromosome 11, leaving 115.\n",
      "Pruned 32178 variants from chromosome 12, leaving 113.\n",
      "Pruned 23782 variants from chromosome 13, leaving 80.\n",
      "Pruned 21401 variants from chromosome 14, leaving 71.\n",
      "Pruned 21015 variants from chromosome 15, leaving 68.\n",
      "Pruned 21769 variants from chromosome 16, leaving 77.\n",
      "Pruned 18911 variants from chromosome 17, leaving 67.\n",
      "Pruned 20269 variants from chromosome 18, leaving 66.\n",
      "Pruned 13469 variants from chromosome 19, leaving 47.\n",
      "Pruned 16009 variants from chromosome 20, leaving 63.\n",
      "Pruned 9333 variants from chromosome 21, leaving 35.\n",
      "Pruned 8869 variants from chromosome 22, leaving 35.\n",
      "Pruning complete.  671490 of 673734 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "673734 variants loaded from .bim file.\n",
      "155 people (103 males, 52 females) loaded from .fam.\n",
      "144 phenotype values loaded from .fam.\n",
      "--extract: 2244 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 155 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999273.\n",
      "2244 variants and 155 people pass filters and QC.\n",
      "Among remaining phenotypes, 111 are cases and 33 are controls.  (11 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "2244 variants loaded from .bim file.\n",
      "155 people (103 males, 52 females) loaded from .fam.\n",
      "144 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 155 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999273.\n",
      "2244 variants and 155 people pass filters and QC.\n",
      "Among remaining phenotypes, 111 are cases and 33 are controls.  (11 phenotypes\n",
      "are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_EAS_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_CAS\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1943048 variants loaded from .bim file.\n",
      "322 people (128 males, 194 females) loaded from .fam.\n",
      "319 phenotype values loaded from .fam.\n",
      "--keep: 314 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 314 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 1206 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.998143.\n",
      "1943048 variants and 314 people pass filters and QC.\n",
      "Among remaining phenotypes, 115 are cases and 196 are controls.  (3 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1861436 variants loaded from .bim file.\n",
      "314 people (125 males, 189 females) loaded from .fam.\n",
      "311 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 314 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998175.\n",
      "30293 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "986409 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "844734 variants and 314 people pass filters and QC.\n",
      "Among remaining phenotypes, 115 are cases and 196 are controls.  (3 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "844734 variants loaded from .bim file.\n",
      "314 people (125 males, 189 females) loaded from .fam.\n",
      "311 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 314 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998519.\n",
      "844734 variants and 314 people pass filters and QC.\n",
      "Among remaining phenotypes, 115 are cases and 196 are controls.  (3 phenotypes\n",
      "are missing.)\n",
      "Pruned 64728 variants from chromosome 1, leaving 261.\n",
      "Pruned 71274 variants from chromosome 2, leaving 297.\n",
      "Pruned 61046 variants from chromosome 3, leaving 256.\n",
      "Pruned 56156 variants from chromosome 4, leaving 272.\n",
      "Pruned 50603 variants from chromosome 5, leaving 235.\n",
      "Pruned 66232 variants from chromosome 6, leaving 237.\n",
      "Pruned 46869 variants from chromosome 7, leaving 207.\n",
      "Pruned 43828 variants from chromosome 8, leaving 188.\n",
      "Pruned 36471 variants from chromosome 9, leaving 179.\n",
      "Pruned 41366 variants from chromosome 10, leaving 188.\n",
      "Pruned 41237 variants from chromosome 11, leaving 188.\n",
      "Pruned 40035 variants from chromosome 12, leaving 182.\n",
      "Pruned 29709 variants from chromosome 13, leaving 143.\n",
      "Pruned 26816 variants from chromosome 14, leaving 134.\n",
      "Pruned 25899 variants from chromosome 15, leaving 123.\n",
      "Pruned 27565 variants from chromosome 16, leaving 124.\n",
      "Pruned 24717 variants from chromosome 17, leaving 129.\n",
      "Pruned 24937 variants from chromosome 18, leaving 137.\n",
      "Pruned 17739 variants from chromosome 19, leaving 102.\n",
      "Pruned 20100 variants from chromosome 20, leaving 99.\n",
      "Pruned 11824 variants from chromosome 21, leaving 63.\n",
      "Pruned 11776 variants from chromosome 22, leaving 63.\n",
      "Pruning complete.  840927 of 844734 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "844734 variants loaded from .bim file.\n",
      "314 people (125 males, 189 females) loaded from .fam.\n",
      "311 phenotype values loaded from .fam.\n",
      "--extract: 3807 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 314 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998604.\n",
      "3807 variants and 314 people pass filters and QC.\n",
      "Among remaining phenotypes, 115 are cases and 196 are controls.  (3 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "3807 variants loaded from .bim file.\n",
      "314 people (125 males, 189 females) loaded from .fam.\n",
      "311 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 314 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998604.\n",
      "3807 variants and 314 people pass filters and QC.\n",
      "Among remaining phenotypes, 115 are cases and 196 are controls.  (3 phenotypes\n",
      "are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_CAS_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_SAS\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1946213 variants loaded from .bim file.\n",
      "109 people (68 males, 41 females) loaded from .fam.\n",
      "101 phenotype values loaded from .fam.\n",
      "--keep: 80 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 80 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 411 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.998506.\n",
      "1946213 variants and 80 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 18 are controls.  (6 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1864842 variants loaded from .bim file.\n",
      "80 people (50 males, 30 females) loaded from .fam.\n",
      "74 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 80 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998552.\n",
      "42205 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "1008082 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "814555 variants and 80 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 18 are controls.  (6 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "814555 variants loaded from .bim file.\n",
      "80 people (50 males, 30 females) loaded from .fam.\n",
      "74 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 80 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999236.\n",
      "814555 variants and 80 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 18 are controls.  (6 phenotypes\n",
      "are missing.)\n",
      "Pruned 62688 variants from chromosome 1, leaving 149.\n",
      "Pruned 68899 variants from chromosome 2, leaving 151.\n",
      "Pruned 59011 variants from chromosome 3, leaving 132.\n",
      "Pruned 54213 variants from chromosome 4, leaving 122.\n",
      "Pruned 49120 variants from chromosome 5, leaving 110.\n",
      "Pruned 63332 variants from chromosome 6, leaving 119.\n",
      "Pruned 45364 variants from chromosome 7, leaving 93.\n",
      "Pruned 42495 variants from chromosome 8, leaving 97.\n",
      "Pruned 35348 variants from chromosome 9, leaving 75.\n",
      "Pruned 40108 variants from chromosome 10, leaving 91.\n",
      "Pruned 39588 variants from chromosome 11, leaving 96.\n",
      "Pruned 38801 variants from chromosome 12, leaving 94.\n",
      "Pruned 28784 variants from chromosome 13, leaving 70.\n",
      "Pruned 25866 variants from chromosome 14, leaving 66.\n",
      "Pruned 25005 variants from chromosome 15, leaving 55.\n",
      "Pruned 26638 variants from chromosome 16, leaving 70.\n",
      "Pruned 23933 variants from chromosome 17, leaving 57.\n",
      "Pruned 24191 variants from chromosome 18, leaving 57.\n",
      "Pruned 16969 variants from chromosome 19, leaving 43.\n",
      "Pruned 19538 variants from chromosome 20, leaving 45.\n",
      "Pruned 11476 variants from chromosome 21, leaving 30.\n",
      "Pruned 11339 variants from chromosome 22, leaving 27.\n",
      "Pruning complete.  812706 of 814555 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "814555 variants loaded from .bim file.\n",
      "80 people (50 males, 30 females) loaded from .fam.\n",
      "74 phenotype values loaded from .fam.\n",
      "--extract: 1849 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 80 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999405.\n",
      "1849 variants and 80 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 18 are controls.  (6 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1849 variants loaded from .bim file.\n",
      "80 people (50 males, 30 females) loaded from .fam.\n",
      "74 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 80 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999405.\n",
      "1849 variants and 80 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 18 are controls.  (6 phenotypes\n",
      "are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_SAS_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_FIN\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1932019 variants loaded from .bim file.\n",
      "12 people (8 males, 4 females) loaded from .fam.\n",
      "10 phenotype values loaded from .fam.\n",
      "--keep: 12 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 12 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 23 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.999999.\n",
      "1932019 variants and 12 people pass filters and QC.\n",
      "Among remaining phenotypes, 9 are cases and 1 is a control.  (2 phenotypes are\n",
      "missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1851676 variants loaded from .bim file.\n",
      "12 people (8 males, 4 females) loaded from .fam.\n",
      "10 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 12 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "1168734 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "682942 variants and 12 people pass filters and QC.\n",
      "Among remaining phenotypes, 9 are cases and 1 is a control.  (2 phenotypes are\n",
      "missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "682942 variants loaded from .bim file.\n",
      "12 people (8 males, 4 females) loaded from .fam.\n",
      "10 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 12 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "682942 variants and 12 people pass filters and QC.\n",
      "Among remaining phenotypes, 9 are cases and 1 is a control.  (2 phenotypes are\n",
      "missing.)\n",
      "Pruned 52263 variants from chromosome 1, leaving 99.\n",
      "Pruned 57793 variants from chromosome 2, leaving 109.\n",
      "Pruned 49002 variants from chromosome 3, leaving 96.\n",
      "Pruned 44796 variants from chromosome 4, leaving 93.\n",
      "Pruned 41065 variants from chromosome 5, leaving 78.\n",
      "Pruned 53285 variants from chromosome 6, leaving 92.\n",
      "Pruned 38175 variants from chromosome 7, leaving 73.\n",
      "Pruned 35273 variants from chromosome 8, leaving 73.\n",
      "Pruned 29650 variants from chromosome 9, leaving 59.\n",
      "Pruned 33504 variants from chromosome 10, leaving 64.\n",
      "Pruned 33630 variants from chromosome 11, leaving 64.\n",
      "Pruned 32676 variants from chromosome 12, leaving 62.\n",
      "Pruned 24097 variants from chromosome 13, leaving 51.\n",
      "Pruned 21266 variants from chromosome 14, leaving 41.\n",
      "Pruned 20870 variants from chromosome 15, leaving 41.\n",
      "Pruned 22759 variants from chromosome 16, leaving 44.\n",
      "Pruned 20470 variants from chromosome 17, leaving 43.\n",
      "Pruned 19889 variants from chromosome 18, leaving 39.\n",
      "Pruned 14957 variants from chromosome 19, leaving 34.\n",
      "Pruned 16491 variants from chromosome 20, leaving 35.\n",
      "Pruned 9709 variants from chromosome 21, leaving 20.\n",
      "Pruned 9991 variants from chromosome 22, leaving 21.\n",
      "Pruning complete.  681611 of 682942 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "682942 variants loaded from .bim file.\n",
      "12 people (8 males, 4 females) loaded from .fam.\n",
      "10 phenotype values loaded from .fam.\n",
      "--extract: 1331 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 12 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1331 variants and 12 people pass filters and QC.\n",
      "Among remaining phenotypes, 9 are cases and 1 is a control.  (2 phenotypes are\n",
      "missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1331 variants loaded from .bim file.\n",
      "12 people (8 males, 4 females) loaded from .fam.\n",
      "10 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 12 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1331 variants and 12 people pass filters and QC.\n",
      "Among remaining phenotypes, 9 are cases and 1 is a control.  (2 phenotypes are\n",
      "missing.)\n",
      "Relationship matrix calculation complete.\n",
      "Warning: calculating 12 PCs, since there are only 12 samples.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_FIN_clean_pcs.eigenvec\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_AFR\n",
      "  --keep /data/GP2/raw_genotypes/cnvs/release4/release4.samples\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1787849 variants loaded from .bim file.\n",
      "1775 people (1098 males, 677 females) loaded from .fam.\n",
      "1767 phenotype values loaded from .fam.\n",
      "--keep: 405 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 405 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 2089 het. haploid genotypes present (see\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR.hh ); many\n",
      "commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.997644.\n",
      "1787849 variants and 405 people pass filters and QC.\n",
      "Among remaining phenotypes, 39 are cases and 362 are controls.  (4 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR\n",
      "  --geno 0.02\n",
      "  --hwe 5e-6\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "1714202 variants loaded from .bim file.\n",
      "405 people (156 males, 249 females) loaded from .fam.\n",
      "401 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 405 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997712.\n",
      "22435 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 1 variant removed due to Hardy-Weinberg exact test.\n",
      "772652 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "919114 variants and 405 people pass filters and QC.\n",
      "Among remaining phenotypes, 39 are cases and 362 are controls.  (4 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe.bed\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe.bim\n",
      "+\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe_prune.log.\n",
      "Options in effect:\n",
      "  --autosome\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe\n",
      "  --indep-pairwise 1000 10 0.01\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe_prune\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "919114 variants loaded from .bim file.\n",
      "405 people (156 males, 249 females) loaded from .fam.\n",
      "401 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 405 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997578.\n",
      "919114 variants and 405 people pass filters and QC.\n",
      "Among remaining phenotypes, 39 are cases and 362 are controls.  (4 phenotypes\n",
      "are missing.)\n",
      "Pruned 70582 variants from chromosome 1, leaving 531.\n",
      "Pruned 80253 variants from chromosome 2, leaving 540.\n",
      "Pruned 68356 variants from chromosome 3, leaving 488.\n",
      "Pruned 61375 variants from chromosome 4, leaving 448.\n",
      "Pruned 55535 variants from chromosome 5, leaving 428.\n",
      "Pruned 67030 variants from chromosome 6, leaving 384.\n",
      "Pruned 50735 variants from chromosome 7, leaving 390.\n",
      "Pruned 47985 variants from chromosome 8, leaving 325.\n",
      "Pruned 39074 variants from chromosome 9, leaving 312.\n",
      "Pruned 44285 variants from chromosome 10, leaving 340.\n",
      "Pruned 44813 variants from chromosome 11, leaving 346.\n",
      "Pruned 42749 variants from chromosome 12, leaving 308.\n",
      "Pruned 32456 variants from chromosome 13, leaving 249.\n",
      "Pruned 28927 variants from chromosome 14, leaving 224.\n",
      "Pruned 28210 variants from chromosome 15, leaving 230.\n",
      "Pruned 29779 variants from chromosome 16, leaving 229.\n",
      "Pruned 26936 variants from chromosome 17, leaving 218.\n",
      "Pruned 27483 variants from chromosome 18, leaving 216.\n",
      "Pruned 18746 variants from chromosome 19, leaving 171.\n",
      "Pruned 22087 variants from chromosome 20, leaving 199.\n",
      "Pruned 12332 variants from chromosome 21, leaving 93.\n",
      "Pruned 12602 variants from chromosome 22, leaving 115.\n",
      "Pruning complete.  912330 of 919114 variants removed.\n",
      "Marker lists written to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe_prune.prune.in\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe_prune.prune.out\n",
      ".\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe\n",
      "  --extract /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_maf_geno_hwe_prune.prune.in\n",
      "  --make-bed\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "919114 variants loaded from .bim file.\n",
      "405 people (156 males, 249 females) loaded from .fam.\n",
      "401 phenotype values loaded from .fam.\n",
      "--extract: 6784 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 405 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996881.\n",
      "6784 variants and 405 people pass filters and QC.\n",
      "Among remaining phenotypes, 39 are cases and 362 are controls.  (4 phenotypes\n",
      "are missing.)\n",
      "--make-bed to\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean.bed +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean.bim +\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "[-] Unloading plink  3.6-alpha \n",
      "[+] Loading plink  1.9  on cn4285 \n",
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) plink/3.6-alpha => plink/1.9\n",
      "\n",
      "PLINK v1.90b3.36 64-bit (31 Mar 2016)      https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean_pcs.log.\n",
      "Options in effect:\n",
      "  --bfile /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean\n",
      "  --out /data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean_pcs\n",
      "  --pca\n",
      "\n",
      "515537 MB RAM detected; reserving 257768 MB for main workspace.\n",
      "6784 variants loaded from .bim file.\n",
      "405 people (156 males, 249 females) loaded from .fam.\n",
      "401 phenotype values loaded from .fam.\n",
      "Using up to 127 threads (change this with --threads).\n",
      "Before main variant filters, 405 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.996881.\n",
      "6784 variants and 405 people pass filters and QC.\n",
      "Among remaining phenotypes, 39 are cases and 362 are controls.  (4 phenotypes\n",
      "are missing.)\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean_pcs.eigenval\n",
      "and\n",
      "/data/GP2/raw_genotypes/cnvs/release4/GP2_round4_JAN_24_2023_AFR_clean_pcs.eigenvec\n",
      ".\n"
     ]
    }
   ],
   "source": [
    "# snp QC for cnvs\n",
    "# for label in ['EUR']:\n",
    "for label in ancestry_labels:\n",
    "    geno_name = f'GP2_round4_JAN_24_2023_{label}'\n",
    "    out_path = f'{cnv_path}/genotypes/{geno_name}'\n",
    "    geno_path = f'/data/GP2/quality_control/release4/genotype_qc/{geno_name}'\n",
    "\n",
    "    cmd1 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {geno_path} \\\n",
    "--keep {cnv_path}/genotypes/release4.samples \\\n",
    "--make-bed \\\n",
    "--out {out_path}'\n",
    "\n",
    "    cmd2 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {out_path} \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.02 \\\n",
    "--hwe 5e-6 \\\n",
    "--autosome \\\n",
    "--make-bed \\\n",
    "--out {out_path}_maf_geno_hwe'\n",
    "\n",
    "    cmd3 = f'\\\n",
    "module load plink/1.9; \\\n",
    "plink \\\n",
    "--bfile {out_path}_maf_geno_hwe \\\n",
    "--indep-pairwise 1000 10 0.01 \\\n",
    "--autosome \\\n",
    "--out {out_path}_maf_geno_hwe_prune'\n",
    "    \n",
    "    cmd4 = f'\\\n",
    "module load plink/1.9; \\\n",
    "plink \\\n",
    "--bfile {out_path}_maf_geno_hwe \\\n",
    "--extract {out_path}_maf_geno_hwe_prune.prune.in \\\n",
    "--make-bed \\\n",
    "--out {out_path}_clean'\n",
    "\n",
    "    cmd5 = f'\\\n",
    "module load plink/1.9; \\\n",
    "plink \\\n",
    "--bfile {out_path}_clean \\\n",
    "--pca \\\n",
    "--out {out_path}_clean_pcs'\n",
    "\n",
    "    cmds = [cmd1, cmd2, cmd3, cmd4, cmd5]\n",
    "\n",
    "    for cmd in cmds:\n",
    "        !{cmd}\n",
    "\n",
    "    pcs = pd.read_csv(f'{out_path}_clean_pcs.eigenvec', sep='\\s+')\n",
    "    pc_num = pcs.iloc[:, 2:].shape[1]\n",
    "    pc_names = ['FID','GP2sampleID'] + [f'PC{i}' for i in range(1, pc_num+1)]\n",
    "    pcs.columns = pc_names\n",
    "\n",
    "    cov = pcs.merge(release_covars, on=['FID','GP2sampleID'], how='left')\n",
    "    cov.age.fillna(cov.age.mean(), inplace=True)\n",
    "    cov.age_of_onset.fillna(cov.age_of_onset.mean(), inplace=True)\n",
    "    cov.sex_for_qc.fillna(cov.sex_for_qc.median(), inplace=True)\n",
    "    cov.rename(columns={'GP2sampleID':'sampleid','sex_for_qc':'sex'}, inplace=True)\n",
    "    cov.to_csv(f'{out_path}_clean.cov', sep='\\t', header=True, index=False)\n",
    "\n",
    "    samples = cov.merge(release_key[['GP2sampleID','IID']], left_on='sampleid', right_on='GP2sampleID', how='left')\n",
    "    samples['IID'].to_csv(f'{out_path}_clean_barcode.samples', header=False, index=False)\n",
    "\n",
    "    bim = pd.read_csv(f'{out_path}_clean.bim', sep='\\s+', header=None, names=['chr','id','pos','bp','a1','a2'], usecols=['id'])\n",
    "    bim.to_csv(f'{out_path}_clean.snps', sep='\\t', header=False, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c0a75eb9-9d5f-4cfd-b81a-f97ac19841f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MDE | total_n: 90 | n_cases: 77 | n_controls: 13 | n_snps: 2276\n",
      "\n",
      "AMR | total_n: 467 | n_cases: 259 | n_controls: 208 | n_snps: 4381\n",
      "\n",
      "AJ | total_n: 1014 | n_cases: 644 | n_controls: 370 | n_snps: 4720\n",
      "\n",
      "AAC | total_n: 1292 | n_cases: 170 | n_controls: 1122 | n_snps: 13840\n",
      "\n",
      "EUR | total_n: 12437 | n_cases: 8066 | n_controls: 4371 | n_snps: 14629\n",
      "\n",
      "EAS | total_n: 144 | n_cases: 111 | n_controls: 33 | n_snps: 2244\n",
      "\n",
      "CAS | total_n: 311 | n_cases: 115 | n_controls: 196 | n_snps: 3807\n",
      "\n",
      "SAS | total_n: 74 | n_cases: 56 | n_controls: 18 | n_snps: 1849\n",
      "\n",
      "FIN | total_n: 10 | n_cases: 9 | n_controls: 1 | n_snps: 1331\n",
      "\n",
      "AFR | total_n: 401 | n_cases: 39 | n_controls: 362 | n_snps: 6784\n",
      "\n"
     ]
    }
   ],
   "source": [
    "total_sample_n = 0\n",
    "for label in ancestry_labels:\n",
    "# for label in ['EUR']:\n",
    "    out_name = f'GP2_round4_JAN_24_2023_{label}'\n",
    "    out_path = f'{cnv_path}/genotypes/{out_name}'\n",
    "    fam = pd.read_csv(f'{out_path}_clean.fam', sep='\\s+', header=None, names=['fid','iid','pat','mat','sex','pheno'])\n",
    "    pheno_counts = fam.pheno.value_counts()\n",
    "    ncases = pheno_counts[2]\n",
    "    ncontrols = pheno_counts[1]\n",
    "    ntotal = ncases + ncontrols\n",
    "    total_sample_n += ntotal\n",
    "    nsnps = !cat {out_path}_clean.bim | wc -l\n",
    "    print(f'{label} | total_n: {ntotal} | n_cases: {ncases} | n_controls: {ncontrols} | n_snps: {int(nsnps[0])}')\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "94fa73a0-2dfb-4213-b1dc-38e6450d7520",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16240"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_sample_n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d716faf8-48f8-4a74-9c90-8046a49bbb5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "with open(f'{swarm_scripts_dir}/cnvs.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "    # for label in ['EUR']:\n",
    "        out_name = f'GP2_round4_JAN_24_2023_{label}'\n",
    "        geno_path = f'{cnv_path}/genotypes/{out_name}_clean'\n",
    "        bim = f'{geno_path}.bim'\n",
    "        samples = pd.read_csv(f'{geno_path}_barcode.samples', header=None, names=['IID'])\n",
    "        label_dir = f'{cnv_path}/cnv_calls/{label}'\n",
    "        os.makedirs(label_dir, exist_ok=True)\n",
    "        for sample in samples.IID.unique():\n",
    "            code = sample.split('_')[0]\n",
    "            for chrom in chroms:\n",
    "            \n",
    "                mfile = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "                \n",
    "                cnv_out = f'{label_dir}/CNV_{label}_{sample}_chr{chrom}.csv'\n",
    "                intervals = f'/data/GP2/utils/ref_dir/glist_hg38_chr{chrom}.csv'\n",
    "\n",
    "                cmd = f'\\\n",
    "python run_cnv_pipeline.py \\\n",
    "--metrics {mfile} \\\n",
    "--bim {bim} \\\n",
    "--out_path {cnv_out} \\\n",
    "--intervals {intervals} \\\n",
    "--min_variants 10 \\\n",
    "--kb_window 250 \\\n",
    "--min_gentrain 0.2'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fa7199d9-3b87-4d27-832a-28f98b0dd800",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58290655\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnvs.swarm -g 4 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --module python/3.8 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2e9e6292-58c4-4565-8074-cfa5a39f883a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MDE\n",
      "120\n",
      "\n",
      "\n",
      "AMR\n",
      "489\n",
      "\n",
      "\n",
      "AJ\n",
      "1434\n",
      "\n",
      "\n",
      "AAC\n",
      "1304\n",
      "\n",
      "\n",
      "EUR\n",
      "13611\n",
      "\n",
      "\n",
      "EAS\n",
      "154\n",
      "\n",
      "\n",
      "CAS\n",
      "313\n",
      "\n",
      "\n",
      "SAS\n",
      "79\n",
      "\n",
      "\n",
      "FIN\n",
      "11\n",
      "\n",
      "\n",
      "AFR\n",
      "404\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "missing_cnvs = list()\n",
    "missing_metrics = list()\n",
    "for label in ancestry_labels:\n",
    "    # for label in ['EUR']:\n",
    "    out_name = f'GP2_round4_JAN_24_2023_{label}'\n",
    "    label_dir = f'{cnv_path}/cnv_calls/{label}'\n",
    "    geno_path = f'{cnv_path}/genotypes/{out_name}_clean'\n",
    "    samples = pd.read_csv(f'{geno_path}_barcode.samples', header=None, names=['IID'])\n",
    "    \n",
    "    print(label)\n",
    "    print(samples.shape[0])\n",
    "    print()\n",
    "    print()\n",
    "    \n",
    "    \n",
    "    for sample in samples.IID.unique():\n",
    "        code = sample.split('_')[0]\n",
    "        for chrom in chroms:\n",
    "            \n",
    "            mfile = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "            cnv_out = f'{label_dir}/CNV_{label}_{sample}_chr{chrom}.csv'\n",
    "            \n",
    "            if os.path.isfile(mfile):\n",
    "                pass\n",
    "            else:\n",
    "                missing_metrics.append(mfile)\n",
    "                \n",
    "            if os.path.isfile(cnv_out):\n",
    "                pass\n",
    "            else:\n",
    "                missing_cnvs.append(cnv_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "75f31342-5241-48f9-bdf5-e73b1c46798d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# write dosages per chromosome per ancestry per cnv-type\n",
    "\n",
    "with open(f'{swarm_scripts_dir}/cnv_dosages.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        label_dir = f'{cnv_path}/dosages/{label}'\n",
    "        os.makedirs(label_dir, exist_ok=True)\n",
    "        for chrom in chroms:\n",
    "\n",
    "            cnv_files_list = glob.glob(f'{cnv_path}/cnv_calls/{label}/CNV_{label}_*_chr{chrom}.csv')\n",
    "            cnv_files_df = pd.DataFrame({'filename': cnv_files_list})\n",
    "            cnv_files_df.to_csv(f'{label_dir}/CNV_{label}_chr{chrom}_files.csv', header=False, index=False)\n",
    "\n",
    "            for cnv_type in cnv_types:\n",
    "                \n",
    "                cmd = f'\\\n",
    "python /data/vitaled2/CNVs/run_cnv_dosage_pipeline.py \\\n",
    "--files {label_dir}/CNV_{label}_chr{chrom}_files.csv \\\n",
    "--label {label} \\\n",
    "--chrom {chrom} \\\n",
    "--cnv_type {cnv_type} \\\n",
    "--out_path {label_dir}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "5bce1bd2-a140-47d7-93cb-f6515adea344",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58322813\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_dosages.swarm -g 32 -t 16 --logdir {swarm_scripts_dir}/logs --module python/3.8 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "ab5d95cc-a362-4abe-8e1f-3ea30fd5041d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "label = 'EUR'\n",
    "label_dir = f'/data/GP2/raw_genotypes/cnvs/release4/cnv_calls/{label}'\n",
    "chrom = '1'\n",
    "cnv_files = f'{cnv_path}/dosages/{label}/CNV_{label}_chr{chrom}_files.csv' \n",
    "cnv_type = 'PERCENT_BAF_INSERTION'\n",
    "\n",
    "cnv_files_df = pd.read_csv(cnv_files, header=None)\n",
    "files_list = cnv_files_df.loc[:,0]\n",
    "\n",
    "dosage_ = pd.DataFrame()\n",
    "\n",
    "for sample_chrom_file in files_list:\n",
    "\n",
    "    sample = sample_chrom_file.split('/')[-1].replace(f'CNV_{label}_','').replace(f'_chr{chrom}.csv','')\n",
    "    cnvs = pd.read_csv(sample_chrom_file, usecols=['INTERVAL', cnv_type, 'NUM_VARIANTS'])\n",
    "    cnvs.loc[:,'sampleid'] = sample\n",
    "    cnvs_final = cnvs.loc[~cnvs[cnv_type].isna()]\n",
    "    dosage = cnvs_final.loc[:,[cnv_type,'INTERVAL','sampleid']]\n",
    "    dosage_pivot = dosage.pivot(index='sampleid', columns='INTERVAL', values=cnv_type)\n",
    "\n",
    "    dosage_ = pd.concat([dosage_, dosage_pivot])\n",
    "# dosage_.to_csv(out_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5493f8c-6cdc-4cdc-9157-680924abd629",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d8eecf1-4cbf-4aa6-a81c-e21239f97a19",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e67bb1ce-7d63-47aa-a0ab-0fb64fdf1a47",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c273d2fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cnv_dosage = pd.read_csv('/data/GP2/raw_genotypes/cnvs/release4/dosages/EUR/CNV_EUR_chr1_PERCENT_BAF_INSERTION.csv')\n",
    "dose_vc = cnv_dosage.apply(lambda x: x.isnull().value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "3b0970fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleid</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>AADACL3</th>\n",
       "      <th>AADACL4</th>\n",
       "      <th>ABCA4</th>\n",
       "      <th>ABCB10</th>\n",
       "      <th>ABCD3</th>\n",
       "      <th>ABL2</th>\n",
       "      <th>ACADM</th>\n",
       "      <th>ACAP3</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNHIT6</th>\n",
       "      <th>ZP4</th>\n",
       "      <th>ZRANB2</th>\n",
       "      <th>ZRANB2-AS1</th>\n",
       "      <th>ZRANB2-AS2</th>\n",
       "      <th>ZSCAN20</th>\n",
       "      <th>ZSWIM5</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>13611.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>True</th>\n",
       "      <td>NaN</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>...</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "      <td>13611.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 2581 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       sampleid  A3GALT2  AADACL3  AADACL4    ABCA4   ABCB10    ABCD3  \\\n",
       "False   13611.0      NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "True        NaN  13611.0  13611.0  13611.0  13611.0  13611.0  13611.0   \n",
       "\n",
       "          ABL2    ACADM    ACAP3  ...   ZNHIT6      ZP4   ZRANB2  ZRANB2-AS1  \\\n",
       "False      NaN      NaN      NaN  ...      NaN      NaN      NaN         NaN   \n",
       "True   13611.0  13611.0  13611.0  ...  13611.0  13611.0  13611.0     13611.0   \n",
       "\n",
       "       ZRANB2-AS2  ZSCAN20   ZSWIM5   ZYG11A   ZYG11B     ZZZ3  \n",
       "False         NaN      NaN      NaN      NaN      NaN      NaN  \n",
       "True      13611.0  13611.0  13611.0  13611.0  13611.0  13611.0  \n",
       "\n",
       "[2 rows x 2581 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dose_vc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "8eaefe81-9ec6-4179-9f2c-776ba966b8ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/update_cnv_ids.swarm','w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        label_dir = f'{cnv_path}/dosages/{label}'\n",
    "        out_dir = f'/data/GP2/releases/gp2tier2/release4_14022023/cnvs/{label}'\n",
    "        os.makedirs(out_dir, exist_ok=True)\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "\n",
    "                dosagefile = f'{label_dir}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                dosagefile_out = f'{out_dir}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                pheno_out = f'{label_dir}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "                cmd = f'\\\n",
    "python update_cnv_ids.py \\\n",
    "--dosagefile {dosagefile} \\\n",
    "--key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv \\\n",
    "--pheno_out {pheno_out} \\\n",
    "--out_path {dosagefile_out}'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "051dc35b-f2f2-4eb8-877a-a720e822fb04",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58356484\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/update_cnv_ids.swarm -g 8 -t 8 --logdir {swarm_scripts_dir}/logs --module python/3.8 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "7936e3e6-fb58-4957-bf60-17ae52a0a4a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr1_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr1_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr1_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr2_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr2_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr2_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr3_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr3_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr3_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr4_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr4_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr4_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr5_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr5_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr5_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr6_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr6_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr6_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr7_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr7_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr7_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr8_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr8_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr8_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr9_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr9_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr9_PERCENT_BAF_INSERTION_gp2ids.csv\n",
      "python update_cnv_ids.py --dosagefile /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/CNV_MDE_chr10_PERCENT_BAF_INSERTION.csv --key /data/GP2/raw_genotypes/cnvs/release4/key/release4_key.csv --pheno_out /data/GP2/raw_genotypes/cnvs/release4/dosages/MDE/GP2_MDE_chr10_PERCENT_BAF_INSERTION.pheno --out_path /data/GP2/releases/gp2tier2/release4_14022023/cnvs/MDE/CNV_MDE_chr10_PERCENT_BAF_INSERTION_gp2ids.csv\n"
     ]
    }
   ],
   "source": [
    "!head {swarm_scripts_dir}/update_cnv_ids.swarm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "3b3a4dec-9b45-40ac-a1bc-0c51090c2819",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_out_path = '/data/GP2/raw_genotypes/cnvwas'\n",
    "with open(f'{swarm_scripts_dir}/cnvwas.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "\n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                pheno = f'{cnv_path}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "                covar = f'{cnv_path}/GP2_round4_JAN_24_2023_{label}_clean.cov'\n",
    "                out_path = f'{cnv_out_path}/GP2_{label}_chr{chrom}_{cnv_type}_cnvwas.tab'\n",
    "\n",
    "                cmd = f'\\\n",
    "python run_cnvwas_pipeline.py \\\n",
    "--cnv_dosage_file {dosagefile} \\\n",
    "--out_path {out_path} \\\n",
    "--pheno {pheno} \\\n",
    "--covar {covar}'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "9098b284-30f8-4c79-9944-9dc8da375090",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58077962\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnvwas.swarm -g 32 -t 16 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "931f3e00-37b1-4a89-b7b3-529d0ca4c7ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n",
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n",
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from GWAS.gwas import calculate_inflation\n",
    "label = 'EUR'\n",
    "# cnv_type = 'PERCENT_BAF_INSERTION'\n",
    "# cnv_type = 'PERCENT_L2R_DUPLICATION'\n",
    "# chrom = '1'\n",
    "pheno_df = pd.read_csv(f'{cnv_path}/GP2_EUR_chr1_{cnv_type}.pheno', sep='\\t', engine='python')\n",
    "ncontrols = pheno_df.pheno.value_counts()[0]\n",
    "ncases = pheno_df.pheno.value_counts()[1]\n",
    "cnv_lambda_dict = dict()\n",
    "for cnv_type in cnv_types:\n",
    "    total_cnvtype_df = pd.DataFrame()\n",
    "    for chrom in chroms:\n",
    "        tab = f'{cnv_out_path}/GP2_{label}_chr{chrom}_{cnv_type}_cnvwas.tab'\n",
    "\n",
    "        if os.path.isfile(tab):\n",
    "\n",
    "            sumstats = pd.read_csv(tab, sep='\\s+', engine='python')\n",
    "            # print(tab)\n",
    "            # infl = calculate_inflation(sumstats.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "            # print(infl)\n",
    "            # print()\n",
    "            # print()\n",
    "            total_cnvtype_df = pd.concat([total_cnvtype_df, sumstats], ignore_index=True)\n",
    "        else:\n",
    "            print(f'{tab} does not exist!')\n",
    "\n",
    "    infl = calculate_inflation(total_cnvtype_df.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "    cnv_lambda_dict[cnv_type] = infl\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "69500012-0c99-45cf-8c91-0a8afa7e3ff4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'PERCENT_BAF_INSERTION': {'pass': True,\n",
       "  'step': 'lambda_calculation',\n",
       "  'metrics': {'inflation': 2.101750589371727}},\n",
       " 'PERCENT_L2R_DELETION': {'pass': True,\n",
       "  'step': 'lambda_calculation',\n",
       "  'metrics': {'inflation': 14.555569305672298}},\n",
       " 'PERCENT_L2R_DUPLICATION': {'pass': True,\n",
       "  'step': 'lambda_calculation',\n",
       "  'metrics': {'inflation': 1.3423052908502526}}}"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_lambda_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "20feb548-fbfc-49d5-8d92-98ee2afc218a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'lambda_calculation',\n",
       " 'metrics': {'inflation': 1.3423052908502526}}"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "pheno = pd.read_csv(f'{cnv_path}/GP2_EUR_chr1_{cnv_type}.pheno', sep='\\t', engine='python')\n",
    "ncontrols = pheno.pheno.value_counts()[0]\n",
    "ncases = pheno.pheno.value_counts()[1]\n",
    "\n",
    "infl = calculate_inflation(total_cnvtype_df.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "infl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "0e4d119f-b74a-4af8-8548-dd4c2b844198",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8066"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "02570930-f095-4a36-a28d-d6d0b93c723c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13911"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno.pheno.value_counts()[1] + pheno.pheno.value_counts()[0] + 1475"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d68612b4-bb57-481d-8506-b1cd1ab71891",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    8066\n",
       "0    4370\n",
       "Name: pheno, dtype: int64"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ce8cb4aa-172a-4258-911f-2e38dd93c0a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       " 2    8711\n",
       " 1    4876\n",
       "-9    1475\n",
       "Name: pheno, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam = pd.read_csv(f'/data/GP2/quality_control/release4/genotype_qc/GP2_round4_JAN_24_2023_{label}.fam', sep='\\t', header=None, names=['fid','iid','pat','mat','sex','pheno'], engine='python')\n",
    "fam.pheno.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06cd40f1-e86e-4a8e-9f2a-4f2c36cd8eba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "3087cdf2-024f-4d96-bdb2-5cfb319558f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/GP2/raw_genotypes/cnvs/release4/CNV_EUR_chr1_PERCENT_L2R_DUPLICATION_gp2ids.csv'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "cnv_dosage_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "f68a016a-a0e2-4fd9-a513-49c333668b0f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638843\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639250\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639028\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639249\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639250\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639149\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.636706\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639198\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639250\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639085\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638965\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639251\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639251\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639252\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639221\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639227\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639244\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638848\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639203\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639254\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639143\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639249\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638589\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639251\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639117\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639252\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638968\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639045\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639113\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639254\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638554\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639185\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639046\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.638780\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639253\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639052\n",
      "         Iterations 6\n",
      "Optimization terminated successfully.\n",
      "         Current function value: 0.639190\n",
      "         Iterations 6\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import numpy as np\n",
    "from numpy.core.numeric import NaN\n",
    "import pandas as pd\n",
    "import glob\n",
    "import shutil\n",
    "from sklearn.preprocessing import MinMaxScaler\n",
    "import statsmodels.api as sm\n",
    "\n",
    "# Supress copy warning.\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "from QC.utils import shell_do\n",
    "\n",
    "label = 'EUR'\n",
    "chrom = '1'\n",
    "cnv_type = 'PERCENT_L2R_DUPLICATION'\n",
    "dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "pheno = f'{cnv_path}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "covar = f'{cnv_path}/GP2_round4_JAN_24_2023_{label}_clean.cov'\n",
    "out_path = f'{cnv_out_path}/GP2_{label}_chr{chrom}_{cnv_type}_cnvwas.tab'\n",
    "cnv_dosage_file = dosagefile\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "scaler = MinMaxScaler()\n",
    "dosage_df = pd.read_csv(cnv_dosage_file)\n",
    "#fix column names\n",
    "dosage_df.columns = [x.replace('-','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace('.','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace(' ','_') for x in dosage_df.columns]\n",
    "\n",
    "pheno_df = pd.read_csv(pheno, sep='\\t')\n",
    "covar_df = pd.read_csv(covar, sep='\\t')\n",
    "\n",
    "\n",
    "if covar_df.age_of_onset.isna().all():\n",
    "    covar_df.drop(columns=['age_of_onset'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age_of_onset'] = scaler.fit_transform(covar_df[['age_of_onset']])\n",
    "\n",
    "if covar_df.age.isna().all():\n",
    "    covar_df.drop(columns=['age'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age'] = scaler.fit_transform(covar_df[['age']])\n",
    "\n",
    "if covar_df.sex.isna().all():\n",
    "    covar_df.drop(columns=['sex'], inplace=True)\n",
    "\n",
    "covar_df.drop(columns=['FID'], inplace=True)\n",
    "covar_df.rename(columns={'GP2sampleID':'sampleid'}, inplace=True)\n",
    "\n",
    "data_df = dosage_df.merge(covar_df, on='sampleid', how='left').merge(pheno_df, on='sampleid', how='left').set_index('sampleid')\n",
    "\n",
    "rm_pred = [f'PC{i}' for i in range(1,21)] + ['sex','age_of_onset','age','pheno']\n",
    "\n",
    "pred_list = [x for x in data_df.columns if x not in rm_pred]\n",
    "covars_list = [x for x in data_df.columns if x not in pred_list + [f'PC{i}' for i in range(11,21)] + ['pheno']]\n",
    "\n",
    "\n",
    "genes_list = dosage_df.columns.drop('sampleid')\n",
    "data_df_final = data_df.copy()\n",
    "ctrl_stds = data_df.loc[data_df['pheno']==0, genes_list].std()\n",
    "ctrl_means = data_df.loc[data_df['pheno']==0, genes_list].mean()\n",
    "\n",
    "for gene in genes_list:\n",
    "    lower = (ctrl_means[gene]-(ctrl_stds[gene]*2))\n",
    "    upper = (ctrl_means[gene]+(ctrl_stds[gene]*2))         \n",
    "    data_df_final.loc[:, gene] = data_df_final.loc[:, gene].between(lower, upper, inclusive='both')\n",
    "    \n",
    "data_df_final.loc[:, genes_list] = data_df_final.loc[:, genes_list].astype(int)\n",
    "\n",
    "results = []\n",
    "fails = []\n",
    "\n",
    "for pred in range(len(pred_list)):\n",
    "    pred_name = pred_list[pred]\n",
    "    formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "\n",
    "    fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df_final).fit()\n",
    "\n",
    "    beta_coef  = fitted.params.loc[pred_name]\n",
    "    beta_se  = fitted.bse.loc[pred_name]\n",
    "    p_val = fitted.pvalues.loc[pred_name]\n",
    "\n",
    "    results.append((pred_name, beta_coef, beta_se, p_val))\n",
    "\n",
    "\n",
    "output = pd.DataFrame(results, columns=('PREDICTOR', 'BETA_COEF', 'BETA_SE','P_VAL'))\n",
    "# output.to_csv(out_path, sep='\\t', header=True, index=False)\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# def CNV_WAS(cnv_dosage_file, pheno, covar, out_path):\n",
    "#     scaler = MinMaxScaler()\n",
    "#     dosage_df = pd.read_csv(cnv_dosage_file)\n",
    "#     #fix column names\n",
    "#     dosage_df.columns = [x.replace('-','_') for x in dosage_df.columns]\n",
    "#     dosage_df.columns = [x.replace('.','_') for x in dosage_df.columns]\n",
    "#     dosage_df.columns = [x.replace(' ','_') for x in dosage_df.columns]\n",
    "\n",
    "#     pheno_df = pd.read_csv(pheno, sep='\\t')\n",
    "#     covar_df = pd.read_csv(covar, sep='\\t')\n",
    "\n",
    "\n",
    "#     if covar_df.age_of_onset.isna().all():\n",
    "#         covar_df.drop(columns=['age_of_onset'], inplace=True)\n",
    "#     else:\n",
    "#         covar_df.loc[:,'age_of_onset'] = scaler.fit_transform(covar_df[['age_of_onset']])\n",
    "\n",
    "#     if covar_df.age.isna().all():\n",
    "#         covar_df.drop(columns=['age'], inplace=True)\n",
    "#     else:\n",
    "#         covar_df.loc[:,'age'] = scaler.fit_transform(covar_df[['age']])\n",
    "\n",
    "#     if covar_df.sex.isna().all():\n",
    "#         covar_df.drop(columns=['sex'], inplace=True)\n",
    "\n",
    "#     covar_df.drop(columns=['FID'], inplace=True)\n",
    "#     covar_df.rename(columns={'GP2sampleID':'sampleid'}, inplace=True)\n",
    "\n",
    "#     data_df = dosage_df.merge(covar_df, on='sampleid', how='left').merge(pheno_df, on='sampleid', how='left').set_index('sampleid')\n",
    "\n",
    "#     rm_pred = [f'PC{i}' for i in range(1,21)] + ['sex','age_of_onset','age','pheno']\n",
    "\n",
    "#     pred_list = [x for x in data_df.columns if x not in rm_pred]\n",
    "#     covars_list = [x for x in data_df.columns if x not in pred_list + [f'PC{i}' for i in range(11,21)] + ['pheno']]\n",
    "\n",
    "\n",
    "#     genes_list = dosage_df.columns.drop('sampleid')\n",
    "#     data_df_final = data_df.copy()\n",
    "#     ctrl_stds = data_df.loc[data_df['pheno']==0, genes_list].std()\n",
    "#     ctrl_means = data_df.loc[data_df['pheno']==0, genes_list].mean()\n",
    "\n",
    "#     for gene in genes_list:\n",
    "#         lower = (ctrl_means[gene]-(ctrl_stds[gene]*2))\n",
    "#         upper = (ctrl_means[gene]+(ctrl_stds[gene]*2))         \n",
    "#         data_df_final.loc[:, gene] = data_df_final.loc[:, gene].between(lower, upper, inclusive='both')\n",
    "\n",
    "#     data_df_final.loc[:, genes_list] = data_df_final.loc[:, genes_list].astype(int)\n",
    "\n",
    "#     results = []\n",
    "#     fails = []\n",
    "\n",
    "#     for pred in range(len(pred_list)):\n",
    "#         pred_name = pred_list[pred]\n",
    "#         formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "\n",
    "#         fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df_final).fit()\n",
    "#         beta_coef  = fitted.params.loc[pred_name]\n",
    "#         beta_se  = fitted.bse.loc[pred_name]\n",
    "#         p_val = fitted.pvalues.loc[pred_name]\n",
    "\n",
    "#         results.append((pred_name, beta_coef, beta_se, p_val))\n",
    "\n",
    "\n",
    "#     output = pd.DataFrame(results, columns=('PREDICTOR', 'BETA_COEF', 'BETA_SE','P_VAL'))\n",
    "#     output.to_csv(out_path, sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "9b025936-4e9c-4281-8f63-f3ee1a68ac49",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'lambda_calculation',\n",
       " 'metrics': {'inflation': 1.4366622566944616}}"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from GWAS.gwas import calculate_inflation\n",
    "# pheno_df = pd.read_csv(f'{cnv_path}/GP2_EUR_chr1_{cnv_type}.pheno', sep='\\t', engine='python')\n",
    "ncontrols = pheno_df.pheno.value_counts()[0]\n",
    "ncases = pheno_df.pheno.value_counts()[1]\n",
    "\n",
    "infl = calculate_inflation(output.P_VAL, normalize=True, ncases=ncases, ncontrols=ncontrols)\n",
    "infl\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "20016864-9700-45f7-9cc5-c9c07d995d62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'pheno ~ WDR64 + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + sex + age + age_of_onset'"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "formula"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "2b45610b-ad6e-4de8-a407-512f1e04503c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACTN2</th>\n",
       "      <th>AGTRAP</th>\n",
       "      <th>ARHGEF16</th>\n",
       "      <th>C1orf101</th>\n",
       "      <th>C1orf167</th>\n",
       "      <th>CAMTA1</th>\n",
       "      <th>CSMD2</th>\n",
       "      <th>DAB1</th>\n",
       "      <th>DRAXIN</th>\n",
       "      <th>EDARADD</th>\n",
       "      <th>...</th>\n",
       "      <th>PC15</th>\n",
       "      <th>PC16</th>\n",
       "      <th>PC17</th>\n",
       "      <th>PC18</th>\n",
       "      <th>PC19</th>\n",
       "      <th>PC20</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>age_of_onset</th>\n",
       "      <th>pheno</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CORIELL_000178_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.006651</td>\n",
       "      <td>0.009075</td>\n",
       "      <td>-0.007526</td>\n",
       "      <td>0.000607</td>\n",
       "      <td>-0.005277</td>\n",
       "      <td>0.016480</td>\n",
       "      <td>1</td>\n",
       "      <td>0.646341</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAGE_000432_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.009945</td>\n",
       "      <td>0.006582</td>\n",
       "      <td>0.005732</td>\n",
       "      <td>-0.005267</td>\n",
       "      <td>-0.017189</td>\n",
       "      <td>-0.012584</td>\n",
       "      <td>1</td>\n",
       "      <td>0.767308</td>\n",
       "      <td>0.741228</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_000783_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001914</td>\n",
       "      <td>-0.002026</td>\n",
       "      <td>-0.007720</td>\n",
       "      <td>-0.002451</td>\n",
       "      <td>-0.003864</td>\n",
       "      <td>-0.006665</td>\n",
       "      <td>2</td>\n",
       "      <td>0.524390</td>\n",
       "      <td>0.578947</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_004856_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006815</td>\n",
       "      <td>-0.005230</td>\n",
       "      <td>-0.006799</td>\n",
       "      <td>0.006651</td>\n",
       "      <td>0.005691</td>\n",
       "      <td>-0.003396</td>\n",
       "      <td>1</td>\n",
       "      <td>0.621951</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COPN_000306_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004314</td>\n",
       "      <td>0.003814</td>\n",
       "      <td>0.000499</td>\n",
       "      <td>0.010819</td>\n",
       "      <td>0.001542</td>\n",
       "      <td>0.018862</td>\n",
       "      <td>1</td>\n",
       "      <td>0.548780</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAGE_001054_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.008115</td>\n",
       "      <td>-0.003489</td>\n",
       "      <td>0.001157</td>\n",
       "      <td>-0.002412</td>\n",
       "      <td>-0.000864</td>\n",
       "      <td>-0.004991</td>\n",
       "      <td>1</td>\n",
       "      <td>0.722300</td>\n",
       "      <td>0.653509</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LCC_000146_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008954</td>\n",
       "      <td>-0.009757</td>\n",
       "      <td>0.003442</td>\n",
       "      <td>-0.000514</td>\n",
       "      <td>-0.001066</td>\n",
       "      <td>-0.003567</td>\n",
       "      <td>1</td>\n",
       "      <td>0.548780</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_002459_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001957</td>\n",
       "      <td>0.003340</td>\n",
       "      <td>0.000891</td>\n",
       "      <td>-0.007043</td>\n",
       "      <td>-0.000739</td>\n",
       "      <td>0.000489</td>\n",
       "      <td>1</td>\n",
       "      <td>0.536585</td>\n",
       "      <td>0.484211</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCM_000036_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.006514</td>\n",
       "      <td>-0.008966</td>\n",
       "      <td>-0.010393</td>\n",
       "      <td>-0.007601</td>\n",
       "      <td>0.000784</td>\n",
       "      <td>-0.014942</td>\n",
       "      <td>2</td>\n",
       "      <td>0.571582</td>\n",
       "      <td>0.577225</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_001134_s1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006826</td>\n",
       "      <td>0.008794</td>\n",
       "      <td>-0.007286</td>\n",
       "      <td>0.005986</td>\n",
       "      <td>-0.005052</td>\n",
       "      <td>-0.010628</td>\n",
       "      <td>1</td>\n",
       "      <td>0.304878</td>\n",
       "      <td>0.378947</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12436 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   ACTN2  AGTRAP  ARHGEF16  C1orf101  C1orf167  CAMTA1  CSMD2  \\\n",
       "sampleid                                                                        \n",
       "CORIELL_000178_s1      1       1         1         1         1       1      1   \n",
       "PAGE_000432_s1         1       1         1         1         1       1      0   \n",
       "CORIELL_000783_s1      1       1         1         1         1       1      1   \n",
       "CORIELL_004856_s1      1       1         1         1         1       1      1   \n",
       "COPN_000306_s1         1       1         1         1         1       1      1   \n",
       "...                  ...     ...       ...       ...       ...     ...    ...   \n",
       "PAGE_001054_s1         1       1         1         1         1       1      1   \n",
       "LCC_000146_s1          1       1         1         1         1       1      1   \n",
       "CORIELL_002459_s1      1       1         1         1         1       1      1   \n",
       "BCM_000036_s1          1       1         1         1         1       1      1   \n",
       "CORIELL_001134_s1      1       1         0         1         1       1      0   \n",
       "\n",
       "                   DAB1  DRAXIN  EDARADD  ...      PC15      PC16      PC17  \\\n",
       "sampleid                                  ...                                 \n",
       "CORIELL_000178_s1     1       1        1  ... -0.006651  0.009075 -0.007526   \n",
       "PAGE_000432_s1        0       1        1  ... -0.009945  0.006582  0.005732   \n",
       "CORIELL_000783_s1     1       1        1  ...  0.001914 -0.002026 -0.007720   \n",
       "CORIELL_004856_s1     1       1        1  ...  0.006815 -0.005230 -0.006799   \n",
       "COPN_000306_s1        1       1        1  ...  0.004314  0.003814  0.000499   \n",
       "...                 ...     ...      ...  ...       ...       ...       ...   \n",
       "PAGE_001054_s1        1       1        1  ... -0.008115 -0.003489  0.001157   \n",
       "LCC_000146_s1         1       1        1  ...  0.008954 -0.009757  0.003442   \n",
       "CORIELL_002459_s1     1       1        1  ...  0.001957  0.003340  0.000891   \n",
       "BCM_000036_s1         1       1        1  ... -0.006514 -0.008966 -0.010393   \n",
       "CORIELL_001134_s1     1       1        1  ...  0.006826  0.008794 -0.007286   \n",
       "\n",
       "                       PC18      PC19      PC20  sex       age  age_of_onset  \\\n",
       "sampleid                                                                       \n",
       "CORIELL_000178_s1  0.000607 -0.005277  0.016480    1  0.646341      0.600000   \n",
       "PAGE_000432_s1    -0.005267 -0.017189 -0.012584    1  0.767308      0.741228   \n",
       "CORIELL_000783_s1 -0.002451 -0.003864 -0.006665    2  0.524390      0.578947   \n",
       "CORIELL_004856_s1  0.006651  0.005691 -0.003396    1  0.621951      0.577225   \n",
       "COPN_000306_s1     0.010819  0.001542  0.018862    1  0.548780      0.577225   \n",
       "...                     ...       ...       ...  ...       ...           ...   \n",
       "PAGE_001054_s1    -0.002412 -0.000864 -0.004991    1  0.722300      0.653509   \n",
       "LCC_000146_s1     -0.000514 -0.001066 -0.003567    1  0.548780      0.577225   \n",
       "CORIELL_002459_s1 -0.007043 -0.000739  0.000489    1  0.536585      0.484211   \n",
       "BCM_000036_s1     -0.007601  0.000784 -0.014942    2  0.571582      0.577225   \n",
       "CORIELL_001134_s1  0.005986 -0.005052 -0.010628    1  0.304878      0.378947   \n",
       "\n",
       "                   pheno  \n",
       "sampleid                  \n",
       "CORIELL_000178_s1      1  \n",
       "PAGE_000432_s1         1  \n",
       "CORIELL_000783_s1      1  \n",
       "CORIELL_004856_s1      0  \n",
       "COPN_000306_s1         1  \n",
       "...                  ...  \n",
       "PAGE_001054_s1         1  \n",
       "LCC_000146_s1          0  \n",
       "CORIELL_002459_s1      1  \n",
       "BCM_000036_s1          1  \n",
       "CORIELL_001134_s1      1  \n",
       "\n",
       "[12436 rows x 61 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "data_df_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d5d2df9-3901-4321-8672-2d427082e1ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41dda188-496d-4149-887d-990acff9b742",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f4b0d49-c3a2-42b8-9156-9172bc4aadf6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41cb69b5-c464-4cc4-be12-d8e1704e84d4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "6081e4d8-a887-412f-98b2-77240260d7b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# output final release samples\n",
    "cnv_out_path = '/data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs'\n",
    "release3_samples = '/data/CARD/PD/GP2/genotypes/GP2/round3/clean/release3.samples'\n",
    "release_key_final = pd.read_csv('/data/CARD/PD/GP2/genotypes/GP2/round3/clean/release3_master_key.csv')\n",
    "release_key_out = release_key_final.loc[release_key_final.pheno!=-9]\n",
    "release_key_out['GP2sampleID'].to_csv(release3_samples, index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "82dc85bb-6d29-405e-bed6-efab5c1cc434",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/cnv_dosage_release_split.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        sample_file = f'/data/CARD/PD/GP2/genotypes/GP2/round3/clean/{label}_release3.samples'\n",
    "        label_samples = release_key_final.loc[release_key_final['label']==label]\n",
    "        label_samples['GP2sampleID'].to_csv(sample_file, header=False, index=False)\n",
    "\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "                \n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                dosage_out = f'{cnv_out_path}/CNV_{label}_chr{chrom}_{cnv_type}_release3.csv'\n",
    "                \n",
    "                cmd = f'\\\n",
    "python split_release_samples.py \\\n",
    "--cnv_dosage_file {dosagefile} \\\n",
    "--samples {sample_file} \\\n",
    "--out_path {dosage_out}'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "56353619-d819-4347-ab81-aec8d05de647",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "53715605\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_dosage_release_split.swarm -g 8 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "64f9b5b4-92c2-430c-a9ec-b09aae6aab6b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AMR PERCENT_BAF_INSERTION\n",
      "412\n",
      "\n",
      "AMR PERCENT_L2R_DELETION\n",
      "412\n",
      "\n",
      "AMR PERCENT_L2R_DUPLICATION\n",
      "412\n",
      "\n",
      "EUR PERCENT_BAF_INSERTION\n",
      "10727\n",
      "\n",
      "EUR PERCENT_L2R_DELETION\n",
      "10727\n",
      "\n",
      "EUR PERCENT_L2R_DUPLICATION\n",
      "10727\n",
      "\n",
      "SAS PERCENT_BAF_INSERTION\n",
      "59\n",
      "\n",
      "SAS PERCENT_L2R_DELETION\n",
      "59\n",
      "\n",
      "SAS PERCENT_L2R_DUPLICATION\n",
      "59\n",
      "\n",
      "MDE PERCENT_BAF_INSERTION\n",
      "62\n",
      "\n",
      "MDE PERCENT_L2R_DELETION\n",
      "62\n",
      "\n",
      "MDE PERCENT_L2R_DUPLICATION\n",
      "62\n",
      "\n",
      "EAS PERCENT_BAF_INSERTION\n",
      "137\n",
      "\n",
      "EAS PERCENT_L2R_DELETION\n",
      "137\n",
      "\n",
      "EAS PERCENT_L2R_DUPLICATION\n",
      "137\n",
      "\n",
      "AAC PERCENT_BAF_INSERTION\n",
      "1215\n",
      "\n",
      "AAC PERCENT_L2R_DELETION\n",
      "1215\n",
      "\n",
      "AAC PERCENT_L2R_DUPLICATION\n",
      "1215\n",
      "\n",
      "AFR PERCENT_BAF_INSERTION\n",
      "404\n",
      "\n",
      "AFR PERCENT_L2R_DELETION\n",
      "404\n",
      "\n",
      "AFR PERCENT_L2R_DUPLICATION\n",
      "404\n",
      "\n",
      "CAS PERCENT_BAF_INSERTION\n",
      "303\n",
      "\n",
      "CAS PERCENT_L2R_DELETION\n",
      "303\n",
      "\n",
      "CAS PERCENT_L2R_DUPLICATION\n",
      "303\n",
      "\n",
      "AJ PERCENT_BAF_INSERTION\n",
      "923\n",
      "\n",
      "AJ PERCENT_L2R_DELETION\n",
      "923\n",
      "\n",
      "AJ PERCENT_L2R_DUPLICATION\n",
      "923\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# CHECK COUNTS\n",
    "total_dosages = 0\n",
    "for label in ancestry_labels:\n",
    "    for cnv_type in cnv_types:\n",
    "        dosage_out = f'{cnv_out_path}/CNV_{label}_chr1_{cnv_type}_release3.csv'\n",
    "        print(label, cnv_type)\n",
    "        label_count = !cat {dosage_out} | wc -l\n",
    "        label_count = int(label_count[0])\n",
    "        total_dosages += label_count\n",
    "        print(label_count)\n",
    "        print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "3df93e96-b169-4e45-ad32-76904be6cf22",
   "metadata": {},
   "outputs": [],
   "source": [
    "# push to release bucket\n",
    "with open(f'{swarm_scripts_dir}/gcp_push_cnvs.swarm','w') as f:\n",
    "    \n",
    "    for label in ancestry_labels:\n",
    "        for chrom in chroms:\n",
    "    \n",
    "#         !module load google-cloud-sdk/397.0.0; gsutil cp {cnv_out_path}/CNV_{label}_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/\n",
    "            cp1 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "            cp2 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_L2R_DELETION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "            cp3 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_L2R_DUPLICATION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "\n",
    "            cps = [cp1, cp2, cp3]\n",
    "\n",
    "            for cp in cps:\n",
    "                f.write(f'{cp}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ddb097af-919f-41f3-a434-2b624c4dbde4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "53719849\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/gcp_push_cnvs.swarm -g 8 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --module google-cloud-sdk/397.0.0 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "70c7c14a-cb24-4e02-961f-f4fb69834166",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "594\n"
     ]
    }
   ],
   "source": [
    "!module load google-cloud-sdk/397.0.0; gsutil cp /data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs/CNV_AMR_chr1_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/AMR/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "2a133f0b-dc2b-4bce-91d5-5040ca024786",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-rw----+ 1 vitaled2 CARD 4.2K Dec  5 19:27 /data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs/CNV_AJ_chr22_PERCENT_BAF_INSERTION_release3.csv\n"
     ]
    }
   ],
   "source": [
    "!ls -lh {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_BAF_INSERTION_release3.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e4aa47ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11696\n"
     ]
    }
   ],
   "source": [
    "!cat /data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_chr1_PERCENT_BAF_INSERTION_gp2ids.csv | wc -l"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.8",
   "language": "python",
   "name": "py3.8"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
