{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "163da59f-cf9c-4665-8221-8524f6a364e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "import numpy as np\n",
    "import glob\n",
    "from sklearn.preprocessing import MinMaxScaler\n",
    "import statsmodels.api as sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ac49c984-53cf-4aca-9b10-2c20e1321d8d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_93564/4258865316.py:17: DtypeWarning: Columns (36,37) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  key = pd.read_csv(f'{key_file}', sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "idat_path = '/data/CARD/PD/GP2/raw_genotypes/GP2_idats'\n",
    "swarm_scripts_dir = f'/data/CARD/PD/GP2/swarm_scripts'\n",
    "\n",
    "ilmn_files_path = '/data/CARD/PD/GP2/ilmn_files'\n",
    "# A1 is hg19, A2 is hg38. csv needed for indel calls\n",
    "bpm_csv = f'{ilmn_files_path}/NeuroBooster_20042459_A2.csv' \n",
    "bpm = f'{ilmn_files_path}/NeuroBooster_20042459_A2.bpm'\n",
    "egt = f'{ilmn_files_path}/recluster_09272022.egt'\n",
    "iaap = f'{ilmn_files_path}/iaap-cli/iaap-cli'\n",
    "\n",
    "ref_fasta = '/data/vitaled2/ref/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna'\n",
    "\n",
    "chroms = [str(i) for i in range(1,23)] + ['X','Y','M']\n",
    "\n",
    "\n",
    "key_file = '/data/CARD/PD/GP2/clinical/key_merge/GP2_merge/GP2_master_key_full.txt'\n",
    "key = pd.read_csv(f'{key_file}', sep='\\t')\n",
    "key.loc[:,'filename'] = key.loc[:,'SentrixBarcode_A'].astype(str) + '_' + key.loc[:,'SentrixPosition_A'].astype(str)\n",
    "\n",
    "cohorts = [\n",
    "    'SYNAPS-KZ', \n",
    "    'CORIELL', \n",
    "    'IPDGCEA-JU', \n",
    "    'KOC', \n",
    "    'IPDGCAF-NG', \n",
    "    'IPDGCEA-MU', \n",
    "    'PAGE', \n",
    "    'NZP3', \n",
    "    'S4', \n",
    "    'BCM',\n",
    "    'BLAACPD-RUSH', \n",
    "    'BIOFIND', \n",
    "    'UMD', \n",
    "    'BLAACPD-UAB', \n",
    "    'LCC', \n",
    "    'SYNAPS-GR', \n",
    "    'MDGAP-QSBB', \n",
    "    'APGS', \n",
    "    'MDGAP-KINGS', \n",
    "    'MDGAP-IBB', \n",
    "    'BLAACPD-UC', \n",
    "    'BLAACPD-KPM', \n",
    "    'PPMI', \n",
    "    'IPDGCEA-UKM'\n",
    "]\n",
    "\n",
    "release_key = key.loc[key['study'].isin(cohorts)]\n",
    "release_key.to_csv('/data/CARD/PD/GP2/clinical/key_merge/GP2_merge/release3_key.csv', index=False, header=True)\n",
    "\n",
    "samples_list = release_key.loc[:,'filename'].unique()\n",
    "barcodes_list = list(set([x.split('_')[0] for x in samples_list]))\n",
    "\n",
    "genes = pd.read_csv('/data/CARD/PD/GP2/ref_panel/glist-hg38', sep='\\s+', header=None, names=['chr','start','end','symbol'], dtype={'chr':str,'start':int,'end':int})\n",
    "genes.columns = ['CHR','START','STOP','NAME']\n",
    "genes_out = genes.loc[genes.CHR.isin(chroms)]\n",
    "genes_out[['NAME','CHR','START','STOP']].to_csv('/data/CARD/PD/GP2/ref_panel/glist_hg38_intervals.csv', index=False)\n",
    "# gene_list originally from plink glist\n",
    "gene_list = '/data/CARD/PD/GP2/ref_panel/glist_hg38_intervals.csv'\n",
    "\n",
    "raw_geno_path = '/data/CARD/PD/GP2/raw_genotypes'\n",
    "snp_metrics_path = f'{raw_geno_path}/GP2_snp_metrics'\n",
    "idat_path = f'{raw_geno_path}/GP2_idats'\n",
    "plink_file_path = f'{raw_geno_path}/GP2_plink'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "59346c9d-e1bd-47fd-b503-c9f592ff4dd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21895"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# len(samples_list)\n",
    "# sorted(cohorts)\n",
    "len(release_key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d146821f-30ff-496e-a9ad-bd2e1b90ac08",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get snp metrics\n",
    "bcftools_plugins_path = 'bin'\n",
    "with open(f'{swarm_scripts_dir}/snp_metrics.swarm', 'w') as f:\n",
    "    for code in barcodes_list:\n",
    "        idat_path_ = f'{idat_path}/{code}'\n",
    "        metrics_out = f'{snp_metrics_path}'\n",
    "\n",
    "        cmd = f'\\\n",
    "python run_snp_metrics_pipeline.py \\\n",
    "--idat_path {idat_path_} \\\n",
    "--bpm {bpm} \\\n",
    "--bpm_csv {bpm_csv} \\\n",
    "--egt {egt} \\\n",
    "--ref_fasta {ref_fasta} \\\n",
    "--out_path {metrics_out} \\\n",
    "--iaap {iaap} \\\n",
    "--bcftools_plugins_path {bcftools_plugins_path}'\n",
    "        f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "117e910d-509a-4ff5-af24-e5f3d07da1fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "!swarm -f {swarm_scripts_dir}/snp_metrics.swarm -g 16 -t 16 --time=04:00:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "cffc06e3-dfa9-4654-b599-d63c485b3b28",
   "metadata": {},
   "outputs": [],
   "source": [
    "# check if snp metrics produced across all samples\n",
    "completed_list = []\n",
    "missing_list = []\n",
    "for sample in samples_list:\n",
    "    code = sample.split('_')[0]\n",
    "    metrics_path = f'{snp_metrics_path}/{code}'\n",
    "    for chrom in chroms:\n",
    "        mfile = f'{metrics_path}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "        if os.path.isfile(mfile):\n",
    "            completed_list.append(sample)\n",
    "        else:\n",
    "            missing_list.append(sample)\n",
    "completed_final = list(set(completed_list))\n",
    "missing_final =list(set(missing_list))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "b7c4dff8-569a-4d46-b4f2-186363aacb39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21503\n",
      "144\n"
     ]
    }
   ],
   "source": [
    "print(len(completed_final))\n",
    "print(len(missing_final))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcd9284a-4a05-46d9-892e-6bda06354195",
   "metadata": {},
   "outputs": [],
   "source": [
    "# now check if we have idats for missing snp metrics\n",
    "missing_idats = []\n",
    "exists_idats = []\n",
    "for sample in missing_final:\n",
    "    code = sample.split('_')[0]\n",
    "    grn = f'{idat_path}/{code}/{sample}_Grn.idat'\n",
    "    red = f'{idat_path}/{code}/{sample}_Red.idat'\n",
    "    if os.path.isfile(grn) & os.path.isfile(red):\n",
    "        exists_idats.append(sample)\n",
    "    else:\n",
    "        missing_idats.append(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc859aa9-0359-427f-ae76-0f0fb2a4596e",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(len(missing_idats))\n",
    "print(len(exists_idats))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7bf097b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AAC.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AAC_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AFR.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AFR_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AJ.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AJ_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AMR.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AMR_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_CAS.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_CAS_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_EAS.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_EAS_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_EUR.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_EUR_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_MDE.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_MDE_clean.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_SAS.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_SAS_clean.bim\n"
     ]
    }
   ],
   "source": [
    "!ls {cnv_path}/*.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8a949c9a-3b34-422e-bc96-b224e6b86681",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_path = f'/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3'\n",
    "# using just autosomes for CNV analysis due to call quality\n",
    "chroms = [str(i) for i in range(1,23)]\n",
    "\n",
    "ancestry_labels = [x.split('/')[-1].replace('.bed','').split('_')[-1] for x in glob.glob('/data/CARD/PD/GP2/genotypes/GP2/round3/clean/GP2_round3_NEW_MERGE*.bed') if '_maf_hwe' not in x]\n",
    "\n",
    "release_key[['FID','GP2sampleID']].to_csv(f'{cnv_path}/release3.samples', sep='\\t', header=False, index=False)\n",
    "release_key[['GP2sampleID','IID']].to_csv(f'{cnv_path}/release3_sample_id_key.csv')\n",
    "release_covars = release_key.loc[:,['FID', 'GP2sampleID','sex_for_qc', 'age', 'age_of_onset']]\n",
    "\n",
    "cnv_types = ['PERCENT_BAF_INSERTION','PERCENT_L2R_DELETION','PERCENT_L2R_DUPLICATION']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea017911-2cff-4adf-9952-8bb3b71f5a14",
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp QC for cnvs\n",
    "for label in ancestry_labels:\n",
    "    out_name = f'GP2_round3_NEW_MERGE_{label}'\n",
    "    out_path = f'{cnv_path}/{out_name}'\n",
    "    geno_path = f'/data/CARD/PD/GP2/genotypes/GP2/round3/clean/{out_name}'\n",
    "\n",
    "    cmd1 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {geno_path} \\\n",
    "--keep {cnv_path}/release3.samples \\\n",
    "--make-bed \\\n",
    "--out {out_path}'\n",
    "\n",
    "    cmd2 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {out_path} \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.02 \\\n",
    "--hwe 5e-6 \\\n",
    "--autosome \\\n",
    "--make-bed \\\n",
    "--out {out_path}_clean'\n",
    "\n",
    "    cmd3 = f'\\\n",
    "module load plink/1.9; plink \\\n",
    "--bfile {out_path}_clean \\\n",
    "--pca \\\n",
    "--out {out_path}_clean_pcs'\n",
    "\n",
    "    cmds = [cmd1, cmd2, cmd3]\n",
    "\n",
    "    for cmd in cmds:\n",
    "        !{cmd}\n",
    "\n",
    "    pcs = pd.read_csv(f'{out_path}_clean_pcs.eigenvec', sep='\\s+')\n",
    "    pc_num = pcs.iloc[:, 2:].shape[1]\n",
    "    pc_names = ['FID','GP2sampleID'] + [f'PC{i}' for i in range(1, pc_num+1)]\n",
    "    pcs.columns = pc_names\n",
    "\n",
    "    cov = pcs.merge(release_covars, on=['FID','GP2sampleID'], how='left')\n",
    "    cov.age.fillna(cov.age.mean(), inplace=True)\n",
    "    cov.age_of_onset.fillna(cov.age_of_onset.mean(), inplace=True)\n",
    "    cov.sex_for_qc.fillna(cov.sex_for_qc.median(), inplace=True)\n",
    "    cov.rename(columns={'GP2sampleID':'sampleid','sex_for_qc':'sex'}, inplace=True)\n",
    "    cov.to_csv(f'{out_path}_clean.cov', sep='\\t', header=True, index=False)\n",
    "\n",
    "    samples = cov.merge(release_key[['GP2sampleID','IID']], left_on='sampleid', right_on='GP2sampleID', how='left')\n",
    "    samples['IID'].to_csv(f'{out_path}_clean_barcode.samples', header=False, index=False)\n",
    "\n",
    "    bim = pd.read_csv(f'{out_path}_clean.bim', sep='\\s+', header=None, names=['chr','id','pos','bp','a1','a2'], usecols=['id'])\n",
    "    bim.to_csv(f'{out_path}_clean.snps', sep='\\t', header=False, index=False)\n",
    "        \n",
    "\n",
    "        \n",
    "# skip ld pruning until we have better quality rare calls\n",
    "# --exclude exclusion_regions_hg38.txt \\\n",
    "\n",
    "\n",
    "#         cmd3 = f'\\\n",
    "# plink \\\n",
    "# --bfile {geno}_release2_maf_geno_hwe \\\n",
    "# --indep-pairwise 1000 10 0.02 \\\n",
    "# --autosome \\\n",
    "# --out {geno}_release2_prune'\n",
    "    \n",
    "#         cmd4 = f'\\\n",
    "# plink \\\n",
    "# --bfile {geno}_release2_maf_geno_hwe \\\n",
    "# --extract {geno}_release2_prune.prune.in \\\n",
    "# --make-bed \\\n",
    "# --out {geno}_release2_clean'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "c0a75eb9-9d5f-4cfd-b81a-f97ac19841f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AMR\n",
      "419\n",
      "\n",
      "EUR\n",
      "12052\n",
      "\n",
      "SAS\n",
      "85\n",
      "\n",
      "MDE\n",
      "81\n",
      "\n",
      "EAS\n",
      "1394\n",
      "\n",
      "AAC\n",
      "1258\n",
      "\n",
      "AFR\n",
      "1423\n",
      "\n",
      "CAS\n",
      "310\n",
      "\n",
      "AJ\n",
      "1302\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for label in ancestry_labels:\n",
    "    out_name = f'GP2_round3_NEW_MERGE_{label}'\n",
    "    out_path = f'{cnv_path}/{out_name}'\n",
    "    print(label)\n",
    "    !cat {out_path}_clean.cov | wc -l\n",
    "    print()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eef546d6",
   "metadata": {},
   "source": [
    "# some testing for cnv caller"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "071759b1-b04e-42e5-8723-3a89c53e5772",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   CHR     START      STOP     NAME\n",
      "0    1  33306765  33321098  A3GALT2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_93564/3773821577.py:50: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  temp_df['BAF_insertion'] = np.where( (temp_df['BAlleleFreq'].between(0.65, 0.85, inclusive='neither')) | (temp_df['BAlleleFreq'].between(0.15, 0.35, inclusive='neither')), 1, 0)\n",
      "/tmp/ipykernel_93564/3773821577.py:51: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  temp_df['L2R_deletion'] = np.where( temp_df['LogRRatio'] < -0.2, 1, 0)\n",
      "/tmp/ipykernel_93564/3773821577.py:52: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  temp_df['L2R_insertion'] = np.where( temp_df['LogRRatio'] > 0.2, 1, 0)\n"
     ]
    }
   ],
   "source": [
    "from numpy.core.numeric import NaN\n",
    "sample = '204958250147_R03C01'\n",
    "chrom = '1'\n",
    "code = sample.split('_')[0]\n",
    "label = 'EUR'\n",
    "\n",
    "\n",
    "snp_metrics_file = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "cnv_out = f'{cnv_path}/CNV_{label}_{sample}_chr{chrom}.csv'\n",
    "intervals_file = f'/data/CARD/PD/GP2/ref_panel/glist_hg38_chr{chrom}.csv'\n",
    "min_variants = 10\n",
    "kb_window = 250\n",
    "bim_path = '/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_EUR_clean.bim'\n",
    "\n",
    "\n",
    "# Load in the data.\n",
    "metrics_df = pd.read_csv(snp_metrics_file, engine='c')\n",
    "bim = pd.read_csv(bim_path, sep='\\s+', header=None, names=['chr','id','pos','bp','a1','a2'], usecols=['id'])\n",
    "sample_df = metrics_df.loc[metrics_df.snpID.isin(bim.id)]\n",
    "\n",
    "temp_interval_df = pd.read_csv(intervals_file, engine='c')\n",
    "temp_interval_df.drop_duplicates(subset = [\"NAME\"], inplace=True, keep='first')\n",
    "intervals_df = temp_interval_df.copy()\n",
    "\n",
    "\n",
    "# # Now reduce just to the intervals of interest and summarize each interval.\n",
    "\n",
    "# # Break down L2R and BAF per gene.\n",
    "\n",
    "results = []\n",
    "\n",
    "# interval_list = intervals_df['NAME'].unique()\n",
    "interval_list = ['A3GALT2']\n",
    "\n",
    "for INTERVAL in interval_list:\n",
    "    print(intervals_df.loc[intervals_df['NAME']==INTERVAL])\n",
    "\n",
    "    interval_CHR = intervals_df.loc[intervals_df['NAME'] == INTERVAL, 'CHR'].item()\n",
    "    interval_START_gene = intervals_df.loc[intervals_df['NAME'] == INTERVAL, 'START'].item()\n",
    "    interval_STOP_gene = intervals_df.loc[intervals_df['NAME'] == INTERVAL, 'STOP'].item()\n",
    "    interval_START = interval_START_gene - (kb_window*1000)\n",
    "    interval_STOP = interval_STOP_gene + (kb_window*1000)\n",
    "\n",
    "    temp_df = sample_df.loc[(sample_df['chromosome'] == interval_CHR) & (sample_df['position'] >= interval_START) & (sample_df['position'] <= interval_STOP)]\n",
    "\n",
    "    if temp_df.shape[0] < min_variants:\n",
    "\n",
    "        results.append((INTERVAL, temp_df.shape[0], NaN, NaN, NaN, interval_START, interval_START_gene, interval_STOP_gene, interval_STOP))\n",
    "    else:\n",
    "        temp_df['BAF_insertion'] = np.where( (temp_df['BAlleleFreq'].between(0.65, 0.85, inclusive='neither')) | (temp_df['BAlleleFreq'].between(0.15, 0.35, inclusive='neither')), 1, 0)\n",
    "        temp_df['L2R_deletion'] = np.where( temp_df['LogRRatio'] < -0.2, 1, 0)\n",
    "        temp_df['L2R_insertion'] = np.where( temp_df['LogRRatio'] > 0.2, 1, 0)\n",
    "        PERCENT_BAF_INSERTION = temp_df['BAF_insertion'].mean()\n",
    "        PERCENT_L2R_DELETION = temp_df['L2R_deletion'].mean()\n",
    "        PERCENT_L2R_INSERTION = temp_df['L2R_insertion'].mean()\n",
    "        results.append((INTERVAL, temp_df.shape[0], PERCENT_BAF_INSERTION, PERCENT_L2R_DELETION, PERCENT_L2R_INSERTION, interval_START, interval_START_gene, interval_STOP_gene, interval_STOP))\n",
    "    \n",
    "    output = pd.DataFrame(results, columns=('INTERVAL', 'NUM_VARIANTS', 'PERCENT_BAF_INSERTION', 'PERCENT_L2R_DELETION','PERCENT_L2R_DUPLICATION','START_PLUS_WINDOW','START','STOP','STOP_PLUS_WINDOW'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "3056027d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "111"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "6ffbdc5b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1445070"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "21895*22*3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3ec1960b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "60.97015503875969"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(metrics_df.snpID.unique())/2580"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "17181570",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>snpID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>ALLELE_A</th>\n",
       "      <th>ALLELE_B</th>\n",
       "      <th>BAlleleFreq</th>\n",
       "      <th>LogRRatio</th>\n",
       "      <th>R</th>\n",
       "      <th>Theta</th>\n",
       "      <th>GenTrain_Score</th>\n",
       "      <th>GType</th>\n",
       "      <th>Alt1</th>\n",
       "      <th>Alt2</th>\n",
       "      <th>GT</th>\n",
       "      <th>a1</th>\n",
       "      <th>a2</th>\n",
       "      <th>maf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>49554</td>\n",
       "      <td>1:49554-G-A</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.033875</td>\n",
       "      <td>-0.016783</td>\n",
       "      <td>1.691960</td>\n",
       "      <td>0.050030</td>\n",
       "      <td>0.699477</td>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AA</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.142857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1</td>\n",
       "      <td>826578</td>\n",
       "      <td>JHU_1.761957</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.001940</td>\n",
       "      <td>-0.111973</td>\n",
       "      <td>1.155930</td>\n",
       "      <td>0.962306</td>\n",
       "      <td>0.832417</td>\n",
       "      <td>AA</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>1</td>\n",
       "      <td>833439</td>\n",
       "      <td>rs12562811</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.490758</td>\n",
       "      <td>0.097902</td>\n",
       "      <td>0.764211</td>\n",
       "      <td>0.708393</td>\n",
       "      <td>0.443799</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.214286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>1</td>\n",
       "      <td>839745</td>\n",
       "      <td>JHU_1.775124</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.005320</td>\n",
       "      <td>-0.112962</td>\n",
       "      <td>1.098080</td>\n",
       "      <td>0.957153</td>\n",
       "      <td>0.790478</td>\n",
       "      <td>AA</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.083333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1</td>\n",
       "      <td>858952</td>\n",
       "      <td>JHU_1.794331</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.515637</td>\n",
       "      <td>-0.068786</td>\n",
       "      <td>1.557300</td>\n",
       "      <td>0.622296</td>\n",
       "      <td>0.540952</td>\n",
       "      <td>AB</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AB</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.357143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157272</th>\n",
       "      <td>1</td>\n",
       "      <td>248908807</td>\n",
       "      <td>JHU_1.249203005</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.985136</td>\n",
       "      <td>-0.042008</td>\n",
       "      <td>0.977678</td>\n",
       "      <td>0.923677</td>\n",
       "      <td>0.832624</td>\n",
       "      <td>AA</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BB</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.142857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157275</th>\n",
       "      <td>1</td>\n",
       "      <td>248910813</td>\n",
       "      <td>JHU_1.249205011</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.997110</td>\n",
       "      <td>-0.111099</td>\n",
       "      <td>1.261540</td>\n",
       "      <td>0.977864</td>\n",
       "      <td>0.497399</td>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BB</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.142857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157276</th>\n",
       "      <td>1</td>\n",
       "      <td>248914573</td>\n",
       "      <td>JHU_1.249208771</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.984007</td>\n",
       "      <td>0.041096</td>\n",
       "      <td>1.420910</td>\n",
       "      <td>0.949008</td>\n",
       "      <td>0.755758</td>\n",
       "      <td>AA</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BB</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157277</th>\n",
       "      <td>1</td>\n",
       "      <td>248916508</td>\n",
       "      <td>rs6704311</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.991263</td>\n",
       "      <td>-0.027500</td>\n",
       "      <td>1.276190</td>\n",
       "      <td>0.963003</td>\n",
       "      <td>0.777869</td>\n",
       "      <td>AA</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BB</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.071429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157302</th>\n",
       "      <td>1</td>\n",
       "      <td>248918526</td>\n",
       "      <td>JHU_1.249212724</td>\n",
       "      <td>204958250147_R03C01</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001921</td>\n",
       "      <td>0.067395</td>\n",
       "      <td>1.445640</td>\n",
       "      <td>0.026349</td>\n",
       "      <td>0.817646</td>\n",
       "      <td>AA</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AA</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59572 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chromosome   position            snpID            Sample_ID Ref Alt  \\\n",
       "0                1      49554      1:49554-G-A  204958250147_R03C01   A   G   \n",
       "21               1     826578     JHU_1.761957  204958250147_R03C01   C   T   \n",
       "27               1     833439       rs12562811  204958250147_R03C01   C   T   \n",
       "32               1     839745     JHU_1.775124  204958250147_R03C01   C   T   \n",
       "40               1     858952     JHU_1.794331  204958250147_R03C01   G   A   \n",
       "...            ...        ...              ...                  ...  ..  ..   \n",
       "157272           1  248908807  JHU_1.249203005  204958250147_R03C01   G   A   \n",
       "157275           1  248910813  JHU_1.249205011  204958250147_R03C01   C   G   \n",
       "157276           1  248914573  JHU_1.249208771  204958250147_R03C01   C   T   \n",
       "157277           1  248916508        rs6704311  204958250147_R03C01   G   A   \n",
       "157302           1  248918526  JHU_1.249212724  204958250147_R03C01   T   C   \n",
       "\n",
       "        ALLELE_A  ALLELE_B  BAlleleFreq  LogRRatio         R     Theta  \\\n",
       "0              0         1     0.033875  -0.016783  1.691960  0.050030   \n",
       "21             1         0     1.001940  -0.111973  1.155930  0.962306   \n",
       "27             1         0     0.490758   0.097902  0.764211  0.708393   \n",
       "32             1         0     1.005320  -0.112962  1.098080  0.957153   \n",
       "40             1         0     0.515637  -0.068786  1.557300  0.622296   \n",
       "...          ...       ...          ...        ...       ...       ...   \n",
       "157272         1         0     0.985136  -0.042008  0.977678  0.923677   \n",
       "157275         1         0     0.997110  -0.111099  1.261540  0.977864   \n",
       "157276         1         0     0.984007   0.041096  1.420910  0.949008   \n",
       "157277         1         0     0.991263  -0.027500  1.276190  0.963003   \n",
       "157302         0         1     0.001921   0.067395  1.445640  0.026349   \n",
       "\n",
       "        GenTrain_Score GType Alt1 Alt2  GT a1 a2       maf  \n",
       "0             0.699477    AA    G  NaN  AA  A  G  0.142857  \n",
       "21            0.832417    AA    T  NaN  BB  T  C  0.000000  \n",
       "27            0.443799    AB    T  NaN  AB  T  C  0.214286  \n",
       "32            0.790478    AA    T  NaN  BB  T  C  0.083333  \n",
       "40            0.540952    AB    A  NaN  AB  A  G  0.357143  \n",
       "...                ...   ...  ...  ...  .. .. ..       ...  \n",
       "157272        0.832624    AA    A  NaN  BB  A  G  0.142857  \n",
       "157275        0.497399    AA    G  NaN  BB  G  C  0.142857  \n",
       "157276        0.755758    AA    T  NaN  BB  T  C  0.000000  \n",
       "157277        0.777869    AA    A  NaN  BB  A  G  0.071429  \n",
       "157302        0.817646    AA    C  NaN  AA  T  C  0.000000  \n",
       "\n",
       "[59572 rows x 20 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "d716faf8-48f8-4a74-9c90-8046a49bbb5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "with open(f'{swarm_scripts_dir}/cnvs.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        out_name = f'GP2_round3_NEW_MERGE_{label}'\n",
    "        geno_path = f'{cnv_path}/{out_name}_clean'\n",
    "        bim = f'{geno_path}.bim'\n",
    "        samples = pd.read_csv(f'{geno_path}_barcode.samples', header=None, names=['IID'])\n",
    "        \n",
    "        for sample in samples.IID.unique():\n",
    "            code = sample.split('_')[0]\n",
    "            for chrom in chroms:\n",
    "            \n",
    "                mfile = f'{snp_metrics_path}/{code}/snp_metrics_{sample}_chr{chrom}.csv'\n",
    "\n",
    "                cnv_out = f'{cnv_path}/CNV_{label}_{sample}_chr{chrom}.csv'\n",
    "                intervals = f'/data/CARD/PD/GP2/ref_panel/glist_hg38_chr{chrom}.csv'\n",
    "\n",
    "                cmd = f'\\\n",
    "python run_cnv_pipeline.py \\\n",
    "--metrics {mfile} \\\n",
    "--bim {bim} \\\n",
    "--out_path {cnv_out} \\\n",
    "--intervals {intervals} \\\n",
    "--min_variants 10 \\\n",
    "--kb_window 250'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "fa7199d9-3b87-4d27-832a-28f98b0dd800",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "52322078\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnvs.swarm -g 4 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --module python/3.8 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "33352dbb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink is found\n",
      "Plink2 is found\n"
     ]
    }
   ],
   "source": [
    "!python3 run_cnv_pipeline.py --metrics /data/CARD/PD/GP2/raw_genotypes/GP2_snp_metrics/204958250147/snp_metrics_204958250147_R03C01_chr1.csv --bim /data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_EUR_clean.bim --out_path /data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_204958250147_R03C01_chr1.csv --intervals /data/CARD/PD/GP2/ref_panel/glist_hg38_chr1.csv --min_variants 10 --kb_window 250\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "75f31342-5241-48f9-bdf5-e73b1c46798d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# write dosages per chromosome per ancestry per cnv-type\n",
    "\n",
    "with open(f'{swarm_scripts_dir}/cnv_dosages.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        for chrom in chroms:\n",
    "\n",
    "            cnv_files_list = glob.glob(f'{cnv_path}/CNV_{label}_*_chr{chrom}.csv')\n",
    "            cnv_files_df = pd.DataFrame({'filename':cnv_files_list})\n",
    "            cnv_files_df.to_csv(f'{cnv_path}/CNV_{label}_chr{chrom}_files.csv', header=False, index=False)\n",
    "\n",
    "            for cnv_type in cnv_types:\n",
    "                \n",
    "                cmd = f'\\\n",
    "python run_cnv_dosage_pipeline.py \\\n",
    "--files {cnv_path}/CNV_{label}_chr{chrom}_files.csv \\\n",
    "--label {label} \\\n",
    "--chrom {chrom} \\\n",
    "--cnv_type {cnv_type} \\\n",
    "--out_path {cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n",
    "            \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5bce1bd2-a140-47d7-93cb-f6515adea344",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50839144\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_dosages.swarm -g 32 -t 16 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e88e7077",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_204958250147_R03C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_206391680116_R04C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_206046190127_R01C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_206046190102_R06C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_205814910049_R02C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_206046190068_R02C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_204835450146_R04C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_205053530113_R03C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_206430130072_R04C01_chr1.csv\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_206451070033_R04C01_chr1.csv\n"
     ]
    }
   ],
   "source": [
    "!head {cnv_path}/CNV_EUR_chr1_files.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "952c1e38",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/CARD/PD/GP2/raw_genotypes/GP2_plink/204958250147_R03C01.bed\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_plink/204958250147_R03C01.bim\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_plink/204958250147_R03C01.fam\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_plink/204958250147_R03C01.hh\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_plink/204958250147_R03C01.log\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_plink/204958250147_R03C01.map\n",
      "/data/CARD/PD/GP2/raw_genotypes/GP2_plink/204958250147_R03C01.ped\n"
     ]
    }
   ],
   "source": [
    "!ls /data/CARD/PD/GP2/raw_genotypes/GP2_plink/*204958250147_R03C01*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "c273d2fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "448.7984496124031"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# cnv_dosage = pd.read_csv('/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_204958250147_R03C01_chr1.csv')\n",
    "# cnv_dosage[cnv_dosage.NUM_VARIANTS>250]\n",
    "cnv_dosage.NUM_VARIANTS.std()\n",
    "cnv_dosage.NUM_VARIANTS.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "d73db50e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>INTERVAL</th>\n",
       "      <th>NUM_VARIANTS</th>\n",
       "      <th>PERCENT_BAF_INSERTION</th>\n",
       "      <th>PERCENT_L2R_DELETION</th>\n",
       "      <th>PERCENT_L2R_DUPLICATION</th>\n",
       "      <th>START_PLUS_WINDOW</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>STOP_PLUS_WINDOW</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>357</td>\n",
       "      <td>0.005602</td>\n",
       "      <td>0.039216</td>\n",
       "      <td>0.075630</td>\n",
       "      <td>33056765</td>\n",
       "      <td>33306765</td>\n",
       "      <td>33321098</td>\n",
       "      <td>33571098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AADACL3</td>\n",
       "      <td>351</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.034188</td>\n",
       "      <td>0.079772</td>\n",
       "      <td>12466114</td>\n",
       "      <td>12716114</td>\n",
       "      <td>12728759</td>\n",
       "      <td>12978759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AADACL4</td>\n",
       "      <td>376</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.031915</td>\n",
       "      <td>0.077128</td>\n",
       "      <td>12394546</td>\n",
       "      <td>12644546</td>\n",
       "      <td>12667086</td>\n",
       "      <td>12917086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA4</td>\n",
       "      <td>944</td>\n",
       "      <td>0.002119</td>\n",
       "      <td>0.049788</td>\n",
       "      <td>0.056144</td>\n",
       "      <td>93742837</td>\n",
       "      <td>93992837</td>\n",
       "      <td>94121149</td>\n",
       "      <td>94371149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCB10</td>\n",
       "      <td>579</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.022453</td>\n",
       "      <td>0.075993</td>\n",
       "      <td>229266581</td>\n",
       "      <td>229516581</td>\n",
       "      <td>229558695</td>\n",
       "      <td>229808695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2575</th>\n",
       "      <td>ZSCAN20</td>\n",
       "      <td>417</td>\n",
       "      <td>0.004796</td>\n",
       "      <td>0.023981</td>\n",
       "      <td>0.067146</td>\n",
       "      <td>33222630</td>\n",
       "      <td>33472630</td>\n",
       "      <td>33496395</td>\n",
       "      <td>33746395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2576</th>\n",
       "      <td>ZSWIM5</td>\n",
       "      <td>811</td>\n",
       "      <td>0.003699</td>\n",
       "      <td>0.019729</td>\n",
       "      <td>0.081381</td>\n",
       "      <td>44766403</td>\n",
       "      <td>45016403</td>\n",
       "      <td>45206578</td>\n",
       "      <td>45456578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2577</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>363</td>\n",
       "      <td>0.002755</td>\n",
       "      <td>0.024793</td>\n",
       "      <td>0.093664</td>\n",
       "      <td>52592510</td>\n",
       "      <td>52842510</td>\n",
       "      <td>52894575</td>\n",
       "      <td>53144575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2578</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>290</td>\n",
       "      <td>0.003448</td>\n",
       "      <td>0.020690</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>52476458</td>\n",
       "      <td>52726458</td>\n",
       "      <td>52827341</td>\n",
       "      <td>53077341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2579</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>285</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.077193</td>\n",
       "      <td>77314504</td>\n",
       "      <td>77564504</td>\n",
       "      <td>77682658</td>\n",
       "      <td>77932658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2580 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     INTERVAL  NUM_VARIANTS  PERCENT_BAF_INSERTION  PERCENT_L2R_DELETION  \\\n",
       "0     A3GALT2           357               0.005602              0.039216   \n",
       "1     AADACL3           351               0.002849              0.034188   \n",
       "2     AADACL4           376               0.002660              0.031915   \n",
       "3       ABCA4           944               0.002119              0.049788   \n",
       "4      ABCB10           579               0.000000              0.022453   \n",
       "...       ...           ...                    ...                   ...   \n",
       "2575  ZSCAN20           417               0.004796              0.023981   \n",
       "2576   ZSWIM5           811               0.003699              0.019729   \n",
       "2577   ZYG11A           363               0.002755              0.024793   \n",
       "2578   ZYG11B           290               0.003448              0.020690   \n",
       "2579     ZZZ3           285               0.000000              0.010526   \n",
       "\n",
       "      PERCENT_L2R_DUPLICATION  START_PLUS_WINDOW      START       STOP  \\\n",
       "0                    0.075630           33056765   33306765   33321098   \n",
       "1                    0.079772           12466114   12716114   12728759   \n",
       "2                    0.077128           12394546   12644546   12667086   \n",
       "3                    0.056144           93742837   93992837   94121149   \n",
       "4                    0.075993          229266581  229516581  229558695   \n",
       "...                       ...                ...        ...        ...   \n",
       "2575                 0.067146           33222630   33472630   33496395   \n",
       "2576                 0.081381           44766403   45016403   45206578   \n",
       "2577                 0.093664           52592510   52842510   52894575   \n",
       "2578                 0.100000           52476458   52726458   52827341   \n",
       "2579                 0.077193           77314504   77564504   77682658   \n",
       "\n",
       "      STOP_PLUS_WINDOW  \n",
       "0             33571098  \n",
       "1             12978759  \n",
       "2             12917086  \n",
       "3             94371149  \n",
       "4            229808695  \n",
       "...                ...  \n",
       "2575          33746395  \n",
       "2576          45456578  \n",
       "2577          53144575  \n",
       "2578          53077341  \n",
       "2579          77932658  \n",
       "\n",
       "[2580 rows x 9 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_dosage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a92f1c77",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_dosage[cnv_dosage]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "df9063fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "32.01899224806201"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "82609/2580"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8eaefe81-9ec6-4179-9f2c-776ba966b8ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/update_cnv_ids.swarm','w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "\n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}.csv'\n",
    "                dosagefile_out = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                pheno_out = f'{cnv_path}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "                cmd = f'\\\n",
    "python update_cnv_ids.py \\\n",
    "--dosagefile {dosagefile} \\\n",
    "--key /data/CARD/PD/GP2/clinical/key_merge/GP2_merge/release3_key.csv \\\n",
    "--pheno_out {pheno_out} \\\n",
    "--out_path {dosagefile_out}'\n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "051dc35b-f2f2-4eb8-877a-a720e822fb04",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50843587\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/update_cnv_ids.swarm -g 8 -t 8 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3b3a4dec-9b45-40ac-a1bc-0c51090c2819",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv_out_path = '/data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs'\n",
    "with open(f'{swarm_scripts_dir}/cnvwas.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "\n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                pheno = f'{cnv_path}/GP2_{label}_chr{chrom}_{cnv_type}.pheno'\n",
    "                covar = f'{cnv_path}/GP2_round3_NEW_MERGE_{label}_clean.cov'\n",
    "                out_path = f'{cnv_out_path}/GP2_{label}_chr{chrom}_{cnv_type}_cnvwas.tab'\n",
    "\n",
    "                cmd = f'\\\n",
    "python run_cnvwas_pipeline.py \\\n",
    "--cnv_dosage_file {dosagefile} \\\n",
    "--out_path {out_path} \\\n",
    "--pheno {pheno} \\\n",
    "--covar {covar}'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "9098b284-30f8-4c79-9944-9dc8da375090",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50860098\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnvwas.swarm -g 32 -t 16 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6081e4d8-a887-412f-98b2-77240260d7b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# output final release samples\n",
    "cnv_out_path = '/data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs'\n",
    "release3_samples = '/data/CARD/PD/GP2/genotypes/GP2/round3/clean/release3.samples'\n",
    "release_key_final = pd.read_csv('/data/CARD/PD/GP2/genotypes/GP2/round3/clean/release3_master_key.csv')\n",
    "release_key_out = release_key_final.loc[release_key_final.pheno!=-9]\n",
    "release_key_out['GP2sampleID'].to_csv(release3_samples, index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "82dc85bb-6d29-405e-bed6-efab5c1cc434",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{swarm_scripts_dir}/cnv_dosage_release_split.swarm', 'w') as f:\n",
    "    for label in ancestry_labels:\n",
    "        sample_file = f'/data/CARD/PD/GP2/genotypes/GP2/round3/clean/{label}_release3.samples'\n",
    "        label_samples = release_key_final.loc[release_key_final['label']==label]\n",
    "        label_samples['GP2sampleID'].to_csv(sample_file, header=False, index=False)\n",
    "\n",
    "        for cnv_type in cnv_types:\n",
    "            for chrom in chroms:\n",
    "                \n",
    "                dosagefile = f'{cnv_path}/CNV_{label}_chr{chrom}_{cnv_type}_gp2ids.csv'\n",
    "                dosage_out = f'{cnv_out_path}/CNV_{label}_chr{chrom}_{cnv_type}_release3.csv'\n",
    "                \n",
    "                cmd = f'\\\n",
    "python split_release_samples.py \\\n",
    "--cnv_dosage_file {dosagefile} \\\n",
    "--samples {sample_file} \\\n",
    "--out_path {dosage_out}'\n",
    "            \n",
    "                f.write(f'{cmd}\\n')\n",
    "f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "56353619-d819-4347-ab81-aec8d05de647",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "51651310\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/cnv_dosage_release_split.swarm -g 8 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "64f9b5b4-92c2-430c-a9ec-b09aae6aab6b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AMR PERCENT_BAF_INSERTION\n",
      "412\n",
      "\n",
      "AMR PERCENT_L2R_DELETION\n",
      "412\n",
      "\n",
      "AMR PERCENT_L2R_DUPLICATION\n",
      "412\n",
      "\n",
      "EUR PERCENT_BAF_INSERTION\n",
      "10727\n",
      "\n",
      "EUR PERCENT_L2R_DELETION\n",
      "10727\n",
      "\n",
      "EUR PERCENT_L2R_DUPLICATION\n",
      "10727\n",
      "\n",
      "SAS PERCENT_BAF_INSERTION\n",
      "59\n",
      "\n",
      "SAS PERCENT_L2R_DELETION\n",
      "59\n",
      "\n",
      "SAS PERCENT_L2R_DUPLICATION\n",
      "59\n",
      "\n",
      "MDE PERCENT_BAF_INSERTION\n",
      "62\n",
      "\n",
      "MDE PERCENT_L2R_DELETION\n",
      "62\n",
      "\n",
      "MDE PERCENT_L2R_DUPLICATION\n",
      "62\n",
      "\n",
      "EAS PERCENT_BAF_INSERTION\n",
      "137\n",
      "\n",
      "EAS PERCENT_L2R_DELETION\n",
      "137\n",
      "\n",
      "EAS PERCENT_L2R_DUPLICATION\n",
      "137\n",
      "\n",
      "AAC PERCENT_BAF_INSERTION\n",
      "1215\n",
      "\n",
      "AAC PERCENT_L2R_DELETION\n",
      "1215\n",
      "\n",
      "AAC PERCENT_L2R_DUPLICATION\n",
      "1215\n",
      "\n",
      "AFR PERCENT_BAF_INSERTION\n",
      "404\n",
      "\n",
      "AFR PERCENT_L2R_DELETION\n",
      "404\n",
      "\n",
      "AFR PERCENT_L2R_DUPLICATION\n",
      "404\n",
      "\n",
      "CAS PERCENT_BAF_INSERTION\n",
      "303\n",
      "\n",
      "CAS PERCENT_L2R_DELETION\n",
      "303\n",
      "\n",
      "CAS PERCENT_L2R_DUPLICATION\n",
      "303\n",
      "\n",
      "AJ PERCENT_BAF_INSERTION\n",
      "923\n",
      "\n",
      "AJ PERCENT_L2R_DELETION\n",
      "923\n",
      "\n",
      "AJ PERCENT_L2R_DUPLICATION\n",
      "923\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# CHECK COUNTS\n",
    "total_dosages = 0\n",
    "for label in ancestry_labels:\n",
    "    for cnv_type in cnv_types:\n",
    "        dosage_out = f'{cnv_out_path}/CNV_{label}_chr1_{cnv_type}_release3.csv'\n",
    "        print(label, cnv_type)\n",
    "        label_count = !cat {dosage_out} | wc -l\n",
    "        label_count = int(label_count[0])\n",
    "        total_dosages += label_count\n",
    "        print(label_count)\n",
    "        print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3df93e96-b169-4e45-ad32-76904be6cf22",
   "metadata": {},
   "outputs": [],
   "source": [
    "# push to release bucket\n",
    "with open(f'{swarm_scripts_dir}/gcp_push_cnvs.swarm','w') as f:\n",
    "    \n",
    "    for label in ancestry_labels:\n",
    "        for chrom in chroms:\n",
    "    \n",
    "#         !module load google-cloud-sdk/397.0.0; gsutil cp {cnv_out_path}/CNV_{label}_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/\n",
    "            cp1 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "            cp2 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_L2R_DELETION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "            cp3 = f'gsutil cp {cnv_out_path}/CNV_{label}_chr{chrom}_PERCENT_L2R_DUPLICATION_release3.csv gs://gp2tier2/release3_31102022/cnvs/{label}/'\n",
    "\n",
    "            cps = [cp1, cp2, cp3]\n",
    "\n",
    "            for cp in cps:\n",
    "                f.write(f'{cp}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "ddb097af-919f-41f3-a434-2b624c4dbde4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "51673863\n"
     ]
    }
   ],
   "source": [
    "!swarm -f {swarm_scripts_dir}/gcp_push_cnvs.swarm -g 8 -t 8 --time=00:30:00 --logdir {swarm_scripts_dir}/logs --module google-cloud-sdk/397.0.0 --gres=lscratch:20 --partition=norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "70c7c14a-cb24-4e02-961f-f4fb69834166",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "594\n"
     ]
    }
   ],
   "source": [
    "!module load google-cloud-sdk/397.0.0; gsutil cp /data/CARD/PD/GP2/genotypes/GP2/round3/GP2_cnvs/CNV_AMR_chr1_BAF_INSERTION_release3.csv gs://gp2tier2/release3_31102022/cnvs/AMR/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "2a133f0b-dc2b-4bce-91d5-5040ca024786",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27.0"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1ede61fd-453f-47d1-8000-0bbbabb6331f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleid</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>AADACL3</th>\n",
       "      <th>AADACL4</th>\n",
       "      <th>ABCA4</th>\n",
       "      <th>ABCB10</th>\n",
       "      <th>ABCD3</th>\n",
       "      <th>ABL2</th>\n",
       "      <th>ACADM</th>\n",
       "      <th>ACAP3</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNHIT6</th>\n",
       "      <th>ZP4</th>\n",
       "      <th>ZRANB2</th>\n",
       "      <th>ZRANB2_AS1</th>\n",
       "      <th>ZRANB2_AS2</th>\n",
       "      <th>ZSCAN20</th>\n",
       "      <th>ZSWIM5</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CORIELL_000178_s1</td>\n",
       "      <td>0.005602</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.002119</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.003546</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004796</td>\n",
       "      <td>0.003699</td>\n",
       "      <td>0.002755</td>\n",
       "      <td>0.003448</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PAGE_000432_s1</td>\n",
       "      <td>0.008403</td>\n",
       "      <td>0.008547</td>\n",
       "      <td>0.007979</td>\n",
       "      <td>0.003178</td>\n",
       "      <td>0.039724</td>\n",
       "      <td>0.007653</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.144208</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003311</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014925</td>\n",
       "      <td>0.014634</td>\n",
       "      <td>0.011538</td>\n",
       "      <td>0.004796</td>\n",
       "      <td>0.018496</td>\n",
       "      <td>0.011019</td>\n",
       "      <td>0.013793</td>\n",
       "      <td>0.010526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CORIELL_000783_s1</td>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.001727</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003311</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002398</td>\n",
       "      <td>0.001233</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CORIELL_004856_s1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.005181</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CORIELL_007135_s1</td>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003454</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003546</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002466</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11690</th>\n",
       "      <td>PAGE_001054_s1</td>\n",
       "      <td>0.005602</td>\n",
       "      <td>0.008547</td>\n",
       "      <td>0.007979</td>\n",
       "      <td>0.003178</td>\n",
       "      <td>0.012090</td>\n",
       "      <td>0.007653</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.076832</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007692</td>\n",
       "      <td>0.002398</td>\n",
       "      <td>0.008631</td>\n",
       "      <td>0.013774</td>\n",
       "      <td>0.013793</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11691</th>\n",
       "      <td>LCC_000146_s1</td>\n",
       "      <td>0.008403</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.010363</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008274</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004796</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11692</th>\n",
       "      <td>CORIELL_002459_s1</td>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.008636</td>\n",
       "      <td>0.005102</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006623</td>\n",
       "      <td>0.001447</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002398</td>\n",
       "      <td>0.004932</td>\n",
       "      <td>0.002755</td>\n",
       "      <td>0.003448</td>\n",
       "      <td>0.014035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11693</th>\n",
       "      <td>BCM_000036_s1</td>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001233</td>\n",
       "      <td>0.005510</td>\n",
       "      <td>0.006897</td>\n",
       "      <td>0.007018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11694</th>\n",
       "      <td>CORIELL_001134_s1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002762</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002755</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11695 rows × 2554 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                sampleid   A3GALT2   AADACL3   AADACL4     ABCA4    ABCB10  \\\n",
       "0      CORIELL_000178_s1  0.005602  0.002849  0.002660  0.002119  0.000000   \n",
       "1         PAGE_000432_s1  0.008403  0.008547  0.007979  0.003178  0.039724   \n",
       "2      CORIELL_000783_s1  0.002801  0.000000  0.000000  0.001059  0.001727   \n",
       "3      CORIELL_004856_s1  0.000000  0.002849  0.002660  0.001059  0.005181   \n",
       "4      CORIELL_007135_s1  0.002801  0.000000  0.000000  0.000000  0.003454   \n",
       "...                  ...       ...       ...       ...       ...       ...   \n",
       "11690     PAGE_001054_s1  0.005602  0.008547  0.007979  0.003178  0.012090   \n",
       "11691      LCC_000146_s1  0.008403  0.002849  0.002660  0.001059  0.010363   \n",
       "11692  CORIELL_002459_s1  0.002801  0.002849  0.002660  0.001059  0.008636   \n",
       "11693      BCM_000036_s1  0.002801  0.000000  0.000000  0.000000  0.000000   \n",
       "11694  CORIELL_001134_s1  0.000000  0.002849  0.002660  0.000000  0.000000   \n",
       "\n",
       "          ABCD3      ABL2     ACADM     ACAP3  ...    ZNHIT6       ZP4  \\\n",
       "0      0.000000  0.000000  0.001992  0.003546  ...  0.000000  0.000000   \n",
       "1      0.007653  0.000000  0.001992  0.144208  ...  0.003311  0.000000   \n",
       "2      0.002551  0.000000  0.000000  0.004728  ...  0.003311  0.000000   \n",
       "3      0.000000  0.000000  0.000000  0.000000  ...  0.000000  0.000000   \n",
       "4      0.002551  0.000000  0.000000  0.003546  ...  0.000000  0.000000   \n",
       "...         ...       ...       ...       ...  ...       ...       ...   \n",
       "11690  0.007653  0.000000  0.000000  0.076832  ...  0.000000  0.000000   \n",
       "11691  0.002551  0.000000  0.000000  0.008274  ...  0.000000  0.000000   \n",
       "11692  0.005102  0.000000  0.001992  0.004728  ...  0.006623  0.001447   \n",
       "11693  0.002551  0.000000  0.001992  0.000000  ...  0.000000  0.000000   \n",
       "11694  0.000000  0.002762  0.001992  0.000000  ...  0.000000  0.000000   \n",
       "\n",
       "         ZRANB2  ZRANB2_AS1  ZRANB2_AS2   ZSCAN20    ZSWIM5    ZYG11A  \\\n",
       "0      0.000000    0.000000    0.000000  0.004796  0.003699  0.002755   \n",
       "1      0.014925    0.014634    0.011538  0.004796  0.018496  0.011019   \n",
       "2      0.000000    0.000000    0.000000  0.002398  0.001233  0.000000   \n",
       "3      0.000000    0.000000    0.000000  0.000000  0.000000  0.000000   \n",
       "4      0.000000    0.000000    0.000000  0.000000  0.002466  0.000000   \n",
       "...         ...         ...         ...       ...       ...       ...   \n",
       "11690  0.000000    0.000000    0.007692  0.002398  0.008631  0.013774   \n",
       "11691  0.000000    0.000000    0.000000  0.004796  0.000000  0.000000   \n",
       "11692  0.000000    0.000000    0.000000  0.002398  0.004932  0.002755   \n",
       "11693  0.000000    0.000000    0.000000  0.000000  0.001233  0.005510   \n",
       "11694  0.000000    0.000000    0.000000  0.000000  0.000000  0.002755   \n",
       "\n",
       "         ZYG11B      ZZZ3  \n",
       "0      0.003448  0.000000  \n",
       "1      0.013793  0.010526  \n",
       "2      0.000000  0.003509  \n",
       "3      0.000000  0.000000  \n",
       "4      0.000000  0.000000  \n",
       "...         ...       ...  \n",
       "11690  0.013793  0.000000  \n",
       "11691  0.000000  0.000000  \n",
       "11692  0.003448  0.014035  \n",
       "11693  0.006897  0.007018  \n",
       "11694  0.000000  0.000000  \n",
       "\n",
       "[11695 rows x 2554 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dosage_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9a5ba71-d37b-449b-9c98-1ab8b39db4bc",
   "metadata": {},
   "source": [
    "# testing for cnvwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b29bf2ed-e75c-4d62-9a54-73dad0cd195f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# test = pd.read_csv('/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_chr1_PERCENT_BAF_INSERTION_gp2ids.csv')\n",
    "# test\n",
    "cnv_dosage_file = '/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_EUR_chr1_PERCENT_BAF_INSERTION_gp2ids.csv'\n",
    "pheno = '/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_EUR_chr1_PERCENT_BAF_INSERTION.pheno'\n",
    "covar = '/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_EUR_clean.cov'\n",
    "\n",
    "scaler = MinMaxScaler()\n",
    "dosage_df = pd.read_csv(cnv_dosage_file)\n",
    "dosage_df.dropna(axis=1, inplace=True)\n",
    "#fix column names\n",
    "dosage_df.columns = [x.replace('-','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace('.','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace(' ','_') for x in dosage_df.columns]\n",
    "\n",
    "pheno_df = pd.read_csv(pheno, sep='\\t')\n",
    "covar_df = pd.read_csv(covar, sep='\\t')\n",
    "\n",
    "\n",
    "if covar_df.age_of_onset.isna().all():\n",
    "    covar_df.drop(columns=['age_of_onset'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age_of_onset'] = scaler.fit_transform(covar_df[['age_of_onset']])\n",
    "\n",
    "if covar_df.age.isna().all():\n",
    "    covar_df.drop(columns=['age'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age'] = scaler.fit_transform(covar_df[['age']])\n",
    "\n",
    "if covar_df.sex.isna().all():\n",
    "    covar_df.drop(columns=['sex'], inplace=True)\n",
    "\n",
    "covar_df.drop(columns=['FID'], inplace=True)\n",
    "covar_df.rename(columns={'GP2sampleID':'sampleid'}, inplace=True)\n",
    "\n",
    "data_df = dosage_df.merge(covar_df, on='sampleid', how='left').merge(pheno_df, on='sampleid', how='left').set_index('sampleid')\n",
    "data_df.dropna(axis=0, inplace=True)\n",
    "rm_pred = [f'PC{i}' for i in range(1,21)] + ['sex','age_of_onset','age','pheno']\n",
    "\n",
    "pred_list = [x for x in data_df.columns if x not in rm_pred]\n",
    "covars_list = [x for x in data_df.columns if x not in pred_list + [f'PC{i}' for i in range(11,21)] + ['pheno']]\n",
    "\n",
    "results = []\n",
    "fails = []\n",
    "\n",
    "for pred in range(len(pred_list)):\n",
    "    pred_name = pred_list[pred]\n",
    "    \n",
    "    mean_ = data_df.loc[data_df['pheno']==0 ,pred_name].mean()\n",
    "    std_ = data_df.loc[data_df['pheno']==0 ,pred_name].std()\n",
    "    # mean_ = data_df.loc[: ,pred_name].mean()\n",
    "    # std_ = data_df.loc[: ,pred_name].std()\n",
    "    recoded_df = data_df.copy()\n",
    "    recoded_df.loc[:, pred_name] = np.where((data_df[pred_name]>mean_+(2*std_)) | (data_df[pred_name]<mean_-(2*std_)), 1, 0)\n",
    "#     recoded_df.loc[:, pred_name] = np.where((data_df[pred_name]>control_mean+(2*control_std)) | (data_df[pred_name]<control_mean-(2*control_std)), 1, 0)\n",
    "    pred_ = [pred_name] + covars_list\n",
    "    X = recoded_df.loc[:, pred_]\n",
    "    y = recoded_df.loc[:,'pheno']\n",
    "\n",
    "#     formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "#     fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=recoded_df).fit()\n",
    "#     pred_ = [pred_name] + covars_list\n",
    "#     X = data_df.loc[:, pred_]\n",
    "#     y = data_df.loc[:,'pheno']\n",
    "    fitted = sm.OLS(y, X).fit()\n",
    "#     fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df).fit()\n",
    "    beta_coef  = fitted.params.loc[pred_name]\n",
    "    beta_se  = fitted.bse.loc[pred_name]\n",
    "    p_val = fitted.pvalues.loc[pred_name]\n",
    "\n",
    "    results.append((pred_name, beta_coef, beta_se, p_val))\n",
    "\n",
    "\n",
    "output = pd.DataFrame(results, columns=('PREDICTOR', 'BETA_COEF', 'BETA_SE','P_VAL'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1428363e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>AADACL3</th>\n",
       "      <th>AADACL4</th>\n",
       "      <th>ABCA4</th>\n",
       "      <th>ABCB10</th>\n",
       "      <th>ABCD3</th>\n",
       "      <th>ABL2</th>\n",
       "      <th>ACADM</th>\n",
       "      <th>ACAP3</th>\n",
       "      <th>ACBD3</th>\n",
       "      <th>...</th>\n",
       "      <th>PC15</th>\n",
       "      <th>PC16</th>\n",
       "      <th>PC17</th>\n",
       "      <th>PC18</th>\n",
       "      <th>PC19</th>\n",
       "      <th>PC20</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>age_of_onset</th>\n",
       "      <th>pheno</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CORIELL_000178_s1</th>\n",
       "      <td>0.005602</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.002119</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.003546</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.010497</td>\n",
       "      <td>-0.001786</td>\n",
       "      <td>-0.000480</td>\n",
       "      <td>-0.015263</td>\n",
       "      <td>0.022669</td>\n",
       "      <td>-0.001714</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.637500</td>\n",
       "      <td>0.582418</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAGE_000432_s1</th>\n",
       "      <td>0.008403</td>\n",
       "      <td>0.008547</td>\n",
       "      <td>0.007979</td>\n",
       "      <td>0.003178</td>\n",
       "      <td>0.039724</td>\n",
       "      <td>0.007653</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.144208</td>\n",
       "      <td>0.033846</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001478</td>\n",
       "      <td>-0.007132</td>\n",
       "      <td>0.017460</td>\n",
       "      <td>0.014483</td>\n",
       "      <td>0.002081</td>\n",
       "      <td>-0.014048</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.761490</td>\n",
       "      <td>0.729853</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_000783_s1</th>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.001727</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.005627</td>\n",
       "      <td>-0.006498</td>\n",
       "      <td>-0.001086</td>\n",
       "      <td>-0.014374</td>\n",
       "      <td>-0.000299</td>\n",
       "      <td>0.005867</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.512500</td>\n",
       "      <td>0.560440</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_004856_s1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.005181</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.003527</td>\n",
       "      <td>0.015134</td>\n",
       "      <td>-0.002175</td>\n",
       "      <td>0.000688</td>\n",
       "      <td>-0.005928</td>\n",
       "      <td>0.018949</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.612500</td>\n",
       "      <td>0.552213</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_007135_s1</th>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003454</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003546</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001130</td>\n",
       "      <td>0.006795</td>\n",
       "      <td>0.004029</td>\n",
       "      <td>-0.005951</td>\n",
       "      <td>-0.007167</td>\n",
       "      <td>-0.005832</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.775000</td>\n",
       "      <td>0.552213</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAGE_001054_s1</th>\n",
       "      <td>0.005602</td>\n",
       "      <td>0.008547</td>\n",
       "      <td>0.007979</td>\n",
       "      <td>0.003178</td>\n",
       "      <td>0.012090</td>\n",
       "      <td>0.007653</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.076832</td>\n",
       "      <td>0.018462</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.004930</td>\n",
       "      <td>0.004625</td>\n",
       "      <td>-0.004943</td>\n",
       "      <td>0.008030</td>\n",
       "      <td>-0.008413</td>\n",
       "      <td>-0.006565</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.715358</td>\n",
       "      <td>0.638278</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LCC_000146_s1</th>\n",
       "      <td>0.008403</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.010363</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.002144</td>\n",
       "      <td>-0.002246</td>\n",
       "      <td>-0.006940</td>\n",
       "      <td>0.002406</td>\n",
       "      <td>0.005469</td>\n",
       "      <td>-0.010250</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.537500</td>\n",
       "      <td>0.552213</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_002459_s1</th>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>0.008636</td>\n",
       "      <td>0.005102</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.006154</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.010934</td>\n",
       "      <td>0.003952</td>\n",
       "      <td>0.001125</td>\n",
       "      <td>0.011962</td>\n",
       "      <td>0.006304</td>\n",
       "      <td>0.015792</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.525000</td>\n",
       "      <td>0.461538</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCM_000036_s1</th>\n",
       "      <td>0.002801</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003077</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003476</td>\n",
       "      <td>-0.011146</td>\n",
       "      <td>0.016683</td>\n",
       "      <td>0.011988</td>\n",
       "      <td>-0.000492</td>\n",
       "      <td>0.006304</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.556571</td>\n",
       "      <td>0.552213</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORIELL_001134_s1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002762</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.003415</td>\n",
       "      <td>-0.015989</td>\n",
       "      <td>-0.013623</td>\n",
       "      <td>0.018942</td>\n",
       "      <td>-0.020538</td>\n",
       "      <td>0.007129</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.287500</td>\n",
       "      <td>0.351648</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11685 rows × 2577 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    A3GALT2   AADACL3   AADACL4     ABCA4    ABCB10     ABCD3  \\\n",
       "sampleid                                                                        \n",
       "CORIELL_000178_s1  0.005602  0.002849  0.002660  0.002119  0.000000  0.000000   \n",
       "PAGE_000432_s1     0.008403  0.008547  0.007979  0.003178  0.039724  0.007653   \n",
       "CORIELL_000783_s1  0.002801  0.000000  0.000000  0.001059  0.001727  0.002551   \n",
       "CORIELL_004856_s1  0.000000  0.002849  0.002660  0.001059  0.005181  0.000000   \n",
       "CORIELL_007135_s1  0.002801  0.000000  0.000000  0.000000  0.003454  0.002551   \n",
       "...                     ...       ...       ...       ...       ...       ...   \n",
       "PAGE_001054_s1     0.005602  0.008547  0.007979  0.003178  0.012090  0.007653   \n",
       "LCC_000146_s1      0.008403  0.002849  0.002660  0.001059  0.010363  0.002551   \n",
       "CORIELL_002459_s1  0.002801  0.002849  0.002660  0.001059  0.008636  0.005102   \n",
       "BCM_000036_s1      0.002801  0.000000  0.000000  0.000000  0.000000  0.002551   \n",
       "CORIELL_001134_s1  0.000000  0.002849  0.002660  0.000000  0.000000  0.000000   \n",
       "\n",
       "                       ABL2     ACADM     ACAP3     ACBD3  ...      PC15  \\\n",
       "sampleid                                                   ...             \n",
       "CORIELL_000178_s1  0.000000  0.001992  0.003546  0.000000  ...  0.010497   \n",
       "PAGE_000432_s1     0.000000  0.001992  0.144208  0.033846  ...  0.001478   \n",
       "CORIELL_000783_s1  0.000000  0.000000  0.004728  0.000000  ... -0.005627   \n",
       "CORIELL_004856_s1  0.000000  0.000000  0.000000  0.000000  ... -0.003527   \n",
       "CORIELL_007135_s1  0.000000  0.000000  0.003546  0.000000  ...  0.001130   \n",
       "...                     ...       ...       ...       ...  ...       ...   \n",
       "PAGE_001054_s1     0.000000  0.000000  0.076832  0.018462  ... -0.004930   \n",
       "LCC_000146_s1      0.000000  0.000000  0.008274  0.000000  ... -0.002144   \n",
       "CORIELL_002459_s1  0.000000  0.001992  0.004728  0.006154  ... -0.010934   \n",
       "BCM_000036_s1      0.000000  0.001992  0.000000  0.003077  ...  0.003476   \n",
       "CORIELL_001134_s1  0.002762  0.001992  0.000000  0.000000  ... -0.003415   \n",
       "\n",
       "                       PC16      PC17      PC18      PC19      PC20  sex  \\\n",
       "sampleid                                                                   \n",
       "CORIELL_000178_s1 -0.001786 -0.000480 -0.015263  0.022669 -0.001714  1.0   \n",
       "PAGE_000432_s1    -0.007132  0.017460  0.014483  0.002081 -0.014048  1.0   \n",
       "CORIELL_000783_s1 -0.006498 -0.001086 -0.014374 -0.000299  0.005867  2.0   \n",
       "CORIELL_004856_s1  0.015134 -0.002175  0.000688 -0.005928  0.018949  1.0   \n",
       "CORIELL_007135_s1  0.006795  0.004029 -0.005951 -0.007167 -0.005832  2.0   \n",
       "...                     ...       ...       ...       ...       ...  ...   \n",
       "PAGE_001054_s1     0.004625 -0.004943  0.008030 -0.008413 -0.006565  1.0   \n",
       "LCC_000146_s1     -0.002246 -0.006940  0.002406  0.005469 -0.010250  1.0   \n",
       "CORIELL_002459_s1  0.003952  0.001125  0.011962  0.006304  0.015792  1.0   \n",
       "BCM_000036_s1     -0.011146  0.016683  0.011988 -0.000492  0.006304  2.0   \n",
       "CORIELL_001134_s1 -0.015989 -0.013623  0.018942 -0.020538  0.007129  1.0   \n",
       "\n",
       "                        age  age_of_onset  pheno  \n",
       "sampleid                                          \n",
       "CORIELL_000178_s1  0.637500      0.582418      1  \n",
       "PAGE_000432_s1     0.761490      0.729853      1  \n",
       "CORIELL_000783_s1  0.512500      0.560440      1  \n",
       "CORIELL_004856_s1  0.612500      0.552213      0  \n",
       "CORIELL_007135_s1  0.775000      0.552213      0  \n",
       "...                     ...           ...    ...  \n",
       "PAGE_001054_s1     0.715358      0.638278      1  \n",
       "LCC_000146_s1      0.537500      0.552213      0  \n",
       "CORIELL_002459_s1  0.525000      0.461538      1  \n",
       "BCM_000036_s1      0.556571      0.552213      1  \n",
       "CORIELL_001134_s1  0.287500      0.351648      1  \n",
       "\n",
       "[11685 rows x 2577 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6cf036e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "26bdc58c-f50d-4a33-baae-3179b680da95",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink is found\n",
      "Plink2 is found\n",
      "Plink is found\n",
      "Plink2 is found\n"
     ]
    }
   ],
   "source": [
    "from GWAS.gwas import calculate_inflation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "6f6809c9-4c00-42e3-895c-b5c434d228e5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUNNING: lambda_calculation\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'lambda_calculation',\n",
       " 'metrics': {'inflation': 30.206209829338817}}"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "calculate_inflation(output.P_VAL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "b085e022-75b7-4eda-acc1-c23112ef5da9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PREDICTOR        ZZZ3\n",
       "BETA_COEF    0.791209\n",
       "BETA_SE      0.117274\n",
       "P_VAL        0.000001\n",
       "dtype: object"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output.max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e9706cf-1f2e-4618-b7db-c6aa24faf737",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fee4434-8a00-4b0b-bd96-c76e65f41ab6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for label in labels:\n",
    "#     for dose in dosage_files:\n",
    "#         tab = f'{cnv_path}/{label}/GP2_round2_{label}_{dose}_gp2ids_cnvwas.tab'\n",
    "#         test_sumstat = pd.read_csv(tab, sep='\\t')\n",
    "#         n_hits = test_sumstat[test_sumstat.P_VAL<=0.05/25000].shape[0]\n",
    "#         print(f'{label}, {dose}:')\n",
    "#         print(f'{n_hits} hits, {calculate_inflation(test_sumstat.P_VAL)} lambda')\n",
    "#         print()\n",
    "#         print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "0c45feb3-03d2-4401-8e9d-aa0f45061a38",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import MinMaxScaler\n",
    "import statsmodels.api as sm\n",
    "scaler = MinMaxScaler()\n",
    "\n",
    "cnv_dosage_file = '/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/CNV_AMR_chr1_PERCENT_BAF_INSERTION_gp2ids.csv'\n",
    "pheno = '/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_AMR_chr1_PERCENT_BAF_INSERTION.pheno'\n",
    "covar = '/data/CARD/PD/GP2/raw_genotypes/GP2_cnvs/release3/GP2_round3_NEW_MERGE_AMR_clean.cov'\n",
    "dosage_df = pd.read_csv(cnv_dosage_file)\n",
    "#fix column names\n",
    "dosage_df.columns = [x.replace('-','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace('.','_') for x in dosage_df.columns]\n",
    "dosage_df.columns = [x.replace(' ','_') for x in dosage_df.columns]\n",
    "\n",
    "pheno_df = pd.read_csv(pheno, sep='\\t')\n",
    "covar_df = pd.read_csv(covar, sep='\\t')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "if covar_df.age_of_onset.isna().all():\n",
    "    covar_df.drop(columns=['age_of_onset'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age_of_onset'] = scaler.fit_transform(covar_df[['age_of_onset']])\n",
    "\n",
    "if covar_df.age.isna().all():\n",
    "    covar_df.drop(columns=['age'], inplace=True)\n",
    "else:\n",
    "    covar_df.loc[:,'age'] = scaler.fit_transform(covar_df[['age']])\n",
    "\n",
    "if covar_df.sex.isna().all():\n",
    "    covar_df.drop(columns=['sex'], inplace=True)\n",
    "\n",
    "covar_df.drop(columns=['FID'], inplace=True)\n",
    "covar_df.rename(columns={'GP2sampleID':'sampleid'}, inplace=True)\n",
    "\n",
    "data_df = dosage_df.merge(covar_df, on='sampleid', how='left').merge(pheno_df, on='sampleid', how='left').set_index('sampleid')\n",
    "\n",
    "rm_pred = [f'PC{i}' for i in range(1,21)] + ['sex','age_of_onset','age','pheno']\n",
    "\n",
    "pred_list = [x for x in data_df.columns if x not in rm_pred]\n",
    "covars_list = [x for x in data_df.columns if x not in pred_list + [f'PC{i}' for i in range(11,21)] + ['pheno']]\n",
    "\n",
    "results = []\n",
    "fails = []\n",
    "\n",
    "for pred in range(len(pred_list)):\n",
    "    pred_name = pred_list[pred]\n",
    "    formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "\n",
    "    fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df).fit()\n",
    "    beta_coef  = fitted.params.loc[pred_name]\n",
    "    beta_se  = fitted.bse.loc[pred_name]\n",
    "    p_val = fitted.pvalues.loc[pred_name]\n",
    "\n",
    "    results.append((pred_name, beta_coef, beta_se, p_val))\n",
    "\n",
    "\n",
    "output = pd.DataFrame(results, columns=('PREDICTOR', 'BETA_COEF', 'BETA_SE','P_VAL'))\n",
    "# output.to_csv(out_path, sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bc9c4281-aeca-40a1-9c0d-b68be449f6d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "pred_name = 'ADAMTSL4-AS1'\n",
    "formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df).fit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b377fdf2-257d-4ed3-996a-593e04b55a62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PREDICTOR</th>\n",
       "      <th>BETA_COEF</th>\n",
       "      <th>BETA_SE</th>\n",
       "      <th>P_VAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [PREDICTOR, BETA_COEF, BETA_SE, P_VAL]\n",
       "Index: []"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output[output.P_VAL<=0.05/output.shape[0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "3b60b8cd-37dc-40b4-831c-4b5ebd6a8741",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.9546520719311964e-05"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "0.05/output.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "415a2046-81a1-44a6-be36-15752ba33199",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['sampleid', 'A3GALT2', 'AADACL3', 'AADACL4', 'ABCA4', 'ABCB10', 'ABCD3',\n",
       "       'ABL2', 'ACADM', 'ACAP3',\n",
       "       ...\n",
       "       'ZNHIT6', 'ZP4', 'ZRANB2', 'ZRANB2_AS1', 'ZRANB2_AS2', 'ZSCAN20',\n",
       "       'ZSWIM5', 'ZYG11A', 'ZYG11B', 'ZZZ3'],\n",
       "      dtype='object', length=2559)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dosage_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "995fdab8-a4bb-4847-9a30-d35aa957a052",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FID\tsampleid\tPC1\tPC2\tPC3\tPC4\tPC5\tPC6\tPC7\tPC8\tPC9\tPC10\tPC11\tPC12\tPC13\tPC14\tPC15\tPC16\tPC17\tPC18\tPC19\tPC20\tsex\tage\tage_of_onset\n",
      "0\tAPGS_000002_s1\t0.0055697\t5.58512e-05\t0.00699746\t-0.00260114\t-0.0122619\t-0.010135\t0.00431493\t-0.012719\t-0.00293343\t0.00668866\t0.0053588\t0.008123\t-0.000890374\t-0.0141451\t0.0119144\t0.000390307\t-0.00671753\t-0.000866958\t-0.0013909\t0.00290549\t1\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000003_s1\t0.00326122\t-0.00888383\t-0.00871568\t0.000830058\t-0.00808219\t-0.00634886\t0.00276938\t0.00309513\t-0.00118999\t-0.00757583\t0.00328179\t0.00329943\t0.0057767\t-0.0128158\t0.0133359\t0.000640707\t-0.00633914\t0.0100417\t0.00191561\t0.0077697\t1\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000005_s1\t0.000738409\t-0.00199674\t0.00794567\t0.00732722\t0.0111577\t0.00256656\t0.00058314\t-0.00580364\t-0.00200246\t-0.00346736\t0.0119227\t0.00547475\t0.00136185\t-0.0146136\t0.00129355\t-0.00678011\t-0.00567244\t0.0212485\t-0.00767652\t-0.00502232\t1\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000006_s1\t0.00713627\t0.0010006\t-0.0110152\t0.00725338\t-0.00556761\t-0.00353322\t0.00169541\t0.00356565\t-0.00750236\t0.00309578\t0.00304494\t-0.000424422\t-0.00590244\t0.00619993\t-0.00184106\t-0.00173787\t-0.00425422\t0.00255309\t0.00575037\t-0.0157719\t1\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000007_s1\t0.00729202\t-0.0123239\t-0.0265346\t0.00487111\t-0.00470293\t-0.00533713\t0.00318663\t-0.0047251\t-0.00766736\t0.00289025\t-0.0110145\t-0.00265537\t-0.00410958\t0.011642\t-0.00404926\t0.00685318\t-0.00564891\t0.0115163\t0.00266377\t0.00105195\t1\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000009_s1\t-0.00795051\t-0.000221271\t-0.0273655\t0.00370781\t-0.00243791\t0.00106283\t-0.00989402\t0.0117013\t-0.005206\t-0.00332964\t-0.0106294\t0.00975295\t0.0136149\t0.00301406\t-0.00418775\t-0.00853897\t-0.00115325\t-0.00368746\t-0.00456272\t-0.00806704\t1\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000010_s1\t0.00918211\t0.00967839\t-0.012686\t0.00273761\t-0.00973895\t-0.000671072\t-0.00358073\t0.00343093\t-0.0110792\t0.0011068\t0.00856866\t-0.00418448\t0.000157018\t0.0148575\t-0.002082\t0.0022195\t-0.00798496\t0.0152374\t-0.000357822\t0.000848431\t1\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000012_s1\t-0.00281887\t0.0164004\t-0.00259928\t-0.0048986\t0.00745725\t-0.0002599\t0.00847199\t0.00293515\t0.00685554\t-0.00129901\t-0.00431071\t-0.00393714\t0.00346845\t-0.000392658\t-0.0100082\t-0.00405609\t0.00349906\t-0.00355833\t0.0065031\t0.00192376\t2\t65.52570310775978\t57.25140654302988\n",
      "0\tAPGS_000013_s1\t0.00620733\t0.00255616\t-0.00171368\t-0.00905064\t-0.000418481\t0.00142644\t0.00329859\t0.000929818\t0.00372047\t-0.00813476\t-0.00295353\t-0.0165052\t0.00321408\t0.00279867\t0.00628431\t0.00235017\t-0.00695878\t0.000646445\t0.00218762\t0.00324519\t1\t65.52570310775978\t57.25140654302988\n"
     ]
    }
   ],
   "source": [
    "def CNV_WAS(cnv_dosage_file, pheno, covar, out_path):\n",
    "    scaler = MinMaxScaler()\n",
    "    dosage_df = pd.read_csv(cnv_dosage_file)\n",
    "    pheno_df = pd.read_csv(pheno, sep='\\t')\n",
    "    covar_df = pd.read_csv(covar, sep='\\t')\n",
    "\n",
    "\n",
    "    if covar_df.age_of_onset.isna().all():\n",
    "        covar_df.drop(columns=['age_of_onset'], inplace=True)\n",
    "    else:\n",
    "        covar_df.loc[:,'age_of_onset'] = scaler.fit_transform(covar_df[['age_of_onset']])\n",
    "\n",
    "    if covar_df.age.isna().all():\n",
    "        covar_df.drop(columns=['age'], inplace=True)\n",
    "    else:\n",
    "        covar_df.loc[:,'age'] = scaler.fit_transform(covar_df[['age']])\n",
    "\n",
    "    if covar_df.sex.isna().all():\n",
    "        covar_df.drop(columns=['sex'], inplace=True)\n",
    "\n",
    "    covar_df.drop(columns=['FID'], inplace=True)\n",
    "    covar_df.rename(columns={'GP2sampleID':'sampleid'}, inplace=True)\n",
    "\n",
    "    data_df = dosage_df.merge(covar_df, on='sampleid', how='left').merge(pheno_df, on='sampleid', how='left').set_index('sampleid')\n",
    "\n",
    "    rm_pred = [f'PC{i}' for i in range(1,21)] + ['sex','age_of_onset','age','pheno']\n",
    "\n",
    "    pred_list = [x for x in data_df.columns if x not in rm_pred]\n",
    "    covars_list = [x for x in data_df.columns if x not in pred_list + [f'PC{i}' for i in range(11,21)] + ['pheno']]\n",
    "\n",
    "    results = []\n",
    "    fails = []\n",
    "\n",
    "    for pred in range(len(pred_list)):\n",
    "        pred_name = pred_list[pred]\n",
    "        formula = \"pheno ~ \" + pred_name + \" + \" + ' + '.join(covars_list)\n",
    "\n",
    "        fitted = sm.formula.glm(formula=formula, family=sm.families.Binomial(), data=data_df).fit()\n",
    "        beta_coef  = fitted.params.loc[pred_name]\n",
    "        beta_se  = fitted.bse.loc[pred_name]\n",
    "        p_val = fitted.pvalues.loc[pred_name]\n",
    "\n",
    "        results.append((pred_name, beta_coef, beta_se, p_val))\n",
    "\n",
    "\n",
    "    output = pd.DataFrame(results, columns=('PREDICTOR', 'BETA_COEF', 'BETA_SE','P_VAL'))\n",
    "    output.to_csv(out_path, sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "9d8fab9c-8631-496f-9d04-61735788cda5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22.0"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "198/9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15c41a1d-f86b-4001-968f-613ea87793e7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea2ad12d-590a-4a16-9569-49d447cf37b3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3419a76-ba11-49a0-913d-f34b85c0ba61",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ff40bf6-6316-43f4-bf4c-e8a950006191",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ecfa17e5-a635-487f-8f47-7a6527f24e94",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce5f6a9d-132b-45b4-92f2-32c2dcb73f76",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00ff6fa4-9530-4737-a080-99e5c9afcd56",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8.12 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
